// cometchat-react-ui-kit/dist/Messages/Bubbles/CometChatDocumentBubble/style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.seperatorStyle = exports.messageSubtitleWrapperStyle = exports.messageKitDocumentBlockStyle = exports.messageDocumentTitleStyle = exports.messageDocumentSubtitleStyle = exports.messageDocumentIconStyle = exports.messageDocumentBtnStyle = exports.messageDocumentBtnItemStyle = exports.messageDocumentBlockStyle = exports.messageBtnItemTextStyle = void 0;

var _ = require("../../../");

var messageKitDocumentBlockStyle = function messageKitDocumentBlockStyle(props, theme) {
  return {
    padding: "16px 0",
    display: "inline-block",
    alignSelf: "flex-end",
    flexDirection: "column",
    justifyContent: "center",
    border: props.style.border,
    width: props.style.width,
    borderRadius: props.style.borderRadius,
    background: props.style.background || theme.palette.background[theme.palette.mode],
    boxSizing: "border-box"
  };
};

exports.messageKitDocumentBlockStyle = messageKitDocumentBlockStyle;

var messageDocumentBlockStyle = function messageDocumentBlockStyle(props) {
  return {
    width: props.style.width,
    display: "flex",
    alignItems: "center",
    justifyContent: "center"
  };
};

exports.messageDocumentBlockStyle = messageDocumentBlockStyle;

var messageDocumentIconStyle = function messageDocumentIconStyle(props, theme) {
  return {
    WebkitMask: "url(".concat(props.iconURL, ") no-repeat left center"),
    background: props.style.iconTint || theme.palette.accent700[theme.palette.mode],
    width: "24px",
    height: "24px",
    display: "inline-block"
  };
};

exports.messageDocumentIconStyle = messageDocumentIconStyle;

var messageDocumentTitleStyle = function messageDocumentTitleStyle(props, theme) {
  var _theme$palette;

  return {
    width: "80%",
    margin: "0",
    whiteSpace: "pre-wrap",
    wordBreak: "break-word",
    textAlign: "left",
    font: props.style.titleFont || (0, _.fontHelper)(theme.typography.title2),
    color: props.style.titleColor || (theme === null || theme === void 0 ? void 0 : (_theme$palette = theme.palette) === null || _theme$palette === void 0 ? void 0 : _theme$palette.getAccent())
  };
};

exports.messageDocumentTitleStyle = messageDocumentTitleStyle;

var messageSubtitleWrapperStyle = function messageSubtitleWrapperStyle(props, theme) {
  return {
    padding: "8px 16px",
    display: "inline-block",
    alignSelf: "flex-end",
    flexDirection: "column",
    justifyContent: "center",
    border: props.style.border,
    width: props.style.width,
    borderRadius: props.style.borderRadius,
    background: props.style.background || theme.palette.background[theme.palette.mode],
    boxSizing: "border-box"
  };
};

exports.messageSubtitleWrapperStyle = messageSubtitleWrapperStyle;

var messageDocumentSubtitleStyle = function messageDocumentSubtitleStyle(props, theme) {
  return {
    width: "80%",
    margin: "0",
    whiteSpace: "pre-wrap",
    wordBreak: "break-word",
    textAlign: "left",
    boxSizing: "border-box",
    font: props.style.subTitleFont || (0, _.fontHelper)(theme.typography.subtitle2),
    color: props.style.subTitleColor || theme.palette.accent600[theme.palette.mode]
  };
};

exports.messageDocumentSubtitleStyle = messageDocumentSubtitleStyle;

var seperatorStyle = function seperatorStyle(props, theme) {
  return {
    width: props.style.width,
    height: "1px",
    background: theme.palette.accent100[theme.palette.mode]
  };
};

exports.seperatorStyle = seperatorStyle;

var messageDocumentBtnStyle = function messageDocumentBtnStyle(props) {
  return {
    width: props.style.width,
    padding: "0",
    margin: "0",
    listStyleType: "none"
  };
};

exports.messageDocumentBtnStyle = messageDocumentBtnStyle;

var messageDocumentBtnItemStyle = function messageDocumentBtnItemStyle(props) {
  return {
    margin: "8px 0 0",
    cursor: "pointer",
    display: "flex",
    position: "relative",
    justifyContent: "center",
    alignItems: "center",
    width: props.style.width,
    background: props.style.buttonBackground,
    borderRadius: props.style.borderRadius,
    boxSizing: "border-box"
  };
};

exports.messageDocumentBtnItemStyle = messageDocumentBtnItemStyle;

var messageBtnItemTextStyle = function messageBtnItemTextStyle(props, theme) {
  var _theme$palette2;

  return {
    margin: "0",
    background: "0 0",
    textAlign: "center",
    display: "inline-block",
    width: props.style.width,
    color: props.style.buttonTextColor || (theme === null || theme === void 0 ? void 0 : (_theme$palette2 = theme.palette) === null || _theme$palette2 === void 0 ? void 0 : _theme$palette2.getPrimary()),
    font: props.style.buttonTextFont || (0, _.fontHelper)(theme.typography.title2)
  };
};

exports.messageBtnItemTextStyle = messageBtnItemTextStyle;
