// cometchat-react-ui-kit/dist/Shared/SecondaryComponents/CometChatAvatar/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatAvatar = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Shared = require("../../../Shared/");

var styles = _interopRequireWildcard(require("./style"));

var _default = _interopRequireDefault(require("./resources/default.jpg"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var CometChatAvatar = function CometChatAvatar(props) {
  var name = props.name,
      image = props.image,
      style = props.style;

  var _style = new _Shared.AvatarStyles(style);
  /**
   * Component template scoping
   */


  var getImageView = function getImageView() {
    if (!image && name) {
      return /*#__PURE__*/_react.default.createElement("p", {
        style: styles.textStyle(_style)
      }, name.length >= 2 ? name.substring(0, 2).toUpperCase() : name);
    } else {
      var imageSource;

      if (image) {
        imageSource = image;
      } else imageSource = _default.default;

      return /*#__PURE__*/_react.default.createElement("span", {
        style: styles.getImageStyle(_style, imageSource)
      });
    }
  };
  /**
   * Component template
   */


  return /*#__PURE__*/_react.default.createElement("div", {
    style: styles.getOuterViewStyle(_style)
  }, /*#__PURE__*/_react.default.createElement("span", {
    style: styles.getContainerStyle(_style)
  }, getImageView()));
};
/**
 * Component default props values
 */


exports.CometChatAvatar = CometChatAvatar;
CometChatAvatar.defaultProps = {
  image: "",
  name: "",
  style: {
    textColor: "#ffffff",
    textFont: "bold 80px Inter",
    outerView: "",
    outerViewSpacing: "0px",
    backgroundColor: "#141414",
    backgroundSize: "cover",
    width: "36px",
    height: "36px",
    border: "none",
    borderRadius: "50%"
  }
};
/**
 * Component default props
 */

CometChatAvatar.propTypes = {
  image: _propTypes.default.string,
  name: _propTypes.default.string,
  style: _propTypes.default.object
};
