// @cometchat-pro/react-ui-kit/dist/Users/CometChatUsers/UsersConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UsersConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _UserListConfiguration = require("../CometChatUserList/UserListConfiguration");
var _back = _interopRequireDefault(require("./resources/back.svg"));
var _search = _interopRequireDefault(require("./resources/search.svg"));
var _UsersStyles = require("./UsersStyles");
/**
 * @class UsersConfiguration
 * @description UsersConfiguration class is used for defining the template of Users
 * @param  {string} backButtonIconURL
 * @param  {string} searchIconURL
 * @param  {boolean} showBackButton
 * @param  {string} hideSearch
 * @param  {object} userListConfigurations
 *
 */var UsersConfiguration = /*#__PURE__*/(0, _createClass2["default"])(function UsersConfiguration(_ref) {
  var _ref$backButtonIconUR = _ref.backButtonIconURL,
    backButtonIconURL = _ref$backButtonIconUR === void 0 ? _back["default"] : _ref$backButtonIconUR,
    _ref$searchIconURL = _ref.searchIconURL,
    searchIconURL = _ref$searchIconURL === void 0 ? _search["default"] : _ref$searchIconURL,
    _ref$showBackButton = _ref.showBackButton,
    showBackButton = _ref$showBackButton === void 0 ? false : _ref$showBackButton,
    _ref$hideSearch = _ref.hideSearch,
    hideSearch = _ref$hideSearch === void 0 ? false : _ref$hideSearch,
    _ref$style = _ref.style,
    style = _ref$style === void 0 ? new _UsersStyles.UsersStyles({}) : _ref$style,
    _ref$userListConfigur = _ref.userListConfiguration,
    userListConfiguration = _ref$userListConfigur === void 0 ? new _UserListConfiguration.UserListConfiguration({}) : _ref$userListConfigur;
  (0, _classCallCheck2["default"])(this, UsersConfiguration);
  this.backButtonIconURL = backButtonIconURL;
  this.searchIconURL = searchIconURL;
  this.showBackButton = showBackButton;
  this.hideSearch = hideSearch;
  this.style = new _UsersStyles.UsersStyles(style || {});
  this.userListConfiguration = new _UserListConfiguration.UserListConfiguration(userListConfiguration || {});
});
exports.UsersConfiguration = UsersConfiguration;
