// react-ui-kit-testing/dist/Shared/UtilityComponents/CometChatActionSheet/ActionSheetStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ActionSheetStyles = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _ = require("../..");
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
 * @class ActionSheetStyles
 * @param {String} layoutModeIconTint
 * @param {String} titleFont
 * @param {String} titleColor
 */
var ActionSheetStyles = /*#__PURE__*/function (_BaseStyles) {
  (0, _inherits2.default)(ActionSheetStyles, _BaseStyles);
  var _super = _createSuper(ActionSheetStyles);
  function ActionSheetStyles(_ref) {
    var _this;
    var _ref$layoutModeIconTi = _ref.layoutModeIconTint,
      layoutModeIconTint = _ref$layoutModeIconTi === void 0 ? "" : _ref$layoutModeIconTi,
      _ref$titleFont = _ref.titleFont,
      titleFont = _ref$titleFont === void 0 ? "" : _ref$titleFont,
      _ref$titleColor = _ref.titleColor,
      titleColor = _ref$titleColor === void 0 ? "" : _ref$titleColor,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "295px" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "350px" : _ref$height,
      _ref$background = _ref.background,
      background = _ref$background === void 0 ? "" : _ref$background,
      _ref$border = _ref.border,
      border = _ref$border === void 0 ? "" : _ref$border,
      _ref$borderRadius = _ref.borderRadius,
      borderRadius = _ref$borderRadius === void 0 ? "8px" : _ref$borderRadius,
      _ref$activeBackground = _ref.activeBackground,
      activeBackground = _ref$activeBackground === void 0 ? "" : _ref$activeBackground;
    (0, _classCallCheck2.default)(this, ActionSheetStyles);
    _this = _super.call(this, {
      width: width,
      height: height,
      background: background,
      activeBackground: activeBackground,
      border: border,
      borderRadius: borderRadius
    });
    _this.layoutModeIconTint = layoutModeIconTint;
    _this.titleFont = titleFont;
    _this.titleColor = titleColor;
    return _this;
  }
  return (0, _createClass2.default)(ActionSheetStyles);
}(_.BaseStyles);
exports.ActionSheetStyles = ActionSheetStyles;
