// cometchat-react-ui-kit/dist/Messages/CometChatStickerKeyboard/StickerKeyboardConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StickerKeyboardConfiguration = void 0;

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _ = require("../../../");

var _StickerKeyboardStyles = require("./StickerKeyboardStyles");

/**
 * @class StickerKeyboardConfiguration
 * @description StickerKeyboardConfiguration class is used for defining the StickerKeyboard templates.
 * @param {String} emptyText
 * @param {String} loadingText
 * @param {String} errorText
 * @param {Function} onClick
 * @param {Object} style
 */
var StickerKeyboardConfiguration = /*#__PURE__*/(0, _createClass2.default)(function StickerKeyboardConfiguration(_ref) {
  var _ref$emptyText = _ref.emptyText,
      emptyText = _ref$emptyText === void 0 ? (0, _.localize)("NO_STICKERS_FOUND") : _ref$emptyText,
      _ref$loadingText = _ref.loadingText,
      loadingText = _ref$loadingText === void 0 ? (0, _.localize)("LOADING") : _ref$loadingText,
      _ref$errorText = _ref.errorText,
      errorText = _ref$errorText === void 0 ? (0, _.localize)("SOMETHING_WRONG") : _ref$errorText,
      _ref$onClick = _ref.onClick,
      onClick = _ref$onClick === void 0 ? null : _ref$onClick,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? new _StickerKeyboardStyles.StickerKeyboardStyles({}) : _ref$style;
  (0, _classCallCheck2.default)(this, StickerKeyboardConfiguration);
  this.emptyText = emptyText;
  this.loadingText = loadingText;
  this.errorText = errorText;
  this.onClick = onClick;
  this.style = new _StickerKeyboardStyles.StickerKeyboardStyles(style !== null && style !== void 0 ? style : {});
});
exports.StickerKeyboardConfiguration = StickerKeyboardConfiguration;
