// cometchat-react-ui-kit/dist/Messages/CometChatStickerKeyboard/style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.stickerWrapperStyle = exports.stickerSectionListStyle = exports.stickerMsgTxtStyle = exports.stickerMsgStyle = exports.stickerListStyle = exports.stickerItemStyle = exports.stickerImageStyle = exports.stickerCategoryImageStyle = exports.sectionListItemStyle = void 0;

var _Shared = require("../../Shared");

var stickerWrapperStyle = function stickerWrapperStyle(props, theme) {
  var _props$style;

  return {
    height: props.style.height,
    width: props.style.width,
    border: props.style.border,
    borderRadius: props.style.borderRadius,
    background: (props === null || props === void 0 ? void 0 : (_props$style = props.style) === null || _props$style === void 0 ? void 0 : _props$style.background) || theme.palette.background[theme.palette.mode],
    justifyContent: "center",
    position: "absolute",
    bottom: "90px",
    right: "8px"
  };
};

exports.stickerWrapperStyle = stickerWrapperStyle;

var stickerSectionListStyle = function stickerSectionListStyle(props, theme) {
  var _props$style2;

  return {
    display: "flex",
    overflowX: "auto",
    alignItems: "center",
    justifyContent: "space-between",
    background: (props === null || props === void 0 ? void 0 : (_props$style2 = props.style) === null || _props$style2 === void 0 ? void 0 : _props$style2.categoryBackground) || theme.palette.accent50[theme.palette.mode],
    borderRadius: props.style.borderRadius,
    paddingLeft: "24px"
  };
};

exports.stickerSectionListStyle = stickerSectionListStyle;

var sectionListItemStyle = function sectionListItemStyle() {
  return {
    height: "35px",
    width: "35px",
    cursor: "pointer",
    flexShrink: "0",
    padding: "8px 0px"
  };
};

exports.sectionListItemStyle = sectionListItemStyle;

var stickerListStyle = function stickerListStyle() {
  return {
    height: "100%",
    display: "flex",
    overflowX: "hidden",
    overflowY: "auto",
    flexWrap: "wrap",
    justifyContent: "center",
    alignItems: "center"
  };
};

exports.stickerListStyle = stickerListStyle;

var stickerItemStyle = function stickerItemStyle() {
  return {
    minWidth: "50px",
    minHeight: "50px",
    maxWidth: "70px",
    maxHeight: "70px",
    cursor: "pointer",
    flexShrink: "0",
    margin: "8px"
  };
};

exports.stickerItemStyle = stickerItemStyle;

var stickerMsgStyle = function stickerMsgStyle() {
  return {
    overflow: "hidden",
    width: "100%",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    top: "45%"
  };
};

exports.stickerMsgStyle = stickerMsgStyle;

var stickerMsgTxtStyle = function stickerMsgTxtStyle(props, theme) {
  var _props$style3;

  return {
    margin: "0",
    height: "30px",
    color: (props === null || props === void 0 ? void 0 : (_props$style3 = props.style) === null || _props$style3 === void 0 ? void 0 : _props$style3.textColor) || theme.palette.accent600[theme.palette.mode],
    font: props.style.textFont || (0, _Shared.fontHelper)(theme.typography.title2)
  };
};

exports.stickerMsgTxtStyle = stickerMsgTxtStyle;

var stickerImageStyle = function stickerImageStyle() {
  return {
    width: "100%",
    height: "100%",
    objectFit: "cover"
  };
};

exports.stickerImageStyle = stickerImageStyle;

var stickerCategoryImageStyle = function stickerCategoryImageStyle() {
  return {
    width: "24px",
    height: "24px",
    objectFit: "cover"
  };
};

exports.stickerCategoryImageStyle = stickerCategoryImageStyle;
