// react-ui-kit-testing/dist/Groups/CometChatJoinProtectedGroup/JoinProtectedGroupsStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.JoinProtectedGroupStyles = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _BaseStyles2 = require("../../Shared/Base/BaseStyles");
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
 * @class JoinProtectedGroupStyles
 * @description JoinProtectedGroupStyles class is used for defining the styles for JoinProtectedGroup component
 * boxShadow
 *@param {string} titleTextFont
 *@param {string} titleTextColor
 *@param {string} errorTextFont
 *@param {string} errorTextColor
 *@param {string} passwordTextFont
 *@param {string} passwordTextColor
 *@param {string} passwordPlaceholderTextFont
 *@param {string} passwordPlaceholderTextColor
 *@param {string} passwordInputBackground
 *@param {string} passwordInputBorder
 *@param {string} passwordInputBorderRadius
 *@param {string} passwordInputBoxShadow
 *@param {string} joinButtonTextFont
 *@param {string} joinButtonTextColor
 *@param {string} joinButtonBackground
 *@param {string} joinButtonBorderRadius
 */
var JoinProtectedGroupStyles = /*#__PURE__*/function (_BaseStyles) {
  (0, _inherits2.default)(JoinProtectedGroupStyles, _BaseStyles);
  var _super = _createSuper(JoinProtectedGroupStyles);
  function JoinProtectedGroupStyles(_ref) {
    var _this;
    var _ref$width = _ref.width,
      width = _ref$width === void 0 ? "100%" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "100%" : _ref$height,
      _ref$background = _ref.background,
      background = _ref$background === void 0 ? "" : _ref$background,
      _ref$border = _ref.border,
      border = _ref$border === void 0 ? "" : _ref$border,
      _ref$borderRadius = _ref.borderRadius,
      borderRadius = _ref$borderRadius === void 0 ? "12px" : _ref$borderRadius,
      _ref$boxShadow = _ref.boxShadow,
      boxShadow = _ref$boxShadow === void 0 ? "" : _ref$boxShadow,
      _ref$titleTextFont = _ref.titleTextFont,
      titleTextFont = _ref$titleTextFont === void 0 ? "" : _ref$titleTextFont,
      _ref$titleTextColor = _ref.titleTextColor,
      titleTextColor = _ref$titleTextColor === void 0 ? "" : _ref$titleTextColor,
      _ref$errorTextFont = _ref.errorTextFont,
      errorTextFont = _ref$errorTextFont === void 0 ? "" : _ref$errorTextFont,
      _ref$errorTextColor = _ref.errorTextColor,
      errorTextColor = _ref$errorTextColor === void 0 ? "" : _ref$errorTextColor,
      _ref$passwordTextFont = _ref.passwordTextFont,
      passwordTextFont = _ref$passwordTextFont === void 0 ? "" : _ref$passwordTextFont,
      _ref$passwordTextColo = _ref.passwordTextColor,
      passwordTextColor = _ref$passwordTextColo === void 0 ? "" : _ref$passwordTextColo,
      _ref$passwordPlacehol = _ref.passwordPlaceholderTextFont,
      passwordPlaceholderTextFont = _ref$passwordPlacehol === void 0 ? "" : _ref$passwordPlacehol,
      _ref$passwordPlacehol2 = _ref.passwordPlaceholderTextColor,
      passwordPlaceholderTextColor = _ref$passwordPlacehol2 === void 0 ? "" : _ref$passwordPlacehol2,
      _ref$passwordInputBac = _ref.passwordInputBackground,
      passwordInputBackground = _ref$passwordInputBac === void 0 ? "" : _ref$passwordInputBac,
      _ref$passwordInputBor = _ref.passwordInputBorder,
      passwordInputBorder = _ref$passwordInputBor === void 0 ? "none" : _ref$passwordInputBor,
      _ref$passwordInputBor2 = _ref.passwordInputBorderRadius,
      passwordInputBorderRadius = _ref$passwordInputBor2 === void 0 ? "8px" : _ref$passwordInputBor2,
      _ref$passwordInputBox = _ref.passwordInputBoxShadow,
      passwordInputBoxShadow = _ref$passwordInputBox === void 0 ? "" : _ref$passwordInputBox,
      _ref$joinButtonTextFo = _ref.joinButtonTextFont,
      joinButtonTextFont = _ref$joinButtonTextFo === void 0 ? "" : _ref$joinButtonTextFo,
      _ref$joinButtonTextCo = _ref.joinButtonTextColor,
      joinButtonTextColor = _ref$joinButtonTextCo === void 0 ? "" : _ref$joinButtonTextCo,
      _ref$joinButtonBackgr = _ref.joinButtonBackground,
      joinButtonBackground = _ref$joinButtonBackgr === void 0 ? "" : _ref$joinButtonBackgr,
      _ref$joinButtonBorder = _ref.joinButtonBorderRadius,
      joinButtonBorderRadius = _ref$joinButtonBorder === void 0 ? "8px" : _ref$joinButtonBorder;
    (0, _classCallCheck2.default)(this, JoinProtectedGroupStyles);
    _this = _super.call(this, {
      width: width,
      height: height,
      background: background,
      border: border,
      borderRadius: borderRadius
    });
    _this.boxShadow = boxShadow;
    _this.titleTextFont = titleTextColor;
    _this.titleTextColor = titleTextFont;
    _this.errorTextFont = errorTextFont;
    _this.errorTextColor = errorTextColor;
    _this.passwordTextFont = passwordTextFont;
    _this.passwordTextColor = passwordTextColor;
    _this.passwordPlaceholderTextFont = passwordPlaceholderTextFont;
    _this.passwordPlaceholderTextColor = passwordPlaceholderTextColor;
    _this.passwordInputBackground = passwordInputBackground;
    _this.passwordInputBorder = passwordInputBorder;
    _this.passwordInputBorderRadius = passwordInputBorderRadius;
    _this.passwordInputBoxShadow = passwordInputBoxShadow;
    _this.joinButtonTextFont = joinButtonTextFont;
    _this.joinButtonTextColor = joinButtonTextColor;
    _this.joinButtonBackground = joinButtonBackground;
    _this.joinButtonBorderRadius = joinButtonBorderRadius;
    return _this;
  }
  return (0, _createClass2.default)(JoinProtectedGroupStyles);
}(_BaseStyles2.BaseStyles);
exports.JoinProtectedGroupStyles = JoinProtectedGroupStyles;
