// react-ui-kit-testing/dist/Groups/CometChatJoinProtectedGroup/style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.messageHeaderStyle = exports.joinGroupTitleStyle = exports.joinGroupProtectedWrapperStyle = exports.joinGroupPasswordInputStyle = exports.joinGroupPasswordInputContainerStyle = exports.joinGroupContainerStyle = exports.joinGroupButtonStyle = exports.errorTextStyle = exports.errorMessageBoxStyle = exports.errorIconStyle = exports.errorContainerStyle = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _ = require("../..");
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
var joinGroupContainerStyle = function joinGroupContainerStyle(style, theme) {
  var _theme$pallete;
  return {
    margin: "auto",
    width: style === null || style === void 0 ? void 0 : style.width,
    height: style === null || style === void 0 ? void 0 : style.height,
    borderRadius: style === null || style === void 0 ? void 0 : style.borderRadius,
    border: style.border,
    zIndex: "2",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    background: (style === null || style === void 0 ? void 0 : style.background) || (theme === null || theme === void 0 ? void 0 : (_theme$pallete = theme.pallete) === null || _theme$pallete === void 0 ? void 0 : _theme$pallete.getBackground())
  };
};
exports.joinGroupContainerStyle = joinGroupContainerStyle;
var joinGroupTitleStyle = function joinGroupTitleStyle(style, theme) {
  var _theme$typography, _theme$palette;
  return {
    font: (style === null || style === void 0 ? void 0 : style.titleTextFont) || (0, _.fontHelper)(theme === null || theme === void 0 ? void 0 : (_theme$typography = theme.typography) === null || _theme$typography === void 0 ? void 0 : _theme$typography.heading),
    color: (style === null || style === void 0 ? void 0 : style.titleTextColor) || (theme === null || theme === void 0 ? void 0 : (_theme$palette = theme.palette) === null || _theme$palette === void 0 ? void 0 : _theme$palette.getAccent700())
  };
};
exports.joinGroupTitleStyle = joinGroupTitleStyle;
var joinGroupPasswordInputStyle = function joinGroupPasswordInputStyle(style, theme) {
  var _theme$typography2, _theme$palette2, _theme$palette3, _theme$palette4;
  return {
    width: style === null || style === void 0 ? void 0 : style.width,
    height: "46px",
    color: (style === null || style === void 0 ? void 0 : style.passwordTextFont) || (0, _.fontHelper)(theme === null || theme === void 0 ? void 0 : (_theme$typography2 = theme.typography) === null || _theme$typography2 === void 0 ? void 0 : _theme$typography2.subtitle2),
    font: (style === null || style === void 0 ? void 0 : style.passwordTextColor) || (theme === null || theme === void 0 ? void 0 : (_theme$palette2 = theme.palette) === null || _theme$palette2 === void 0 ? void 0 : _theme$palette2.getAccent900()),
    background: (style === null || style === void 0 ? void 0 : style.passwordInputBackground) || (theme === null || theme === void 0 ? void 0 : (_theme$palette3 = theme.palette) === null || _theme$palette3 === void 0 ? void 0 : _theme$palette3.getAccent50()),
    border: (style === null || style === void 0 ? void 0 : style.passwordInputBorder) || "none",
    borderRadius: (style === null || style === void 0 ? void 0 : style.passwordInputBorderRadius) || "8px",
    boxShadow: (style === null || style === void 0 ? void 0 : style.passwordInputBoxShadow) || "".concat(theme === null || theme === void 0 ? void 0 : (_theme$palette4 = theme.palette) === null || _theme$palette4 === void 0 ? void 0 : _theme$palette4.getAccent50(), " 0px 0px 0px 1px"),
    margin: "38px 0px 16px 0px",
    outline: "none",
    padding: "6px 12px"
  };
};
exports.joinGroupPasswordInputStyle = joinGroupPasswordInputStyle;
var joinGroupButtonStyle = function joinGroupButtonStyle(style, theme) {
  var _theme$typography3, _theme$palette5, _theme$palette6;
  return {
    width: style === null || style === void 0 ? void 0 : style.width,
    height: "46px",
    font: (style === null || style === void 0 ? void 0 : style.joinButtonTextFont) || (0, _.fontHelper)(theme === null || theme === void 0 ? void 0 : (_theme$typography3 = theme.typography) === null || _theme$typography3 === void 0 ? void 0 : _theme$typography3.title2),
    color: (style === null || style === void 0 ? void 0 : style.joinButtonTextColor) || (theme === null || theme === void 0 ? void 0 : (_theme$palette5 = theme.palette) === null || _theme$palette5 === void 0 ? void 0 : _theme$palette5.getAccent900("light")),
    background: (style === null || style === void 0 ? void 0 : style.joinButtonBackground) || (theme === null || theme === void 0 ? void 0 : (_theme$palette6 = theme.palette) === null || _theme$palette6 === void 0 ? void 0 : _theme$palette6.getPrimary()),
    borderRadius: style === null || style === void 0 ? void 0 : style.joinButtonBorderRadius,
    border: "none"
  };
};
exports.joinGroupButtonStyle = joinGroupButtonStyle;
var errorContainerStyle = function errorContainerStyle(style, theme) {
  return {
    width: style === null || style === void 0 ? void 0 : style.width,
    height: "60px",
    margin: "16px 0px 32px 0px",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    background: "rgba(255, 59, 48, 0.1)",
    //not in theme
    border: "none",
    borderRadius: "8px"
  };
};
exports.errorContainerStyle = errorContainerStyle;
var errorIconStyle = function errorIconStyle(theme) {
  var _theme$palette7;
  return {
    iconBackground: (theme === null || theme === void 0 ? void 0 : (_theme$palette7 = theme.palette) === null || _theme$palette7 === void 0 ? void 0 : _theme$palette7.getError()) || "RGB(255, 59, 48)",
    iconTint: "white",
    iconBackgroundWidth: "38px",
    iconBackgroundHeight: "38px",
    iconWidth: "24px",
    iconHeight: "24px",
    margin: "0px 12px"
  };
};
exports.errorIconStyle = errorIconStyle;
var errorTextStyle = function errorTextStyle(style, theme) {
  var _theme$palette8, _theme$typography4;
  return {
    color: (style === null || style === void 0 ? void 0 : style.errorTextColor) || (theme === null || theme === void 0 ? void 0 : (_theme$palette8 = theme.palette) === null || _theme$palette8 === void 0 ? void 0 : _theme$palette8.getError()),
    font: (style === null || style === void 0 ? void 0 : style.errorTextFont) || (0, _.fontHelper)(theme === null || theme === void 0 ? void 0 : (_theme$typography4 = theme.typography) === null || _theme$typography4 === void 0 ? void 0 : _theme$typography4.text1)
  };
};
exports.errorTextStyle = errorTextStyle;
var joinGroupPasswordInputContainerStyle = function joinGroupPasswordInputContainerStyle(style) {
  return {
    width: style === null || style === void 0 ? void 0 : style.width,
    height: "auto",
    margin: "0",
    padding: "0"
  };
};
exports.joinGroupPasswordInputContainerStyle = joinGroupPasswordInputContainerStyle;
var errorMessageBoxStyle = function errorMessageBoxStyle(style) {
  return {
    width: style === null || style === void 0 ? void 0 : style.width,
    height: "auto",
    margin: "0",
    padding: "0"
  };
};
exports.errorMessageBoxStyle = errorMessageBoxStyle;
var joinGroupProtectedWrapperStyle = function joinGroupProtectedWrapperStyle() {
  return {
    margin: "auto",
    width: "343px",
    height: "100%",
    borderRadius: "8px",
    zIndex: "4",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center"
  };
};
exports.joinGroupProtectedWrapperStyle = joinGroupProtectedWrapperStyle;
var messageHeaderStyle = function messageHeaderStyle(theme) {
  var _theme$palette9, _theme$palette10;
  return _objectSpread(_objectSpread({}, new _.BaseStyles({
    width: "100%",
    height: "auto",
    background: theme === null || theme === void 0 ? void 0 : (_theme$palette9 = theme.palette) === null || _theme$palette9 === void 0 ? void 0 : _theme$palette9.getBackground(),
    border: "none",
    borderRadius: "none",
    activeBackground: "none"
  })), {}, {
    backButtonIconTint: theme === null || theme === void 0 ? void 0 : (_theme$palette10 = theme.palette) === null || _theme$palette10 === void 0 ? void 0 : _theme$palette10.getPrimary()
  });
};
exports.messageHeaderStyle = messageHeaderStyle;
