// react-ui-kit-testing/dist/Groups/CometChatGroupList/GroupListStyle.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GroupListStyle = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _Shared = require("../../Shared");
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
 * @class GroupListStyle
 * @description GroupListStyle class is used for defining the GroupList template.
 * @param {any} width
 * @param {any} height
 * @param {any} background
 * @param {string} border
 * @param {string} borderRadius
 * @param {string} loadingIconTint
 * @param {string} emptyTextFont
 * @param {string} emptyTextColor
 * @param {string} errorTextFont
 * @param {string} errorTextColor
 */
var GroupListStyle = /*#__PURE__*/function (_BaseStyles) {
  (0, _inherits2.default)(GroupListStyle, _BaseStyles);
  var _super = _createSuper(GroupListStyle);
  function GroupListStyle(_ref) {
    var _this;
    var _ref$width = _ref.width,
      width = _ref$width === void 0 ? "100%" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "100%" : _ref$height,
      _ref$background = _ref.background,
      background = _ref$background === void 0 ? "" : _ref$background,
      _ref$border = _ref.border,
      border = _ref$border === void 0 ? "0 solid black" : _ref$border,
      _ref$borderRadius = _ref.borderRadius,
      borderRadius = _ref$borderRadius === void 0 ? "0" : _ref$borderRadius,
      _ref$loadingIconTint = _ref.loadingIconTint,
      loadingIconTint = _ref$loadingIconTint === void 0 ? "black" : _ref$loadingIconTint,
      _ref$emptyTextFont = _ref.emptyTextFont,
      emptyTextFont = _ref$emptyTextFont === void 0 ? "" : _ref$emptyTextFont,
      _ref$emptyTextColor = _ref.emptyTextColor,
      emptyTextColor = _ref$emptyTextColor === void 0 ? "#141414" : _ref$emptyTextColor,
      _ref$errorTextFont = _ref.errorTextFont,
      errorTextFont = _ref$errorTextFont === void 0 ? "" : _ref$errorTextFont,
      _ref$errorTextColor = _ref.errorTextColor,
      errorTextColor = _ref$errorTextColor === void 0 ? "#141414" : _ref$errorTextColor;
    (0, _classCallCheck2.default)(this, GroupListStyle);
    _this = _super.call(this, {
      width: width,
      height: height,
      background: background,
      border: border,
      borderRadius: borderRadius
    });
    _this.loadingIconTint = loadingIconTint;
    _this.emptyTextFont = emptyTextFont;
    _this.emptyTextColor = emptyTextColor;
    _this.errorTextFont = errorTextFont;
    _this.errorTextColor = errorTextColor;
    return _this;
  }
  return (0, _createClass2.default)(GroupListStyle);
}(_Shared.BaseStyles);
exports.GroupListStyle = GroupListStyle;
