// react-ui-kit-testing/dist/Groups/CometChatGroupList/GroupListConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GroupListConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _Shared = require("../../Shared");
var _spinner = _interopRequireDefault(require("./resources/spinner.svg"));
var _GroupListStyle = require("./GroupListStyle");
/**
 * @class GroupListConfiguration
 * @description GroupListConfiguration class is used for defining the GroupList template.
 * @param {number} limit
 * @param {string} searchKeyword
 * @param {bool} joinedOnly
 * @param {array} tags
 * @param {object} style
 * @param {null} customView
 * @param {null} hideError
 * @param {string} loadingIconURL
 * @param {object} dataItemConfiguration
 */
var GroupListConfiguration = /*#__PURE__*/(0, _createClass2.default)(function GroupListConfiguration(_ref) {
  var _ref$limit = _ref.limit,
    limit = _ref$limit === void 0 ? 30 : _ref$limit,
    _ref$searchKeyword = _ref.searchKeyword,
    searchKeyword = _ref$searchKeyword === void 0 ? "" : _ref$searchKeyword,
    _ref$joinedOnly = _ref.joinedOnly,
    joinedOnly = _ref$joinedOnly === void 0 ? false : _ref$joinedOnly,
    _ref$tags = _ref.tags,
    tags = _ref$tags === void 0 ? null : _ref$tags,
    _ref$loadingIconURL = _ref.loadingIconURL,
    loadingIconURL = _ref$loadingIconURL === void 0 ? _spinner.default : _ref$loadingIconURL,
    _ref$style = _ref.style,
    style = _ref$style === void 0 ? new _GroupListStyle.GroupListStyle({}) : _ref$style,
    _ref$customView = _ref.customView,
    customView = _ref$customView === void 0 ? new _Shared.CustomView({
      loading: "",
      empty: "",
      error: ""
    }) : _ref$customView,
    _ref$hideError = _ref.hideError,
    hideError = _ref$hideError === void 0 ? false : _ref$hideError,
    _ref$dataItemConfigur = _ref.dataItemConfiguration,
    dataItemConfiguration = _ref$dataItemConfigur === void 0 ? new _Shared.DataItemConfiguration({}) : _ref$dataItemConfigur;
  (0, _classCallCheck2.default)(this, GroupListConfiguration);
  this.limit = limit;
  this.searchKeyword = searchKeyword;
  this.joinedOnly = joinedOnly;
  this.tags = tags;
  this.loadingIconURL = loadingIconURL;
  this.style = new _GroupListStyle.GroupListStyle(style || {});
  this.customView = new _GroupListStyle.GroupListStyle(customView || {});
  this.hideError = hideError;
  this.dataItemConfiguration = new _Shared.DataItemConfiguration(dataItemConfiguration || {});
});
exports.GroupListConfiguration = GroupListConfiguration;
