// @cometchat-pro/react-ui-kit/dist/Shared/SecondaryComponents/CometChatBadgeCount/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatBadgeCount = void 0;
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _Shared = require("../../../Shared/");
var CometChatBadgeCount = function CometChatBadgeCount(props) {
  /**
   * Component template scoping
   */
  if (props !== null && props !== void 0 && props.count) {
    var getStyle = function getStyle() {
      var _props$style, _props$style2, _props$style3, _props$style4, _props$style5, _props$style6, _props$style7, _props$style8;
      return {
        border: props === null || props === void 0 ? void 0 : (_props$style = props.style) === null || _props$style === void 0 ? void 0 : _props$style.border,
        borderRadius: props === null || props === void 0 ? void 0 : (_props$style2 = props.style) === null || _props$style2 === void 0 ? void 0 : _props$style2.borderRadius,
        backgroundColor: props === null || props === void 0 ? void 0 : (_props$style3 = props.style) === null || _props$style3 === void 0 ? void 0 : _props$style3.background,
        color: props === null || props === void 0 ? void 0 : (_props$style4 = props.style) === null || _props$style4 === void 0 ? void 0 : _props$style4.textColor,
        font: props === null || props === void 0 ? void 0 : (_props$style5 = props.style) === null || _props$style5 === void 0 ? void 0 : _props$style5.textFont,
        minWidth: props === null || props === void 0 ? void 0 : (_props$style6 = props.style) === null || _props$style6 === void 0 ? void 0 : _props$style6.width,
        height: props === null || props === void 0 ? void 0 : (_props$style7 = props.style) === null || _props$style7 === void 0 ? void 0 : _props$style7.height,
        lineHeight: props === null || props === void 0 ? void 0 : (_props$style8 = props.style) === null || _props$style8 === void 0 ? void 0 : _props$style8.height,
        textAlign: "center",
        display: "inline-block",
        padding: "0px 10px"
      };
    };
    return /*#__PURE__*/_react["default"].createElement("div", {
      style: getStyle()
    }, (props === null || props === void 0 ? void 0 : props.count) >= 100 ? "99+" : props === null || props === void 0 ? void 0 : props.count);
  }
  return null;
};

/**
 * Component default props values
 */
exports.CometChatBadgeCount = CometChatBadgeCount;
CometChatBadgeCount.defaultProps = {
  count: 0,
  style: {
    textFont: "600 12px Inter",
    textColor: "#ffffff",
    width: "fit-content",
    height: "20px",
    background: "rgba(51, 153, 255, 1)",
    border: "none",
    borderRadius: "11px"
  }
};

/**
 * Component default props
 */
CometChatBadgeCount.propTypes = {
  count: _propTypes["default"].number,
  style: _propTypes["default"].object
};
