// cometchat-react-ui-kit/dist/Messages/CometChatSmartReplies/SmartRepliesConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SmartRepliesConfiguration = void 0;

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _close = _interopRequireDefault(require("../../Messages/CometChatSmartReplies/resources/close.svg"));

/**
 * @class SmartRepliesConfiguration
 * @description SmartRepliesConfiguration class is used for defining the smart Replies Template
 * @param {String} customOutgoingMessageSound
 * @param {String} enableSoundForMessages
 * @param {Function} onClick
 * @param {Function} onClose
 * @param {String} closeIconURL
 */
var SmartRepliesConfiguration = /*#__PURE__*/(0, _createClass2.default)(function SmartRepliesConfiguration(_ref) {
  var _ref$customOutgoingMe = _ref.customOutgoingMessageSound,
      customOutgoingMessageSound = _ref$customOutgoingMe === void 0 ? null : _ref$customOutgoingMe,
      _ref$enableSoundForMe = _ref.enableSoundForMessages,
      enableSoundForMessages = _ref$enableSoundForMe === void 0 ? true : _ref$enableSoundForMe,
      _ref$onClick = _ref.onClick,
      onClick = _ref$onClick === void 0 ? null : _ref$onClick,
      _ref$onClose = _ref.onClose,
      onClose = _ref$onClose === void 0 ? null : _ref$onClose,
      _ref$closeIconURL = _ref.closeIconURL,
      closeIconURL = _ref$closeIconURL === void 0 ? _close.default : _ref$closeIconURL;
  (0, _classCallCheck2.default)(this, SmartRepliesConfiguration);
  this.customOutgoingMessageSound = customOutgoingMessageSound;
  this.enableSoundForMessages = enableSoundForMessages;
  this.onClick = onClick;
  this.onClose = onClose;
  this.closeIconURL = closeIconURL;
});
exports.SmartRepliesConfiguration = SmartRepliesConfiguration;
