// cometchat-react-ui-kit/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AvatarConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.AvatarConfiguration;
  }
});
Object.defineProperty(exports, "BadgeCountConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.BadgeCountConfiguration;
  }
});
Object.defineProperty(exports, "BaseStyles", {
  enumerable: true,
  get: function get() {
    return _Shared.BaseStyles;
  }
});
Object.defineProperty(exports, "CometChatActionSheet", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatActionSheet;
  }
});
Object.defineProperty(exports, "CometChatActionSheetItem", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatActionSheetItem;
  }
});
Object.defineProperty(exports, "CometChatAudioBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatAudioBubble;
  }
});
Object.defineProperty(exports, "CometChatAvatar", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatAvatar;
  }
});
Object.defineProperty(exports, "CometChatBackdrop", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatBackdrop;
  }
});
Object.defineProperty(exports, "CometChatBadgeCount", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatBadgeCount;
  }
});
Object.defineProperty(exports, "CometChatConfirmDialog", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatConfirmDialog;
  }
});
Object.defineProperty(exports, "CometChatConversationEvents", {
  enumerable: true,
  get: function get() {
    return _Chats.CometChatConversationEvents;
  }
});
Object.defineProperty(exports, "CometChatConversationList", {
  enumerable: true,
  get: function get() {
    return _Chats.CometChatConversationList;
  }
});
Object.defineProperty(exports, "CometChatConversationListItem", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatConversationListItem;
  }
});
Object.defineProperty(exports, "CometChatConversations", {
  enumerable: true,
  get: function get() {
    return _Chats.CometChatConversations;
  }
});
Object.defineProperty(exports, "CometChatConversationsWithMessages", {
  enumerable: true,
  get: function get() {
    return _Chats.CometChatConversationsWithMessages;
  }
});
Object.defineProperty(exports, "CometChatCreatePoll", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatCreatePoll;
  }
});
Object.defineProperty(exports, "CometChatCreatePollOptions", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatCreatePollOptions;
  }
});
Object.defineProperty(exports, "CometChatCustomMessageTypes", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatCustomMessageTypes;
  }
});
Object.defineProperty(exports, "CometChatDataItem", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatDataItem;
  }
});
Object.defineProperty(exports, "CometChatDate", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatDate;
  }
});
Object.defineProperty(exports, "CometChatDecoratorMessage", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatDecoratorMessage;
  }
});
Object.defineProperty(exports, "CometChatDeletedMessageBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatDeletedMessageBubble;
  }
});
Object.defineProperty(exports, "CometChatDivider", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatDivider;
  }
});
Object.defineProperty(exports, "CometChatDocumentBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatDocumentBubble;
  }
});
Object.defineProperty(exports, "CometChatEmojiKeyboard", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatEmojiKeyboard;
  }
});
Object.defineProperty(exports, "CometChatFileBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatFileBubble;
  }
});
Object.defineProperty(exports, "CometChatGroupActionBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatGroupActionBubble;
  }
});
Object.defineProperty(exports, "CometChatGroupEvents", {
  enumerable: true,
  get: function get() {
    return _Groups.CometChatGroupEvents;
  }
});
Object.defineProperty(exports, "CometChatGroupList", {
  enumerable: true,
  get: function get() {
    return _Groups.CometChatGroupList;
  }
});
Object.defineProperty(exports, "CometChatGroupListItem", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatGroupListItem;
  }
});
Object.defineProperty(exports, "CometChatGroups", {
  enumerable: true,
  get: function get() {
    return _Groups.CometChatGroups;
  }
});
Object.defineProperty(exports, "CometChatGroupsWithMessages", {
  enumerable: true,
  get: function get() {
    return _Groups.CometChatGroupsWithMessages;
  }
});
Object.defineProperty(exports, "CometChatImageBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatImageBubble;
  }
});
Object.defineProperty(exports, "CometChatImageViewer", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatImageViewer;
  }
});
Object.defineProperty(exports, "CometChatLinkPreview", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatLinkPreview;
  }
});
Object.defineProperty(exports, "CometChatListBase", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatListBase;
  }
});
Object.defineProperty(exports, "CometChatListItem", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatListItem;
  }
});
Object.defineProperty(exports, "CometChatLiveReactions", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatLiveReactions;
  }
});
Object.defineProperty(exports, "CometChatLocalize", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatLocalize;
  }
});
Object.defineProperty(exports, "CometChatMenuList", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatMenuList;
  }
});
Object.defineProperty(exports, "CometChatMessageBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessageBubble;
  }
});
Object.defineProperty(exports, "CometChatMessageCategories", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessageCategories;
  }
});
Object.defineProperty(exports, "CometChatMessageComposer", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessageComposer;
  }
});
Object.defineProperty(exports, "CometChatMessageEvents", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessageEvents;
  }
});
Object.defineProperty(exports, "CometChatMessageHeader", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessageHeader;
  }
});
Object.defineProperty(exports, "CometChatMessageList", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessageList;
  }
});
Object.defineProperty(exports, "CometChatMessageOptions", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessageOptions;
  }
});
Object.defineProperty(exports, "CometChatMessagePreview", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessagePreview;
  }
});
Object.defineProperty(exports, "CometChatMessageReactions", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessageReactions;
  }
});
Object.defineProperty(exports, "CometChatMessageReceipt", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatMessageReceipt;
  }
});
Object.defineProperty(exports, "CometChatMessageReceiverType", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessageReceiverType;
  }
});
Object.defineProperty(exports, "CometChatMessageTemplate", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessageTemplate;
  }
});
Object.defineProperty(exports, "CometChatMessageTypes", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessageTypes;
  }
});
Object.defineProperty(exports, "CometChatMessages", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatMessages;
  }
});
Object.defineProperty(exports, "CometChatNewMessageIndicator", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatNewMessageIndicator;
  }
});
Object.defineProperty(exports, "CometChatPlaceholderBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatPlaceholderBubble;
  }
});
Object.defineProperty(exports, "CometChatPollBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatPollBubble;
  }
});
Object.defineProperty(exports, "CometChatPollOptionBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatPollOptionBubble;
  }
});
Object.defineProperty(exports, "CometChatPopover", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatPopover;
  }
});
Object.defineProperty(exports, "CometChatSmartReplies", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatSmartReplies;
  }
});
Object.defineProperty(exports, "CometChatSoundManager", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatSoundManager;
  }
});
Object.defineProperty(exports, "CometChatStatusIndicator", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatStatusIndicator;
  }
});
Object.defineProperty(exports, "CometChatStickerBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatStickerBubble;
  }
});
Object.defineProperty(exports, "CometChatStickerKeyboard", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatStickerKeyboard;
  }
});
Object.defineProperty(exports, "CometChatTextBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatTextBubble;
  }
});
Object.defineProperty(exports, "CometChatTheme", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatTheme;
  }
});
Object.defineProperty(exports, "CometChatThreadReplies", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatThreadReplies;
  }
});
Object.defineProperty(exports, "CometChatUserEvents", {
  enumerable: true,
  get: function get() {
    return _Users.CometChatUserEvents;
  }
});
Object.defineProperty(exports, "CometChatUserList", {
  enumerable: true,
  get: function get() {
    return _Users.CometChatUserList;
  }
});
Object.defineProperty(exports, "CometChatUserListItem", {
  enumerable: true,
  get: function get() {
    return _Shared.CometChatUserListItem;
  }
});
Object.defineProperty(exports, "CometChatUsers", {
  enumerable: true,
  get: function get() {
    return _Users.CometChatUsers;
  }
});
Object.defineProperty(exports, "CometChatUsersWithMessages", {
  enumerable: true,
  get: function get() {
    return _Users.CometChatUsersWithMessages;
  }
});
Object.defineProperty(exports, "CometChatVideoBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatVideoBubble;
  }
});
Object.defineProperty(exports, "CometChatWhiteboardBubble", {
  enumerable: true,
  get: function get() {
    return _Messages.CometChatWhiteboardBubble;
  }
});
Object.defineProperty(exports, "ConversationListConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.ConversationListConfiguration;
  }
});
Object.defineProperty(exports, "ConversationListItemConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.ConversationListItemConfiguration;
  }
});
Object.defineProperty(exports, "ConversationOptionConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.ConversationOptionConstants;
  }
});
Object.defineProperty(exports, "ConversationTypeConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.ConversationTypeConstants;
  }
});
Object.defineProperty(exports, "ConversationsConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.ConversationsConfiguration;
  }
});
Object.defineProperty(exports, "ConversationsWithMessagesConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.ConversationsWithMessagesConfiguration;
  }
});
Object.defineProperty(exports, "CustomView", {
  enumerable: true,
  get: function get() {
    return _Shared.CustomView;
  }
});
Object.defineProperty(exports, "DateConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.DateConfiguration;
  }
});
Object.defineProperty(exports, "ExtensionConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.ExtensionConstants;
  }
});
Object.defineProperty(exports, "ExtensionURLs", {
  enumerable: true,
  get: function get() {
    return _Shared.ExtensionURLs;
  }
});
Object.defineProperty(exports, "GroupListItemConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.GroupListItemConfiguration;
  }
});
Object.defineProperty(exports, "GroupMemberOptionConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.GroupMemberOptionConstants;
  }
});
Object.defineProperty(exports, "GroupMemberScope", {
  enumerable: true,
  get: function get() {
    return _Shared.GroupMemberScope;
  }
});
Object.defineProperty(exports, "GroupOptionConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.GroupOptionConstants;
  }
});
Object.defineProperty(exports, "GroupTypeConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.GroupTypeConstants;
  }
});
Object.defineProperty(exports, "GroupsConfiguration", {
  enumerable: true,
  get: function get() {
    return _Groups.GroupsConfiguration;
  }
});
Object.defineProperty(exports, "GroupsWithMessagesConfiguration", {
  enumerable: true,
  get: function get() {
    return _Groups.GroupsWithMessagesConfiguration;
  }
});
Object.defineProperty(exports, "InputData", {
  enumerable: true,
  get: function get() {
    return _Shared.InputData;
  }
});
Object.defineProperty(exports, "ListBaseConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.ListBaseConfiguration;
  }
});
Object.defineProperty(exports, "LiveReactionConfiguration", {
  enumerable: true,
  get: function get() {
    return _Messages.LiveReactionConfiguration;
  }
});
Object.defineProperty(exports, "MessageBubbleAlignmentConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.MessageBubbleAlignmentConstants;
  }
});
Object.defineProperty(exports, "MessageCategoryConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.MessageCategoryConstants;
  }
});
Object.defineProperty(exports, "MessageComposerConfiguration", {
  enumerable: true,
  get: function get() {
    return _Messages.MessageComposerConfiguration;
  }
});
Object.defineProperty(exports, "MessageHeaderConfiguration", {
  enumerable: true,
  get: function get() {
    return _Messages.MessageHeaderConfiguration;
  }
});
Object.defineProperty(exports, "MessageInputData", {
  enumerable: true,
  get: function get() {
    return _Messages.MessageInputData;
  }
});
Object.defineProperty(exports, "MessageListAlignmentConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.MessageListAlignmentConstants;
  }
});
Object.defineProperty(exports, "MessageOptionConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.MessageOptionConstants;
  }
});
Object.defineProperty(exports, "MessageOptionForConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.MessageOptionForConstants;
  }
});
Object.defineProperty(exports, "MessageReceiptConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.MessageReceiptConfiguration;
  }
});
Object.defineProperty(exports, "MessageStatusConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.MessageStatusConstants;
  }
});
Object.defineProperty(exports, "MessageTimeAlignmentConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.MessageTimeAlignmentConstants;
  }
});
Object.defineProperty(exports, "MessageTypeConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.MessageTypeConstants;
  }
});
Object.defineProperty(exports, "MessagesConfiguration", {
  enumerable: true,
  get: function get() {
    return _Messages.MessagesConfiguration;
  }
});
Object.defineProperty(exports, "MetadataConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.MetadataConstants;
  }
});
Object.defineProperty(exports, "Palette", {
  enumerable: true,
  get: function get() {
    return _Shared.Palette;
  }
});
Object.defineProperty(exports, "ReceiverTypeConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.ReceiverTypeConstants;
  }
});
Object.defineProperty(exports, "StatusIndicatorConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.StatusIndicatorConfiguration;
  }
});
Object.defineProperty(exports, "Typography", {
  enumerable: true,
  get: function get() {
    return _Shared.Typography;
  }
});
Object.defineProperty(exports, "UserListConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.UserListConfiguration;
  }
});
Object.defineProperty(exports, "UserListItemConfiguration", {
  enumerable: true,
  get: function get() {
    return _Shared.UserListItemConfiguration;
  }
});
Object.defineProperty(exports, "UserOptionConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.UserOptionConstants;
  }
});
Object.defineProperty(exports, "UserStatusConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.UserStatusConstants;
  }
});
Object.defineProperty(exports, "UsersConfiguration", {
  enumerable: true,
  get: function get() {
    return _Users.UsersConfiguration;
  }
});
Object.defineProperty(exports, "UsersWithMessagesConfiguration", {
  enumerable: true,
  get: function get() {
    return _Users.UsersWithMessagesConfiguration;
  }
});
Object.defineProperty(exports, "bytesToSize", {
  enumerable: true,
  get: function get() {
    return _Shared.bytesToSize;
  }
});
Object.defineProperty(exports, "conversationType", {
  enumerable: true,
  get: function get() {
    return _Chats.conversationType;
  }
});
Object.defineProperty(exports, "emailPattern", {
  enumerable: true,
  get: function get() {
    return _Shared.emailPattern;
  }
});
Object.defineProperty(exports, "fontHelper", {
  enumerable: true,
  get: function get() {
    return _Shared.fontHelper;
  }
});
Object.defineProperty(exports, "getDefaultTypes", {
  enumerable: true,
  get: function get() {
    return _Messages.getDefaultTypes;
  }
});
Object.defineProperty(exports, "getExtensionsData", {
  enumerable: true,
  get: function get() {
    return _Shared.getExtensionsData;
  }
});
Object.defineProperty(exports, "getMetadataByKey", {
  enumerable: true,
  get: function get() {
    return _Shared.getMetadataByKey;
  }
});
Object.defineProperty(exports, "localize", {
  enumerable: true,
  get: function get() {
    return _Shared.localize;
  }
});
Object.defineProperty(exports, "messageAlignment", {
  enumerable: true,
  get: function get() {
    return _Messages.messageAlignment;
  }
});
Object.defineProperty(exports, "messageBubbleAlignment", {
  enumerable: true,
  get: function get() {
    return _Messages.messageBubbleAlignment;
  }
});
Object.defineProperty(exports, "messageConstants", {
  enumerable: true,
  get: function get() {
    return _Shared.messageConstants;
  }
});
Object.defineProperty(exports, "messageTimeAlignment", {
  enumerable: true,
  get: function get() {
    return _Messages.messageTimeAlignment;
  }
});
Object.defineProperty(exports, "phoneNumPattern", {
  enumerable: true,
  get: function get() {
    return _Shared.phoneNumPattern;
  }
});
Object.defineProperty(exports, "urlPattern", {
  enumerable: true,
  get: function get() {
    return _Shared.urlPattern;
  }
});
Object.defineProperty(exports, "wordBoundary", {
  enumerable: true,
  get: function get() {
    return _Shared.wordBoundary;
  }
});

var _Shared = require("./Shared");

var _Chats = require("./Chats");

var _Messages = require("./Messages");

var _Users = require("./Users");

var _Groups = require("./Groups");
