// react-ui-kit-testing/dist/Shared/SDKDerivedComponents/CometChatDataItem/DataItemStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DataItemStyles = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _ = require("../..");
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
 * @class DataItemStyles
 * @param {String} titleFont
 * @param {String} titleColor
 * @param {String} outerView
 * @param {String} outerViewSpacing
 */
var DataItemStyles = /*#__PURE__*/function (_BaseStyles) {
  (0, _inherits2.default)(DataItemStyles, _BaseStyles);
  var _super = _createSuper(DataItemStyles);
  function DataItemStyles(_ref) {
    var _this;
    var _ref$titleColor = _ref.titleColor,
      titleColor = _ref$titleColor === void 0 ? "" : _ref$titleColor,
      _ref$titleFont = _ref.titleFont,
      titleFont = _ref$titleFont === void 0 ? "" : _ref$titleFont,
      _ref$subTitleColor = _ref.subTitleColor,
      subTitleColor = _ref$subTitleColor === void 0 ? "" : _ref$subTitleColor,
      _ref$subTitleFont = _ref.subTitleFont,
      subTitleFont = _ref$subTitleFont === void 0 ? "" : _ref$subTitleFont,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "100%" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "100%" : _ref$height,
      background = _ref.background,
      activeBackground = _ref.activeBackground,
      _ref$border = _ref.border,
      border = _ref$border === void 0 ? "" : _ref$border,
      _ref$borderRadius = _ref.borderRadius,
      borderRadius = _ref$borderRadius === void 0 ? "8px" : _ref$borderRadius;
    (0, _classCallCheck2.default)(this, DataItemStyles);
    _this = _super.call(this, {
      width: width,
      height: height,
      background: background,
      activeBackground: activeBackground,
      border: border,
      borderRadius: borderRadius
    });
    _this.titleColor = titleColor;
    _this.titleFont = titleFont;
    _this.subTitleColor = subTitleColor;
    _this.subTitleFont = subTitleFont;
    return _this;
  }
  return (0, _createClass2.default)(DataItemStyles);
}(_.BaseStyles);
exports.DataItemStyles = DataItemStyles;
