// react-ui-kit-testing/dist/Messages/Bubbles/CometChatMessageBubble/MessageBubbleConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageBubbleConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _DateConfiguration = require("../../../Shared/PrimaryComponents/CometChatConfiguration/DateConfiguration");
var _Shared = require("../../../Shared/");
var _MessageReactionsConfiguration = require("../../CometChatMessageReactions/MessageReactionsConfiguration");
var _TextBubbleConfiguration = require("../CometChatTextBubble/TextBubbleConfiguration");
var _FileBubbleConfiguration = require("../CometChatFileBubble/FileBubbleConfiguration");
var _ImageBubbleConfiguration = require("../CometChatImageBubble/ImageBubbleConfiguration");
var _AudioBubbleConfiguration = require("../../../Messages/Bubbles/CometChatAudioBubble/AudioBubbleConfiguration");
var _VideoBubbleConfiguration = require("../CometChatVideoBubble/VideoBubbleConfiguration");
var _PollBubbleConfiguration = require("../CometChatPollBubble/PollBubbleConfiguration");
var _StickerBubbleConfiguration = require("../CometChatStickerBubble/StickerBubbleConfiguration");
var _DeletedBubbleConfiguration = require("../CometChatDeletedMessageBubble/DeletedBubbleConfiguration");
var _CollaborativeWhiteboardConfiguration = require("../CometChatWhiteboardBubble/CollaborativeWhiteboardConfiguration");
var _CollaborativeDocumentConfiguration = require("../../../Messages/Bubbles/CometChatDocumentBubble/CollaborativeDocumentConfiguration");
var _MessageInputData = require("../../../Shared/InputData/MessageInputData");
var _Shared2 = require("../../../Shared");
/**
 * @class messageBubbleConfiguration
 * @description MessageBubbleConfiguration class is used for defining the message bubble templates
 * @param {Object} messageBubbleData
 * @param {String} timeAlignment
 * @param {Object} dateConfiguration
 * @param {Object} avatarConfiguration
 * @param {Object} messageReceiptConfiguration
 * @param {Object} messageReactionConfiguration 
 * @param {Object} textBubbleConfiguration
 * @param {Object} fileBubbleConfiguration
 * @param {Object} imageBubbleConfiguration
 * @param {Object} audioBubbleConfiguration
 * @param {Object} videoBubbleConfiguration
 * @param {Object} pollBubbleConfiguration
 * @param {Object} stickerBubbleConfiguration
 * @param {Object} deletedBubbleConfiguration
 * @param {Object} collaborativeWhiteboardConfiguration
 * @param {Object} collaborativeDocumentConfiguration
 */
var MessageBubbleConfiguration = /*#__PURE__*/(0, _createClass2.default)(function MessageBubbleConfiguration(_ref) {
  var _ref$messageBubbleDat = _ref.messageBubbleData,
    messageBubbleData = _ref$messageBubbleDat === void 0 ? new _MessageInputData.MessageInputData({}) : _ref$messageBubbleDat,
    _ref$timeAlignment = _ref.timeAlignment,
    timeAlignment = _ref$timeAlignment === void 0 ? _Shared2.MessageTimeAlignmentConstants.bottom : _ref$timeAlignment,
    _ref$dateConfiguratio = _ref.dateConfiguration,
    dateConfiguration = _ref$dateConfiguratio === void 0 ? new _DateConfiguration.DateConfiguration({}) : _ref$dateConfiguratio,
    _ref$avatarConfigurat = _ref.avatarConfiguration,
    avatarConfiguration = _ref$avatarConfigurat === void 0 ? new _Shared.AvatarConfiguration({}) : _ref$avatarConfigurat,
    _ref$messageReceiptCo = _ref.messageReceiptConfiguration,
    messageReceiptConfiguration = _ref$messageReceiptCo === void 0 ? new _Shared.MessageReceiptConfiguration({}) : _ref$messageReceiptCo,
    _ref$messageReactionC = _ref.messageReactionConfiguration,
    messageReactionConfiguration = _ref$messageReactionC === void 0 ? new _MessageReactionsConfiguration.MessageReactionsConfiguration({}) : _ref$messageReactionC,
    _ref$textBubbleConfig = _ref.textBubbleConfiguration,
    textBubbleConfiguration = _ref$textBubbleConfig === void 0 ? new _TextBubbleConfiguration.TextBubbleConfiguration({}) : _ref$textBubbleConfig,
    _ref$fileBubbleConfig = _ref.fileBubbleConfiguration,
    fileBubbleConfiguration = _ref$fileBubbleConfig === void 0 ? new _FileBubbleConfiguration.FileBubbleConfiguration({}) : _ref$fileBubbleConfig,
    _ref$imageBubbleConfi = _ref.imageBubbleConfiguration,
    imageBubbleConfiguration = _ref$imageBubbleConfi === void 0 ? new _ImageBubbleConfiguration.ImageBubbleConfiguration({}) : _ref$imageBubbleConfi,
    _ref$audioBubbleConfi = _ref.audioBubbleConfiguration,
    audioBubbleConfiguration = _ref$audioBubbleConfi === void 0 ? new _AudioBubbleConfiguration.AudioBubbleConfiguration({}) : _ref$audioBubbleConfi,
    _ref$videoBubbleConfi = _ref.videoBubbleConfiguration,
    videoBubbleConfiguration = _ref$videoBubbleConfi === void 0 ? new _VideoBubbleConfiguration.VideoBubbleConfiguration({}) : _ref$videoBubbleConfi,
    _ref$pollBubbleConfig = _ref.pollBubbleConfiguration,
    pollBubbleConfiguration = _ref$pollBubbleConfig === void 0 ? new _PollBubbleConfiguration.PollBubbleConfiguration({}) : _ref$pollBubbleConfig,
    _ref$stickerBubbleCon = _ref.stickerBubbleConfiguration,
    stickerBubbleConfiguration = _ref$stickerBubbleCon === void 0 ? new _StickerBubbleConfiguration.StickerBubbleConfiguration({}) : _ref$stickerBubbleCon,
    _ref$deletedBubbleCon = _ref.deletedBubbleConfiguration,
    deletedBubbleConfiguration = _ref$deletedBubbleCon === void 0 ? new _DeletedBubbleConfiguration.DeletedBubbleConfiguration({}) : _ref$deletedBubbleCon,
    _ref$collaborativeWhi = _ref.collaborativeWhiteboardConfiguration,
    collaborativeWhiteboardConfiguration = _ref$collaborativeWhi === void 0 ? new _CollaborativeWhiteboardConfiguration.CollaborativeWhiteboardConfiguration({}) : _ref$collaborativeWhi,
    _ref$collaborativeDoc = _ref.collaborativeDocumentConfiguration,
    collaborativeDocumentConfiguration = _ref$collaborativeDoc === void 0 ? new _CollaborativeDocumentConfiguration.CollaborativeDocumentConfiguration({}) : _ref$collaborativeDoc;
  (0, _classCallCheck2.default)(this, MessageBubbleConfiguration);
  this.messageBubbleData = new _MessageInputData.MessageInputData(messageBubbleData || {});
  this.timeAlignment = timeAlignment;
  this.DateConfiguration = new _DateConfiguration.DateConfiguration(dateConfiguration || {});
  this.AvatarConfiguration = new _Shared.AvatarConfiguration(avatarConfiguration || {});
  this.MessageReceiptConfiguration = new _Shared.MessageReceiptConfiguration(messageReceiptConfiguration || {});
  this.MessageReactionsConfiguration = new _MessageReactionsConfiguration.MessageReactionsConfiguration(messageReactionConfiguration || {});
  this.TextBubbleConfiguration = new _TextBubbleConfiguration.TextBubbleConfiguration(textBubbleConfiguration || {});
  this.FileBubbleConfiguration = new _FileBubbleConfiguration.FileBubbleConfiguration(fileBubbleConfiguration || {});
  this.ImageBubbleConfiguration = new _ImageBubbleConfiguration.ImageBubbleConfiguration(imageBubbleConfiguration || {});
  this.AudioBubbleConfiguration = new _AudioBubbleConfiguration.AudioBubbleConfiguration(audioBubbleConfiguration || {});
  this.VideoBubbleConfiguration = new _VideoBubbleConfiguration.VideoBubbleConfiguration(videoBubbleConfiguration || {});
  this.PollBubbleConfiguration = new _PollBubbleConfiguration.PollBubbleConfiguration(pollBubbleConfiguration || {});
  this.StickerBubbleConfiguration = new _StickerBubbleConfiguration.StickerBubbleConfiguration(stickerBubbleConfiguration || {});
  this.DeletedBubbleConfiguration = new _DeletedBubbleConfiguration.DeletedBubbleConfiguration(deletedBubbleConfiguration || {});
  this.CollaborativeWhiteboardBubbleConfiguration = new _CollaborativeWhiteboardConfiguration.CollaborativeWhiteboardConfiguration(collaborativeWhiteboardConfiguration || {});
  this.CollaborativeDocumentBubbleConfiguration = new _CollaborativeDocumentConfiguration.CollaborativeDocumentConfiguration(collaborativeDocumentConfiguration || {});
});
exports.MessageBubbleConfiguration = MessageBubbleConfiguration;
