// cometchat-react-ui-kit/dist/Messages/Bubbles/CometChatStickerBubble/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatStickerBubble = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _hooks = require("./hooks");

var _style = require("./style");

/**
 *
 * CometChatStickerBubble is UI component for sticker message bubble.
 *
 * @version 1.0.0
 * @author CometChatTeam
 * @copyright © 2022 CometChat Inc.
 *
 */
var CometChatStickerBubble = function CometChatStickerBubble(props) {
  var _React$useState = _react.default.useState(""),
      _React$useState2 = (0, _slicedToArray2.default)(_React$useState, 2),
      stickerURL = _React$useState2[0],
      setStickerURL = _React$useState2[1];

  var stickerMessage = function stickerMessage() {
    return /*#__PURE__*/_react.default.createElement("div", {
      style: (0, _style.messageStickerBlockStyle)(props),
      className: "message_kit__blocks"
    }, /*#__PURE__*/_react.default.createElement("img", {
      className: "message__message-blocks",
      style: (0, _style.messageStickerBubbleBlockStyle)(props),
      src: stickerURL,
      alt: stickerURL
    }));
  };

  (0, _hooks.Hooks)(props, setStickerURL);
  return stickerURL ? stickerMessage() : null;
};

exports.CometChatStickerBubble = CometChatStickerBubble;
CometChatStickerBubble.defaultProps = {
  messageObject: {},
  stickerURL: "",
  stickerName: "",
  style: {
    width: "150px",
    height: "149px",
    border: "0 none",
    borderRadius: "12px",
    background: "none"
  }
};
CometChatStickerBubble.propTypes = {
  messageObject: _propTypes.default.object,
  stickerURL: _propTypes.default.string,
  style: _propTypes.default.object
};
