// react-ui-kit-testing/dist/Messages/Bubbles/CometChatFileBubble/style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.messageTitleStyle = exports.messageSubTitleStyle = exports.messageKitFileBubbleBlockStyle = exports.messageFileIconStyle = exports.messageFileBubbleBlockStyle = void 0;
var _ = require("../../../");
var messageKitFileBubbleBlockStyle = function messageKitFileBubbleBlockStyle(props, theme) {
  return {
    borderRadius: props.style.borderRadius,
    padding: "8px 0px",
    background: props.style.background || theme.palette.accent50[theme.palette.mode],
    display: "flex",
    width: "100%",
    border: props.style.border,
    justifyContent: "center",
    alignItems: "center"
  };
};
exports.messageKitFileBubbleBlockStyle = messageKitFileBubbleBlockStyle;
var messageFileBubbleBlockStyle = function messageFileBubbleBlockStyle(props) {
  return {
    width: "calc(100% - 24px)",
    whiteSpace: "pre-wrap",
    wordWrap: "break-word",
    textAlign: "left",
    margin: "0 0 0 10px"
  };
};
exports.messageFileBubbleBlockStyle = messageFileBubbleBlockStyle;
var messageTitleStyle = function messageTitleStyle(props, theme) {
  var _theme$palette;
  return {
    color: "".concat(props.style.titleColor) || (theme === null || theme === void 0 ? void 0 : (_theme$palette = theme.palette) === null || _theme$palette === void 0 ? void 0 : _theme$palette.getAccent()),
    font: "".concat(props.style.titleFont) || (0, _.fontHelper)(theme.typography.title2)
  };
};
exports.messageTitleStyle = messageTitleStyle;
var messageSubTitleStyle = function messageSubTitleStyle(props, theme) {
  return {
    color: "".concat(props.style.subTitleColor) || theme.palette.accent600[theme.palette.mode],
    font: "".concat(props.style.subTitleFont) || (0, _.fontHelper)(theme.typography.subtitle2)
  };
};
exports.messageSubTitleStyle = messageSubTitleStyle;
var messageFileIconStyle = function messageFileIconStyle(props, url, theme) {
  var _theme$palette2;
  return {
    WebkitMask: "url(".concat(url, ") center center no-repeat"),
    background: "".concat(props.style.iconTint) || (theme === null || theme === void 0 ? void 0 : (_theme$palette2 = theme.palette) === null || _theme$palette2 === void 0 ? void 0 : _theme$palette2.getPrimary()),
    display: "inline-block",
    width: "24px",
    height: "24px",
    padding: "13px 20px",
    cursor: "pointer",
    zIndex: "2"
  };
};
exports.messageFileIconStyle = messageFileIconStyle;
