// react-ui-kit-testing/dist/Groups/CometChatGroups/GroupsConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GroupsConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _CreateGroupConfiguration = require("../CometChatCreateGroup/CreateGroupConfiguration");
var _GroupListConfiguration = require("../CometChatGroupList/GroupListConfiguration");
var _GroupsStyles = require("./GroupsStyles");
var _create = _interopRequireDefault(require("./resources/create.svg"));
var _back = _interopRequireDefault(require("./resources/back.svg"));
var _search = _interopRequireDefault(require("./resources/search.svg"));
/**
 * @class GroupsConfiguration
 * @param {String} backButtonIconURL
 * @param {Boolean} showBackButton
 * @param {String} searchIconURL
 * @param {Boolean} hideSearch
 * @param {String} createGroupIconURL
 * @param {Boolean} hideCreateGroup
 * @param {Object} style
 * @param {Object} groupListConfiguration
 * @param {Object} createGroupConfiguration
 */
var GroupsConfiguration = /*#__PURE__*/(0, _createClass2.default)(function GroupsConfiguration(_ref) {
  var _ref$backButtonIconUR = _ref.backButtonIconURL,
    backButtonIconURL = _ref$backButtonIconUR === void 0 ? _back.default : _ref$backButtonIconUR,
    _ref$showBackButton = _ref.showBackButton,
    showBackButton = _ref$showBackButton === void 0 ? false : _ref$showBackButton,
    _ref$searchIconURL = _ref.searchIconURL,
    searchIconURL = _ref$searchIconURL === void 0 ? _search.default : _ref$searchIconURL,
    _ref$hideSearch = _ref.hideSearch,
    hideSearch = _ref$hideSearch === void 0 ? false : _ref$hideSearch,
    _ref$createGroupIconU = _ref.createGroupIconURL,
    createGroupIconURL = _ref$createGroupIconU === void 0 ? _create.default : _ref$createGroupIconU,
    _ref$hideCreateGroup = _ref.hideCreateGroup,
    hideCreateGroup = _ref$hideCreateGroup === void 0 ? false : _ref$hideCreateGroup,
    _ref$style = _ref.style,
    style = _ref$style === void 0 ? new _GroupsStyles.GroupsStyles({}) : _ref$style,
    _ref$groupListConfigu = _ref.groupListConfiguration,
    groupListConfiguration = _ref$groupListConfigu === void 0 ? new _GroupListConfiguration.GroupListConfiguration({}) : _ref$groupListConfigu,
    _ref$createGroupConfi = _ref.createGroupConfiguration,
    createGroupConfiguration = _ref$createGroupConfi === void 0 ? new _CreateGroupConfiguration.CreateGroupConfiguration({}) : _ref$createGroupConfi;
  (0, _classCallCheck2.default)(this, GroupsConfiguration);
  this.backButtonIconURL = backButtonIconURL;
  this.showBackButton = showBackButton;
  this.searchIconURL = searchIconURL;
  this.hideSearch = hideSearch;
  this.createGroupIconURL = createGroupIconURL;
  this.hideCreateGroup = hideCreateGroup;
  this.style = new _GroupsStyles.GroupsStyles(style || {});
  groupListConfiguration = groupListConfiguration || new _GroupListConfiguration.GroupListConfiguration({});
  createGroupConfiguration = createGroupConfiguration || new _CreateGroupConfiguration.CreateGroupConfiguration({});
});
exports.GroupsConfiguration = GroupsConfiguration;
