// cometchat-react-ui-kit/dist/Shared/UtilityComponents/CometChatActionSheet/ActionSheetConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ActionSheetConfiguration = void 0;

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _fileUpload = _interopRequireDefault(require("../../../Messages/CometChatMessageComposer/resources/file-upload.svg"));

var _Shared = require("../../../Shared");

var _ = require("../..");

/**
 * @class ActionSheetConfiguration
 * @description ActionSheetConfiguration class is used for defining the ActionSheet templates.
 * @param {String} title
 * @param {String} layoutModeIconURL
 * @param {Boolean} hideLayoutMode
 * @param {Array} layoutMode
 * @param {Array} actions
 */
var ActionSheetConfiguration = /*#__PURE__*/(0, _createClass2.default)(function ActionSheetConfiguration(_ref) {
  var _ref$title = _ref.title,
      title = _ref$title === void 0 ? (0, _.localize)("ADD_TO_CHAT") : _ref$title,
      _ref$layoutModeIconUR = _ref.layoutModeIconURL,
      layoutModeIconURL = _ref$layoutModeIconUR === void 0 ? _fileUpload.default : _ref$layoutModeIconUR,
      _ref$layoutMode = _ref.layoutMode,
      layoutMode = _ref$layoutMode === void 0 ? (0, _.localize)("LIST") : _ref$layoutMode,
      _ref$hideLayoutMode = _ref.hideLayoutMode,
      hideLayoutMode = _ref$hideLayoutMode === void 0 ? false : _ref$hideLayoutMode,
      _ref$actions = _ref.actions,
      actions = _ref$actions === void 0 ? null : _ref$actions,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? new _Shared.ActionSheetStyles({}) : _ref$style;
  (0, _classCallCheck2.default)(this, ActionSheetConfiguration);
  this.title = title;
  this.layoutMode = layoutMode;
  this.hideLayoutMode = hideLayoutMode;
  this.layoutModeIconURL = layoutModeIconURL;
  this.actions = actions;
  this.style = new _Shared.ActionSheetStyles(style !== null && style !== void 0 ? style : {});
  this.listItemConfiguration = new _Shared.ListItemConfiguration();
});
exports.ActionSheetConfiguration = ActionSheetConfiguration;
