// cometchat-react-ui-kit/dist/Shared/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ActionSheetConfiguration", {
  enumerable: true,
  get: function get() {
    return _ActionSheetConfiguration.ActionSheetConfiguration;
  }
});
Object.defineProperty(exports, "ActionSheetStyles", {
  enumerable: true,
  get: function get() {
    return _ActionSheetStyles.ActionSheetStyles;
  }
});
Object.defineProperty(exports, "AvatarConfiguration", {
  enumerable: true,
  get: function get() {
    return _AvatarConfiguration.AvatarConfiguration;
  }
});
Object.defineProperty(exports, "AvatarStyles", {
  enumerable: true,
  get: function get() {
    return _AvatarStyles.AvatarStyles;
  }
});
Object.defineProperty(exports, "BadgeCountConfiguration", {
  enumerable: true,
  get: function get() {
    return _BadgeCountConfiguration.BadgeCountConfiguration;
  }
});
Object.defineProperty(exports, "BadgeCountStyles", {
  enumerable: true,
  get: function get() {
    return _BadgeCountStyles.BadgeCountStyles;
  }
});
Object.defineProperty(exports, "BaseStyles", {
  enumerable: true,
  get: function get() {
    return _BaseStyles.BaseStyles;
  }
});
Object.defineProperty(exports, "CometChatActionSheet", {
  enumerable: true,
  get: function get() {
    return _UtilityComponents.CometChatActionSheet;
  }
});
Object.defineProperty(exports, "CometChatActionSheetItem", {
  enumerable: true,
  get: function get() {
    return _UtilityComponents.CometChatActionSheetItem;
  }
});
Object.defineProperty(exports, "CometChatAvatar", {
  enumerable: true,
  get: function get() {
    return _SecondaryComponents.CometChatAvatar;
  }
});
Object.defineProperty(exports, "CometChatBackdrop", {
  enumerable: true,
  get: function get() {
    return _UtilityComponents.CometChatBackdrop;
  }
});
Object.defineProperty(exports, "CometChatBadgeCount", {
  enumerable: true,
  get: function get() {
    return _SecondaryComponents.CometChatBadgeCount;
  }
});
Object.defineProperty(exports, "CometChatConfirmDialog", {
  enumerable: true,
  get: function get() {
    return _UtilityComponents.CometChatConfirmDialog;
  }
});
Object.defineProperty(exports, "CometChatConversationListItem", {
  enumerable: true,
  get: function get() {
    return _SDKDerivedComponents.CometChatConversationListItem;
  }
});
Object.defineProperty(exports, "CometChatDataItem", {
  enumerable: true,
  get: function get() {
    return _SDKDerivedComponents.CometChatDataItem;
  }
});
Object.defineProperty(exports, "CometChatDate", {
  enumerable: true,
  get: function get() {
    return _SecondaryComponents.CometChatDate;
  }
});
Object.defineProperty(exports, "CometChatDecoratorMessage", {
  enumerable: true,
  get: function get() {
    return _UtilityComponents.CometChatDecoratorMessage;
  }
});
Object.defineProperty(exports, "CometChatDivider", {
  enumerable: true,
  get: function get() {
    return _CometChatDivider.CometChatDivider;
  }
});
Object.defineProperty(exports, "CometChatGroupListItem", {
  enumerable: true,
  get: function get() {
    return _SDKDerivedComponents.CometChatGroupListItem;
  }
});
Object.defineProperty(exports, "CometChatListBase", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.CometChatListBase;
  }
});
Object.defineProperty(exports, "CometChatListItem", {
  enumerable: true,
  get: function get() {
    return _UtilityComponents.CometChatListItem;
  }
});
Object.defineProperty(exports, "CometChatLocalize", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.CometChatLocalize;
  }
});
Object.defineProperty(exports, "CometChatMenuList", {
  enumerable: true,
  get: function get() {
    return _UtilityComponents.CometChatMenuList;
  }
});
Object.defineProperty(exports, "CometChatMessageReceipt", {
  enumerable: true,
  get: function get() {
    return _SecondaryComponents.CometChatMessageReceipt;
  }
});
Object.defineProperty(exports, "CometChatPopover", {
  enumerable: true,
  get: function get() {
    return _UtilityComponents.CometChatPopover;
  }
});
Object.defineProperty(exports, "CometChatSoundManager", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.CometChatSoundManager;
  }
});
Object.defineProperty(exports, "CometChatStatusIndicator", {
  enumerable: true,
  get: function get() {
    return _SecondaryComponents.CometChatStatusIndicator;
  }
});
Object.defineProperty(exports, "CometChatTheme", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.CometChatTheme;
  }
});
Object.defineProperty(exports, "CometChatUserListItem", {
  enumerable: true,
  get: function get() {
    return _SDKDerivedComponents.CometChatUserListItem;
  }
});
Object.defineProperty(exports, "ConversationListConfiguration", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.ConversationListConfiguration;
  }
});
Object.defineProperty(exports, "ConversationListItemConfiguration", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.ConversationListItemConfiguration;
  }
});
Object.defineProperty(exports, "ConversationListItemStyles", {
  enumerable: true,
  get: function get() {
    return _SDKDerivedComponents.ConversationListItemStyles;
  }
});
Object.defineProperty(exports, "ConversationOptionConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.ConversationOptionConstants;
  }
});
Object.defineProperty(exports, "ConversationTypeConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.ConversationTypeConstants;
  }
});
Object.defineProperty(exports, "ConversationsConfiguration", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.ConversationsConfiguration;
  }
});
Object.defineProperty(exports, "ConversationsWithMessagesConfiguration", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.ConversationsWithMessagesConfiguration;
  }
});
Object.defineProperty(exports, "CustomView", {
  enumerable: true,
  get: function get() {
    return _CustomView.CustomView;
  }
});
Object.defineProperty(exports, "DataItemConfiguration", {
  enumerable: true,
  get: function get() {
    return _DataItemConfiguration.DataItemConfiguration;
  }
});
Object.defineProperty(exports, "DataItemStyles", {
  enumerable: true,
  get: function get() {
    return _DataItemStyles.DataItemStyles;
  }
});
Object.defineProperty(exports, "DateConfiguration", {
  enumerable: true,
  get: function get() {
    return _DateConfiguration.DateConfiguration;
  }
});
Object.defineProperty(exports, "DateStyles", {
  enumerable: true,
  get: function get() {
    return _DateStyles.DateStyles;
  }
});
Object.defineProperty(exports, "DecoratorMessageStyles", {
  enumerable: true,
  get: function get() {
    return _DecoratorMessageStyles.DecoratorMessageStyles;
  }
});
Object.defineProperty(exports, "DetailsConfiguration", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.DetailsConfiguration;
  }
});
Object.defineProperty(exports, "ExtensionConstants", {
  enumerable: true,
  get: function get() {
    return _ExtensionConstants.ExtensionConstants;
  }
});
Object.defineProperty(exports, "ExtensionURLs", {
  enumerable: true,
  get: function get() {
    return _ExtensionConstants.ExtensionURLs;
  }
});
Object.defineProperty(exports, "GroupListItemConfiguration", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.GroupListItemConfiguration;
  }
});
Object.defineProperty(exports, "GroupMemberOptionConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.GroupMemberOptionConstants;
  }
});
Object.defineProperty(exports, "GroupMemberScope", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.GroupMemberScope;
  }
});
Object.defineProperty(exports, "GroupOptionConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.GroupOptionConstants;
  }
});
Object.defineProperty(exports, "GroupTypeConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.GroupTypeConstants;
  }
});
Object.defineProperty(exports, "InputData", {
  enumerable: true,
  get: function get() {
    return _InputData.InputData;
  }
});
Object.defineProperty(exports, "ListBaseConfiguration", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.ListBaseConfiguration;
  }
});
Object.defineProperty(exports, "ListItemConfiguration", {
  enumerable: true,
  get: function get() {
    return _ListItemConfiguration.ListItemConfiguration;
  }
});
Object.defineProperty(exports, "ListItemStyles", {
  enumerable: true,
  get: function get() {
    return _ListItemStyles.ListItemStyles;
  }
});
Object.defineProperty(exports, "MessageBubbleAlignmentConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.MessageBubbleAlignmentConstants;
  }
});
Object.defineProperty(exports, "MessageCategoryConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.MessageCategoryConstants;
  }
});
Object.defineProperty(exports, "MessageListAlignmentConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.MessageListAlignmentConstants;
  }
});
Object.defineProperty(exports, "MessageOptionConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.MessageOptionConstants;
  }
});
Object.defineProperty(exports, "MessageOptionForConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.MessageOptionForConstants;
  }
});
Object.defineProperty(exports, "MessageReceiptConfiguration", {
  enumerable: true,
  get: function get() {
    return _MessageReceiptConfiguration.MessageReceiptConfiguration;
  }
});
Object.defineProperty(exports, "MessageReceiptStyles", {
  enumerable: true,
  get: function get() {
    return _MessageReceiptStyles.MessageReceiptStyles;
  }
});
Object.defineProperty(exports, "MessageStatusConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.MessageStatusConstants;
  }
});
Object.defineProperty(exports, "MessageTimeAlignmentConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.MessageTimeAlignmentConstants;
  }
});
Object.defineProperty(exports, "MessageTypeConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.MessageTypeConstants;
  }
});
Object.defineProperty(exports, "MetadataConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.MetadataConstants;
  }
});
Object.defineProperty(exports, "Palette", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.Palette;
  }
});
Object.defineProperty(exports, "ReceiverTypeConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.ReceiverTypeConstants;
  }
});
Object.defineProperty(exports, "StatusIndicatorConfiguration", {
  enumerable: true,
  get: function get() {
    return _StatusIndicatorConfiguration.StatusIndicatorConfiguration;
  }
});
Object.defineProperty(exports, "StatusIndicatorStyles", {
  enumerable: true,
  get: function get() {
    return _StatusIndicatorStyles.StatusIndicatorStyles;
  }
});
Object.defineProperty(exports, "Typography", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.Typography;
  }
});
Object.defineProperty(exports, "UserListConfiguration", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.UserListConfiguration;
  }
});
Object.defineProperty(exports, "UserListItemConfiguration", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.UserListItemConfiguration;
  }
});
Object.defineProperty(exports, "UserOptionConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.UserOptionConstants;
  }
});
Object.defineProperty(exports, "UserStatusConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.UserStatusConstants;
  }
});
Object.defineProperty(exports, "bytesToSize", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.bytesToSize;
  }
});
Object.defineProperty(exports, "emailPattern", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.emailPattern;
  }
});
Object.defineProperty(exports, "fontHelper", {
  enumerable: true,
  get: function get() {
    return _Helpers.fontHelper;
  }
});
Object.defineProperty(exports, "getExtensionsData", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.getExtensionsData;
  }
});
Object.defineProperty(exports, "getMetadataByKey", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.getMetadataByKey;
  }
});
Object.defineProperty(exports, "localize", {
  enumerable: true,
  get: function get() {
    return _PrimaryComponents.localize;
  }
});
Object.defineProperty(exports, "messageConstants", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.messageConstants;
  }
});
Object.defineProperty(exports, "phoneNumPattern", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.phoneNumPattern;
  }
});
Object.defineProperty(exports, "urlPattern", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.urlPattern;
  }
});
Object.defineProperty(exports, "wordBoundary", {
  enumerable: true,
  get: function get() {
    return _UIKitConstants.wordBoundary;
  }
});

var _PrimaryComponents = require("./PrimaryComponents");

var _SDKDerivedComponents = require("./SDKDerivedComponents");

var _SecondaryComponents = require("./SecondaryComponents");

var _Helpers = require("./Helpers");

var _UtilityComponents = require("./UtilityComponents");

var _UIKitConstants = require("./Constants/UIKitConstants");

var _ExtensionConstants = require("./Constants/ExtensionConstants/ExtensionConstants");

var _CometChatDivider = require("./CometChatDivider");

var _InputData = require("./InputData/InputData");

var _CustomView = require("./CustomView");

var _AvatarConfiguration = require("./SecondaryComponents/CometChatAvatar/AvatarConfiguration");

var _BadgeCountConfiguration = require("./SecondaryComponents/CometChatBadgeCount/BadgeCountConfiguration");

var _StatusIndicatorConfiguration = require("./SecondaryComponents/CometChatStatusIndicator/StatusIndicatorConfiguration");

var _MessageReceiptConfiguration = require("./SecondaryComponents/CometChatMessageReceipt/MessageReceiptConfiguration");

var _DateConfiguration = require("./SecondaryComponents/CometChatDate/DateConfiguration");

var _ActionSheetConfiguration = require("./UtilityComponents/CometChatActionSheet/ActionSheetConfiguration");

var _DataItemConfiguration = require("./SDKDerivedComponents/CometChatDataItem/DataItemConfiguration");

var _ListItemConfiguration = require("./UtilityComponents/CometChatListItem/ListItemConfiguration");

var _BaseStyles = require("./Base/BaseStyles");

var _AvatarStyles = require("./SecondaryComponents/CometChatAvatar/AvatarStyles");

var _BadgeCountStyles = require("./SecondaryComponents/CometChatBadgeCount/BadgeCountStyles");

var _StatusIndicatorStyles = require("./SecondaryComponents/CometChatStatusIndicator/StatusIndicatorStyles");

var _MessageReceiptStyles = require("./SecondaryComponents/CometChatMessageReceipt/MessageReceiptStyles");

var _DateStyles = require("./SecondaryComponents/CometChatDate/DateStyles");

var _DecoratorMessageStyles = require("./UtilityComponents/CometChatDecoratorMessage/DecoratorMessageStyles");

var _ListItemStyles = require("./UtilityComponents/CometChatListItem/ListItemStyles");

var _DataItemStyles = require("./SDKDerivedComponents/CometChatDataItem/DataItemStyles");

var _ActionSheetStyles = require("./UtilityComponents/CometChatActionSheet/ActionSheetStyles");
