// react-ui-kit-testing/dist/Messages/Bubbles/CometChatImageBubble/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatImageBubble = void 0;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _hooks = require("./hooks");
var _style = require("./style");
var _px = _interopRequireDefault(require("./resources/1px.png"));
var _unsafeContent = _interopRequireDefault(require("./resources/unsafe-content.svg"));
/**
 *
 * CometChatImageBubble is UI component for image message bubble.
 *
 * @version 1.0.0
 * @author CometChatTeam
 * @copyright © 2022 CometChat Inc.
 *
 */
var CometChatImageBubble = function CometChatImageBubble(props) {
  var _React$useState = _react.default.useState(_px.default),
    _React$useState2 = (0, _slicedToArray2.default)(_React$useState, 2),
    imageURL = _React$useState2[0],
    setImageURL = _React$useState2[1];
  var _React$useState3 = _react.default.useState(false),
    _React$useState4 = (0, _slicedToArray2.default)(_React$useState3, 2),
    unsafe = _React$useState4[0],
    setUnsafe = _React$useState4[1];
  var unsafeMessage = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "message__unsafe",
    style: (0, _style.messageUnsafeStyle)(props)
  }, /*#__PURE__*/_react.default.createElement("i", {
    className: "messsage__unsafe__icon",
    style: (0, _style.messageUnsafeIconStyle)(props.overlayImageURL)
  })));
  var imageMessage = function imageMessage() {
    return /*#__PURE__*/_react.default.createElement("div", {
      style: (0, _style.messageImageBubbleStyle)(props),
      className: "message_kit__blocks"
    }, /*#__PURE__*/_react.default.createElement("img", {
      className: "message__message-blocks",
      style: (0, _style.messageImageBubbleBlockStyle)(props, unsafe),
      src: imageURL,
      alt: imageURL
    }), unsafe ? unsafeMessage : null);
  };
  (0, _hooks.Hooks)(props, setImageURL, imageURL, setUnsafe);
  return imageMessage();
};
exports.CometChatImageBubble = CometChatImageBubble;
CometChatImageBubble.defaultProps = {
  messageObject: null,
  overlayImageURL: _unsafeContent.default,
  imageURL: "",
  style: {
    width: "300px",
    height: "200px",
    borderRadius: "12px",
    background: "rgb(255,255,255)",
    border: "0 none"
  }
};
CometChatImageBubble.propTypes = {
  messageObject: _propTypes.default.object,
  overlayImageURL: _propTypes.default.string,
  imageURL: _propTypes.default.string,
  style: _propTypes.default.object
};
