// cometchat-react-ui-kit/dist/Messages/CometChatCreatePollOptions/CreatePollOptionStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CreatePollOptionStyles = void 0;

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _Shared = require("../../Shared");

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

/**
 * @class CreatePollOptionStyles
 * @param {String} deleteIconTint
 * @param {String} boxShadow
 * @param {String} placeholderTextFont
 * @param {String} placeholderTextColor
 * @param {String} inputTextFont
 * @param {String} inputTextColor
 */
var CreatePollOptionStyles = /*#__PURE__*/function (_BaseStyles) {
  (0, _inherits2.default)(CreatePollOptionStyles, _BaseStyles);

  var _super = _createSuper(CreatePollOptionStyles);

  function CreatePollOptionStyles(_ref) {
    var _this;

    var _ref$deleteIconTint = _ref.deleteIconTint,
        deleteIconTint = _ref$deleteIconTint === void 0 ? "" : _ref$deleteIconTint,
        _ref$boxShadow = _ref.boxShadow,
        boxShadow = _ref$boxShadow === void 0 ? "" : _ref$boxShadow,
        _ref$placeholderTextF = _ref.placeholderTextFont,
        placeholderTextFont = _ref$placeholderTextF === void 0 ? "" : _ref$placeholderTextF,
        _ref$placeholderTextC = _ref.placeholderTextColor,
        placeholderTextColor = _ref$placeholderTextC === void 0 ? "" : _ref$placeholderTextC,
        _ref$inputTextFont = _ref.inputTextFont,
        inputTextFont = _ref$inputTextFont === void 0 ? "" : _ref$inputTextFont,
        _ref$inputTextColor = _ref.inputTextColor,
        inputTextColor = _ref$inputTextColor === void 0 ? "" : _ref$inputTextColor,
        _ref$width = _ref.width,
        width = _ref$width === void 0 ? "100%" : _ref$width,
        _ref$height = _ref.height,
        height = _ref$height === void 0 ? "46px" : _ref$height,
        _ref$background = _ref.background,
        background = _ref$background === void 0 ? "" : _ref$background,
        _ref$border = _ref.border,
        border = _ref$border === void 0 ? "" : _ref$border,
        _ref$borderRadius = _ref.borderRadius,
        borderRadius = _ref$borderRadius === void 0 ? "8px" : _ref$borderRadius,
        _ref$activeBackground = _ref.activeBackground,
        activeBackground = _ref$activeBackground === void 0 ? "" : _ref$activeBackground;
    (0, _classCallCheck2.default)(this, CreatePollOptionStyles);
    _this = _super.call(this, {
      width: width,
      height: height,
      background: background,
      activeBackground: activeBackground,
      border: border,
      borderRadius: borderRadius
    });
    _this.deleteIconTint = deleteIconTint;
    _this.boxShadow = boxShadow;
    _this.placeholderTextFont = placeholderTextFont;
    _this.placeholderTextColor = placeholderTextColor;
    _this.inputTextFont = inputTextFont;
    _this.inputTextColor = inputTextColor;
    return _this;
  }

  return (0, _createClass2.default)(CreatePollOptionStyles);
}(_Shared.BaseStyles);

exports.CreatePollOptionStyles = CreatePollOptionStyles;
