// @cometchat-pro/react-ui-kit/dist/Messages/Bubbles/CometChatVideoBubble/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatVideoBubble = void 0;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _hooks = require("./hooks");
var _style = require("./style");
var CometChatVideoBubble = function CometChatVideoBubble(props) {
  var _React$useState = _react["default"].useState(""),
    _React$useState2 = (0, _slicedToArray2["default"])(_React$useState, 2),
    videoURL = _React$useState2[0],
    setVideoURL = _React$useState2[1];
  var videoMessage = function videoMessage() {
    return /*#__PURE__*/_react["default"].createElement("div", {
      style: (0, _style.messageVideoBubbleStyle)(props),
      className: "message_kit__blocks"
    }, /*#__PURE__*/_react["default"].createElement("video", {
      className: "message__message-blocks",
      style: (0, _style.messageVideoBubbleBlockStyle)(props),
      controls: true,
      src: videoURL
    }));
  };
  (0, _hooks.Hooks)(props, setVideoURL);
  return videoMessage();
};
exports.CometChatVideoBubble = CometChatVideoBubble;
CometChatVideoBubble.defaultProps = {
  messageObject: null,
  videoURL: null,
  style: {
    width: "250px",
    height: "200px",
    borderRadius: "12px",
    background: "rgb(225,255,255)",
    border: "0 none"
  }
};
CometChatVideoBubble.propTypes = {
  messageObject: _propTypes["default"].object,
  videoURL: _propTypes["default"].string,
  style: _propTypes["default"].object
};
