// react-ui-kit-testing/dist/Messages/Bubbles/CometChatDeletedMessageBubble/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatDeletedMessageBubble = void 0;
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _ = require("../../..");
var _2 = require("../..");
var _style = require("./style");
var CometChatDeletedMessageBubble = function CometChatDeletedMessageBubble(props) {
  var message = null;
  message = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    style: (0, _style.messageTxtWrapperStyle)(props),
    className: "message__text__wrapper"
  }, /*#__PURE__*/_react.default.createElement("p", {
    style: (0, _style.messageTxtStyle)(props),
    className: "message__text"
  }, props.text ? props.text : message)));
  return message !== null ? /*#__PURE__*/_react.default.createElement("div", {
    style: (0, _style.messageContainerStyle)(props),
    className: "message__deleted"
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: (0, _style.messageWrapperStyle)(props),
    className: "message__wrapper"
  }, message)) : null;
};

// Specifies the default values for props:
exports.CometChatDeletedMessageBubble = CometChatDeletedMessageBubble;
CometChatDeletedMessageBubble.defaultProps = {
  messageObject: {},
  text: "",
  style: {
    width: "",
    height: "",
    border: "",
    background: "",
    borderRadius: "",
    textFont: "",
    textColor: ""
  }
};
CometChatDeletedMessageBubble.propTypes = {
  messageObject: _propTypes.default.object,
  text: _propTypes.default.string,
  style: _propTypes.default.object
};
