// react-ui-kit-testing/dist/Messages/Bubbles/CometChatDeletedMessageBubble/style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.messageWrapperStyle = exports.messageTxtWrapperStyle = exports.messageTxtStyle = exports.messageContainerStyle = void 0;
var messageContainerStyle = function messageContainerStyle(props) {
  return {
    // paddingLeft: "16px",
    // paddingRight: "16px",
    clear: "both",
    flexShrink: "0"
  };
};
exports.messageContainerStyle = messageContainerStyle;
var messageWrapperStyle = function messageWrapperStyle(props) {
  return {
    flex: "1 1",
    display: "flex",
    position: "relative",
    width: props.style.width,
    height: props.style.height
  };
};
exports.messageWrapperStyle = messageWrapperStyle;
var messageTxtWrapperStyle = function messageTxtWrapperStyle(props) {
  return {
    display: "inline-block",
    padding: "8px 12px",
    alignSelf: "flex-end",
    Width: props.style.width,
    height: props.style.height,
    border: props.style.border,
    background: props.style.background,
    borderRadius: props.style.borderRadius
  };
};
exports.messageTxtWrapperStyle = messageTxtWrapperStyle;
var messageTxtStyle = function messageTxtStyle(props) {
  return {
    font: props.style.textFont,
    color: props.style.textColor,
    margin: "0",
    lineHeight: "20px!important"
  };
};
exports.messageTxtStyle = messageTxtStyle;
