// react-ui-kit-testing/dist/Messages/CometChatMessageList/MessageListStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageListStyles = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _Shared = require("../../Shared");
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
 * @class MessageListStyles
 * @description MessageListStyles class is used for defining the messageList component's style
 * @param {String} width
 * @param {String} height
 * @param {String} background
 * @param {String} border
 * @param {String} borderRadius
 * @param {String} loadingIconTint
 * @param {String} TextFont
 * @param {String} TextColor
 * @param {String} emptyStateTextFont
 * @param {String} emptyStateTextColor
 * @param {String} errorStateTextFont
 * @param {String} errorStateTextColor
 */
var MessageListStyles = /*#__PURE__*/function (_BaseStyles) {
  (0, _inherits2.default)(MessageListStyles, _BaseStyles);
  var _super = _createSuper(MessageListStyles);
  function MessageListStyles(_ref) {
    var _this;
    var width = _ref.width,
      height = _ref.height,
      background = _ref.background,
      border = _ref.border,
      borderRadius = _ref.borderRadius,
      loadingIconTint = _ref.loadingIconTint,
      TextFont = _ref.TextFont,
      TextColor = _ref.TextColor,
      emptyStateTextFont = _ref.emptyStateTextFont,
      emptyStateTextColor = _ref.emptyStateTextColor,
      errorStateTextFont = _ref.errorStateTextFont,
      errorStateTextColor = _ref.errorStateTextColor;
    (0, _classCallCheck2.default)(this, MessageListStyles);
    _this = _super.call(this, {
      width: width,
      height: height,
      background: background,
      border: border,
      borderRadius: borderRadius
    });
    _this.loadingIconTint = loadingIconTint;
    _this.TextFont = TextFont;
    _this.TextColor = TextColor;
    _this.emptyStateTextFont = emptyStateTextFont;
    _this.emptyStateTextColor = emptyStateTextColor;
    _this.errorStateTextFont = errorStateTextFont;
    _this.errorStateTextColor = errorStateTextColor;
    return _this;
  }
  return (0, _createClass2.default)(MessageListStyles);
}(_Shared.BaseStyles);
exports.MessageListStyles = MessageListStyles;
