// react-ui-kit-testing/dist/Messages/CometChatMessagePreview/MessagePreviewConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessagePreviewConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _close = _interopRequireDefault(require("../CometChatMessageComposer/resources/close.svg"));
var _ = require("../../../");
var _2 = require("../");
/**
 * @class MessagePreviewConfiguration
 * @description MessagePreviewConfiguration class is used for defining the MessagePreview templates.
 * @param {String} iconURL
 * @param {String} text
 * @param {Object} tail
 * @param {Function} onItemClick
 * @param {Number} id
 */
var MessagePreviewConfiguration = /*#__PURE__*/(0, _createClass2.default)(function MessagePreviewConfiguration(_ref) {
  var _ref$messagePreviewTi = _ref.messagePreviewTitle,
    messagePreviewTitle = _ref$messagePreviewTi === void 0 ? (0, _.localize)("EDIT_MESSAGE") : _ref$messagePreviewTi,
    _ref$messagePreviewSu = _ref.messagePreviewSubtitle,
    messagePreviewSubtitle = _ref$messagePreviewSu === void 0 ? null : _ref$messagePreviewSu,
    _ref$closeIconURL = _ref.closeIconURL,
    closeIconURL = _ref$closeIconURL === void 0 ? _close.default : _ref$closeIconURL,
    _ref$onCloseClick = _ref.onCloseClick,
    onCloseClick = _ref$onCloseClick === void 0 ? null : _ref$onCloseClick,
    _ref$style = _ref.style,
    style = _ref$style === void 0 ? new _2.MessagePreviewStyles({}) : _ref$style;
  (0, _classCallCheck2.default)(this, MessagePreviewConfiguration);
  this.messagePreviewTitle = messagePreviewTitle;
  this.messagePreviewSubtitle = messagePreviewSubtitle;
  this.closeIconURL = closeIconURL;
  this.onCloseClick = onCloseClick;
  this.style = new _2.MessagePreviewStyles(style !== null && style !== void 0 ? style : {});
});
exports.MessagePreviewConfiguration = MessagePreviewConfiguration;
