// react-ui-kit-testing/dist/Messages/CometChatMessagePreview/MessagePreviewStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessagePreviewStyles = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _Shared = require("../../Shared");
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
 * @class MessagePreviewStyles
 * @param {String} border
 * @param {String} background
 * @param {String} messagePreviewTitleFont
 * @param {String} messagePreviewTitleColor
 * @param {String} messagePreviewSubtitleColor
 * @param {String} messagePreviewSubtitleFont
 * @param {String} closeIconTint
 */
var MessagePreviewStyles = /*#__PURE__*/function (_BaseStyles) {
  (0, _inherits2.default)(MessagePreviewStyles, _BaseStyles);
  var _super = _createSuper(MessagePreviewStyles);
  function MessagePreviewStyles(_ref) {
    var _this;
    var _ref$messagePreviewTi = _ref.messagePreviewTitleFont,
      messagePreviewTitleFont = _ref$messagePreviewTi === void 0 ? "" : _ref$messagePreviewTi,
      _ref$messagePreviewTi2 = _ref.messagePreviewTitleColor,
      messagePreviewTitleColor = _ref$messagePreviewTi2 === void 0 ? "" : _ref$messagePreviewTi2,
      _ref$messagePreviewSu = _ref.messagePreviewSubtitleColor,
      messagePreviewSubtitleColor = _ref$messagePreviewSu === void 0 ? "" : _ref$messagePreviewSu,
      _ref$messagePreviewSu2 = _ref.messagePreviewSubtitleFont,
      messagePreviewSubtitleFont = _ref$messagePreviewSu2 === void 0 ? "" : _ref$messagePreviewSu2,
      _ref$closeIconTint = _ref.closeIconTint,
      closeIconTint = _ref$closeIconTint === void 0 ? "" : _ref$closeIconTint,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "100%" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "auto" : _ref$height,
      _ref$background = _ref.background,
      background = _ref$background === void 0 ? "" : _ref$background,
      _ref$border = _ref.border,
      border = _ref$border === void 0 ? "" : _ref$border,
      _ref$borderRadius = _ref.borderRadius,
      borderRadius = _ref$borderRadius === void 0 ? "" : _ref$borderRadius,
      _ref$activeBackground = _ref.activeBackground,
      activeBackground = _ref$activeBackground === void 0 ? "" : _ref$activeBackground;
    (0, _classCallCheck2.default)(this, MessagePreviewStyles);
    _this = _super.call(this, {
      width: width,
      height: height,
      background: background,
      activeBackground: activeBackground,
      border: border,
      borderRadius: borderRadius
    });
    _this.messagePreviewTitleFont = messagePreviewTitleFont;
    _this.messagePreviewTitleColor = messagePreviewTitleColor;
    _this.messagePreviewSubtitleColor = messagePreviewSubtitleColor;
    _this.messagePreviewSubtitleFont = messagePreviewSubtitleFont;
    _this.closeIconTint = closeIconTint;
    return _this;
  }
  return (0, _createClass2.default)(MessagePreviewStyles);
}(_Shared.BaseStyles);
exports.MessagePreviewStyles = MessagePreviewStyles;
