// cometchat-react-ui-kit/dist/Messages/Bubbles/CometChatAudioBubble/style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.messageBlockAudioStyle = exports.messageAudioBubbleStyle = exports.messageAudioBubbleBlockStyle = void 0;

var messageAudioBubbleStyle = function messageAudioBubbleStyle(props) {
  return {
    borderRadius: props.style.borderRadius,
    background: props.style.background,
    border: props.style.border,
    display: "flex",
    width: props.style.width
  };
};

exports.messageAudioBubbleStyle = messageAudioBubbleStyle;

var messageAudioBubbleBlockStyle = function messageAudioBubbleBlockStyle(props) {
  return {
    width: props.style.width,
    height: props.style.height,
    whiteSpace: "pre-wrap",
    wordWrap: "break-word",
    textAlign: "left",
    display: "inline-block"
  };
};

exports.messageAudioBubbleBlockStyle = messageAudioBubbleBlockStyle;

var messageBlockAudioStyle = function messageBlockAudioStyle(props) {
  return {
    width: props.style.width // height: "100%",

  };
};

exports.messageBlockAudioStyle = messageBlockAudioStyle;
