// @cometchat-pro/react-ui-kit/dist/Messages/Bubbles/CometChatMessageBubble/MessageBubbleStyle.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageBubbleStyles = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _Shared = require("../../../Shared");
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
 * @class MessageBubbleStyles
 * @description MessageBubbleStyle is used to defining message bubble style
 * @param {String} width
 * @param {String} height
 * @param {String} nameTextFont
 * @param {String} nameTextColor
 * @param {String} border
 * @param {String} borderRadius
 * @param {String} background
 * @param {String} timestampFont
 * @param {String} timestampColor
 */var MessageBubbleStyles = /*#__PURE__*/function (_BaseStyles) {
  (0, _inherits2["default"])(MessageBubbleStyles, _BaseStyles);
  var _super = _createSuper(MessageBubbleStyles);
  function MessageBubbleStyles(_ref) {
    var _this;
    var width = _ref.width,
      height = _ref.height,
      background = _ref.background,
      border = _ref.border,
      borderRadius = _ref.borderRadius,
      nameTextFont = _ref.nameTextFont,
      nameTextColor = _ref.nameTextColor,
      timestampFont = _ref.timestampFont,
      timestampColor = _ref.timestampColor;
    (0, _classCallCheck2["default"])(this, MessageBubbleStyles);
    _this = _super.call(this, {
      width: width,
      height: height,
      background: background,
      border: border,
      borderRadius: borderRadius
    });
    _this.nameTextFont = nameTextFont;
    _this.nameTextColor = nameTextColor;
    _this.timestampFont = timestampFont;
    _this.timestampColor = timestampColor;
    return _this;
  }
  return (0, _createClass2["default"])(MessageBubbleStyles);
}(_Shared.BaseStyles);
exports.MessageBubbleStyles = MessageBubbleStyles;
