// react-ui-kit-testing/dist/Messages/CometChatCreatePoll/CreatePollConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CreatePollConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _close = _interopRequireDefault(require("./resources/close.svg"));
var _addCircleFilled = _interopRequireDefault(require("../CometChatMessageComposer/resources/add-circle-filled.svg"));
var _delete = _interopRequireDefault(require("../CometChatCreatePoll/resources/delete.svg"));
var _ = require("../");
/**
 * @class CreatePollConfiguration
 * @description CreatePollConfiguration class is used for defining the MessageComposer templates.
 * @param {callback} onClose
 * @param {callback} onCreatePoll
 * @param {string} closeIconURL
 * @param {string} deleteIconURL
 * @param {string} addAnswerIconURL
 * @param {object} style
 */
var CreatePollConfiguration = /*#__PURE__*/(0, _createClass2.default)(function CreatePollConfiguration(_ref) {
  var _ref$addAnswerIconURL = _ref.addAnswerIconURL,
    addAnswerIconURL = _ref$addAnswerIconURL === void 0 ? _addCircleFilled.default : _ref$addAnswerIconURL,
    _ref$closeIconURL = _ref.closeIconURL,
    closeIconURL = _ref$closeIconURL === void 0 ? _close.default : _ref$closeIconURL,
    _ref$deleteIconURL = _ref.deleteIconURL,
    deleteIconURL = _ref$deleteIconURL === void 0 ? _delete.default : _ref$deleteIconURL,
    _ref$onCreatePoll = _ref.onCreatePoll,
    onCreatePoll = _ref$onCreatePoll === void 0 ? null : _ref$onCreatePoll,
    _ref$onClose = _ref.onClose,
    onClose = _ref$onClose === void 0 ? null : _ref$onClose,
    _ref$style = _ref.style,
    style = _ref$style === void 0 ? new _.CreatePollStyles({}) : _ref$style;
  (0, _classCallCheck2.default)(this, CreatePollConfiguration);
  this.onClose = onClose;
  this.onCreatePoll = onCreatePoll;
  this.deleteIconURL = deleteIconURL;
  this.closeIconURL = closeIconURL;
  this.addAnswerIconURL = addAnswerIconURL;
  this.style = new _.CreatePollStyles(style !== null && style !== void 0 ? style : {});
});
exports.CreatePollConfiguration = CreatePollConfiguration;
