// @cometchat-pro/react-ui-kit/dist/Messages/Bubbles/CometChatImageBubble/style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.messageUnsafeStyle = exports.messageUnsafeIconStyle = exports.messageImageBubbleStyle = exports.messageImageBubbleBlockStyle = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
var messageImageBubbleStyle = function messageImageBubbleStyle(props) {
  return {
    padding: "0",
    minWidth: "110px",
    height: props.style.height,
    border: props.style.border,
    background: props.style.background,
    borderRadius: props.style.borderRadius,
    position: "relative"
  };
};
exports.messageImageBubbleStyle = messageImageBubbleStyle;
var messageImageBubbleBlockStyle = function messageImageBubbleBlockStyle(props, unsafe) {
  var unsafeStyle = {};
  if (unsafe) {
    unsafeStyle = {
      filter: "blur(12px)",
      opacity: "0.4"
    };
  }
  return _objectSpread({
    margin: "0",
    borderRadius: "inherit",
    maxWidth: props.style.width,
    height: props.style.height
  }, unsafeStyle);
};

/** message unsafe styles */
exports.messageImageBubbleBlockStyle = messageImageBubbleBlockStyle;
var messageUnsafeStyle = function messageUnsafeStyle(props) {
  return {
    background: "transparent",
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    zIndex: "20",
    width: "90%",
    padding: "20px",
    textAlign: "center"
  };
};
exports.messageUnsafeStyle = messageUnsafeStyle;
var messageUnsafeIconStyle = function messageUnsafeIconStyle(img) {
  return {
    width: "38px",
    height: "48px",
    WebkitMask: "url(".concat(img, ") center center no-repeat"),
    background: "rgba(255,255,255)",
    display: "block",
    margin: "auto",
    marginBottom: "5px",
    cursor: "pointer"
  };
};
exports.messageUnsafeIconStyle = messageUnsafeIconStyle;
