// react-ui-kit-testing/dist/Shared/PrimaryComponents/CometChatConfiguration/ListBaseConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListBaseConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _ListBaseStyles = require("../CometChatListBase/ListBaseStyles");
/**
 * @class ListBaseConfiguration
 * @param {String} title
 * @param {String} searchPlaceholder
 * @param {String} backButtonIconURL
 * @param {String} searchIconURL
 * @param {Boolean} showBackButton
 * @param {Boolean} hideSearch
 * @param {String} searchText
 * @param {Function} onSearch
 * @param {Function} onBackButtonClick
 * @param {Object} style
 */
var ListBaseConfiguration = /*#__PURE__*/(0, _createClass2.default)(function ListBaseConfiguration(_ref) {
  var _ref$title = _ref.title,
    title = _ref$title === void 0 ? "Title" : _ref$title,
    _ref$searchPlaceholde = _ref.searchPlaceholder,
    searchPlaceholder = _ref$searchPlaceholde === void 0 ? "Search" : _ref$searchPlaceholde,
    _ref$onSearch = _ref.onSearch,
    onSearch = _ref$onSearch === void 0 ? function () {} : _ref$onSearch,
    _ref$onBackButtonClic = _ref.onBackButtonClick,
    onBackButtonClick = _ref$onBackButtonClic === void 0 ? function () {} : _ref$onBackButtonClic,
    _ref$backButtonIconUR = _ref.backButtonIconURL,
    backButtonIconURL = _ref$backButtonIconUR === void 0 ? "" : _ref$backButtonIconUR,
    _ref$searchIconURL = _ref.searchIconURL,
    searchIconURL = _ref$searchIconURL === void 0 ? "" : _ref$searchIconURL,
    _ref$showBackButton = _ref.showBackButton,
    showBackButton = _ref$showBackButton === void 0 ? true : _ref$showBackButton,
    _ref$hideSearch = _ref.hideSearch,
    hideSearch = _ref$hideSearch === void 0 ? true : _ref$hideSearch,
    _ref$searchText = _ref.searchText,
    searchText = _ref$searchText === void 0 ? "" : _ref$searchText,
    _ref$style = _ref.style,
    style = _ref$style === void 0 ? new _ListBaseStyles.ListBaseStyles({}) : _ref$style;
  (0, _classCallCheck2.default)(this, ListBaseConfiguration);
  this.title = title;
  this.searchPlaceholder = searchPlaceholder;
  this.onSearch = onSearch;
  this.onBackButtonClick = onBackButtonClick;
  this.backButtonIconURL = backButtonIconURL;
  this.searchIconURL = searchIconURL;
  this.showBackButton = showBackButton;
  this.hideSearch = hideSearch;
  this.searchText = searchText;
  this.style = new _ListBaseStyles.ListBaseStyles(style !== null && style !== void 0 ? style : {});
});
exports.ListBaseConfiguration = ListBaseConfiguration;
