// react-ui-kit-testing/dist/Shared/UtilityComponents/CometChatDecoratorMessage/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatDecoratorMessage = void 0;
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var CometChatDecoratorMessage = function CometChatDecoratorMessage(props) {
  /**
   * Component template scoping
   */
  if (props.text) {
    var getStyle = function getStyle() {
      return {
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        width: "100%",
        height: "100%",
        backgroundColor: props.background,
        color: props.textColor,
        font: props.textFont
      };
    };
    return /*#__PURE__*/_react.default.createElement("div", {
      style: getStyle()
    }, props.text);
  }
  return null;
};

/**
 * Component default props values
 */
exports.CometChatDecoratorMessage = CometChatDecoratorMessage;
CometChatDecoratorMessage.defaultProps = {
  background: "rgba(51, 153, 255, 1)",
  textColor: "white",
  textFont: "600 12px Inter",
  text: ""
};

/**
 * Component default props
 */
CometChatDecoratorMessage.propTypes = {
  background: _propTypes.default.string,
  textColor: _propTypes.default.string,
  textFont: _propTypes.default.string,
  text: _propTypes.default.string
};
