// cometchat-react-ui-kit/dist/Messages/Bubbles/CometChatTextBubble/style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.previewTitleStyle = exports.previewLinkStyle = exports.previewImageStyle = exports.previewDataStyle = exports.messageTextBubbleStyle = exports.messagePreviewWrapperStyle = exports.messagePreviewContainerStyle = exports.messageKitTextBubbleBlockStyle = exports.linkSubtitleStyle = void 0;

var _Shared = require("../../../Shared");

var messageKitTextBubbleBlockStyle = function messageKitTextBubbleBlockStyle(props, theme) {
  return {
    textAlign: "inherit",
    display: "inline-block",
    padding: "8px 12px",
    border: props.style.border,
    width: props.style.width,
    height: props.style.height,
    borderRadius: props.style.borderRadius,
    background: props.style.background //theme not applied jut bcz transparent

  };
};

exports.messageKitTextBubbleBlockStyle = messageKitTextBubbleBlockStyle;

var messageTextBubbleStyle = function messageTextBubbleStyle(props, theme) {
  return {
    color: props.style.textColor || theme.palette.background[theme.palette.mode],
    //text color not sure in theme
    font: props.style.textFont || (0, _Shared.fontHelper)(theme.typography.subtitle1),
    whiteSpace: "pre-wrap",
    wordWrap: "break-word",
    margin: "0",
    width: "auto"
  };
};
/** link previe style */


exports.messageTextBubbleStyle = messageTextBubbleStyle;

var messagePreviewContainerStyle = function messagePreviewContainerStyle(props, theme) {
  return {
    display: "inline-block",
    borderRadius: "12px",
    backgroundColor: props.style.linkPreviewBackgroundColor || theme.palette.background[theme.palette.mode],
    boxShadow: "0px 1px 2px 1px rgba(0,0,0,0.18)",
    alignSelf: "flex-start",
    width: "auto"
  };
};

exports.messagePreviewContainerStyle = messagePreviewContainerStyle;

var messagePreviewWrapperStyle = function messagePreviewWrapperStyle() {
  return {
    display: "flex",
    flexDirection: "column"
  };
};

exports.messagePreviewWrapperStyle = messagePreviewWrapperStyle;

var previewImageStyle = function previewImageStyle(img) {
  return {
    background: "url(".concat(img, ") no-repeat center center"),
    backgroundSize: "cover",
    height: "150px",
    minHeight: "50px",
    borderRadius: "12px 12px 0 0"
  };
};

exports.previewImageStyle = previewImageStyle;

var previewDataStyle = function previewDataStyle() {
  return {
    padding: "16px 16px 0 16px"
  };
};

exports.previewDataStyle = previewDataStyle;

var previewTitleStyle = function previewTitleStyle(props, theme) {
  return {
    whiteSpace: "pre-wrap",
    wordBreak: "break-word",
    textAlign: "left",
    width: "auto",
    color: props.style.linkPreviewTitleColor || theme.palette.accent[theme.palette.mode],
    font: props.style.linkPreviewTitleFont || (0, _Shared.fontHelper)(theme.typography.title2),
    marginBottom: "8px"
  };
};

exports.previewTitleStyle = previewTitleStyle;

var previewLinkStyle = function previewLinkStyle() {
  return {
    display: "flex",
    alignItems: "center",
    justifyContent: "left",
    padding: "0 16px 16px 16px"
  };
};

exports.previewLinkStyle = previewLinkStyle;

var linkSubtitleStyle = function linkSubtitleStyle(props, theme) {
  return {
    display: "inline-block",
    color: props.style.linkPreviewSubtitleColor || theme.palette.accent600[theme.palette.mode],
    font: props.style.linkPreviewSubtitleFont || (0, _Shared.fontHelper)(theme.typography.subtitle2),
    textDecoration: "none"
  };
};

exports.linkSubtitleStyle = linkSubtitleStyle;
