// @cometchat-pro/react-ui-kit/dist/Messages/CometChatCreatePollOptions/style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.removeOptionIconStyle = exports.pollOptionInputStyle = exports.pollOptionAnswerStyle = exports.iconWrapperStyle = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
var pollOptionAnswerStyle = function pollOptionAnswerStyle() {
  return {
    width: "100%",
    margin: "10px 0",
    position: "relative"
  };
};
exports.pollOptionAnswerStyle = pollOptionAnswerStyle;
var pollOptionInputStyle = function pollOptionInputStyle(props) {
  return _objectSpread({}, props.style.inputStyles);
};
exports.pollOptionInputStyle = pollOptionInputStyle;
var removeOptionIconStyle = function removeOptionIconStyle(props, img) {
  return {
    WebkitMask: "url(".concat(props.deleteIconURL || img, ") right center / 24px no-repeat"),
    background: props.style.deleteIconTint,
    cursor: "pointer",
    display: "block",
    height: "28px",
    width: "28px",
    position: "absolute",
    top: "20%",
    left: "90%"
  };
};
exports.removeOptionIconStyle = removeOptionIconStyle;
var iconWrapperStyle = function iconWrapperStyle(props) {
  return {
    width: "28px",
    height: "28px"
  };
};
exports.iconWrapperStyle = iconWrapperStyle;
