// react-ui-kit-testing/dist/Messages/CometChatMessageTemplate/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getDefaultTypes = exports.CometChatMessageTemplate = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _ = require("../..");
var _2 = require("..");
var _stickers = _interopRequireDefault(require("../CometChatMessageList/resources/stickers.svg"));
var _collaborativedocument = _interopRequireDefault(require("../CometChatMessageList/resources/collaborativedocument.svg"));
var _collaborativewhiteboard = _interopRequireDefault(require("../CometChatMessageList/resources/collaborativewhiteboard.svg"));
var _polls = _interopRequireDefault(require("../CometChatMessageList/resources/polls.svg"));
var _editicon = _interopRequireDefault(require("../CometChatMessageList/resources/editicon.svg"));
var _audioFile = _interopRequireDefault(require("../CometChatMessageList/resources/audio-file.svg"));
var _videoUpload = _interopRequireDefault(require("../CometChatMessageList/resources/video-upload.svg"));
var _fileUpload = _interopRequireDefault(require("../CometChatMessageList/resources/file-upload.svg"));
var _image = _interopRequireDefault(require("../CometChatMessageList/resources/image.svg"));
/**
 * @class CometChatMessageTemplate
 * @description CometChatMessageTemplate class is used for defining the message templates.
 * @param {String} type
 * @param {String} icon
 * @param {String} name
 * @param {Function} onActionClick
 * @param {Function} customView
 * @param {Array} options
 * @param {String} category
 */
var CometChatMessageTemplate = /*#__PURE__*/(0, _createClass2.default)(function CometChatMessageTemplate(_ref) {
  var _ref$type = _ref.type,
    type = _ref$type === void 0 ? "" : _ref$type,
    _ref$icon = _ref.icon,
    icon = _ref$icon === void 0 ? "" : _ref$icon,
    _ref$name = _ref.name,
    name = _ref$name === void 0 ? "" : _ref$name,
    _ref$onActionClick = _ref.onActionClick,
    onActionClick = _ref$onActionClick === void 0 ? null : _ref$onActionClick,
    _ref$customView = _ref.customView,
    customView = _ref$customView === void 0 ? null : _ref$customView,
    _ref$options = _ref.options,
    options = _ref$options === void 0 ? [] : _ref$options;
  (0, _classCallCheck2.default)(this, CometChatMessageTemplate);
  (0, _defineProperty2.default)(this, "type", "");
  (0, _defineProperty2.default)(this, "icon", "");
  (0, _defineProperty2.default)(this, "name", "");
  (0, _defineProperty2.default)(this, "onActionClick", null);
  (0, _defineProperty2.default)(this, "customView", null);
  (0, _defineProperty2.default)(this, "options", []);
  (0, _defineProperty2.default)(this, "category", "");
  this.type = type;
  this.icon = icon;
  this.name = name;
  this.onActionClick = onActionClick;
  this.customView = customView;
  this.options = options;
  this.category = CometChatMessageTemplate.getCategory(type);
});
exports.CometChatMessageTemplate = CometChatMessageTemplate;
(0, _defineProperty2.default)(CometChatMessageTemplate, "getCategory", function (type) {
  switch (type) {
    case _.MessageTypeConstants.text:
    case _.MessageTypeConstants.image:
    case _.MessageTypeConstants.video:
    case _.MessageTypeConstants.audio:
    case _.MessageTypeConstants.file:
      {
        return _.MessageCategoryConstants.message;
      }
    case _.MessageTypeConstants.poll:
    case _.MessageTypeConstants.sticker:
    case _.MessageTypeConstants.document:
    case _.MessageTypeConstants.whiteboard:
    case _.MessageTypeConstants.meeting:
      {
        return _.MessageCategoryConstants.custom;
      }
    case _.MessageTypeConstants.groupMember:
    case _.MessageTypeConstants.messageEdited:
    case _.MessageTypeConstants.messageDeleted:
      {
        return _.MessageCategoryConstants.action;
      }
    default:
      {
        return _.MessageCategoryConstants.custom;
      }
  }
});
(0, _defineProperty2.default)(CometChatMessageTemplate, "getMessageTemplate", function (type) {
  switch (type) {
    case _.MessageTypeConstants.text:
      {
        return new CometChatMessageTemplate({
          type: _.MessageTypeConstants.text,
          icon: _editicon.default,
          name: (0, _.localize)("TEXT"),
          onActionClick: null,
          customView: null,
          options: _2.CometChatMessageOptions.getDefaultOptions(_.MessageTypeConstants.text)
        });
      }
    case _.MessageTypeConstants.image:
      {
        return new CometChatMessageTemplate({
          type: _.MessageTypeConstants.image,
          icon: _image.default,
          name: (0, _.localize)("ATTACH_IMAGE"),
          onActionClick: null,
          customView: null,
          options: _2.CometChatMessageOptions.getDefaultOptions(_.MessageTypeConstants.image)
        });
      }
    case _.MessageTypeConstants.video:
      {
        return new CometChatMessageTemplate({
          type: _.MessageTypeConstants.video,
          icon: _videoUpload.default,
          name: (0, _.localize)("ATTACH_VIDEO"),
          onActionClick: null,
          customView: null,
          options: _2.CometChatMessageOptions.getDefaultOptions(_.MessageTypeConstants.video)
        });
      }
    case _.MessageTypeConstants.audio:
      {
        return new CometChatMessageTemplate({
          type: _.MessageTypeConstants.audio,
          icon: _audioFile.default,
          name: (0, _.localize)("ATTACH_AUDIO"),
          onActionClick: null,
          customView: null,
          options: _2.CometChatMessageOptions.getDefaultOptions(_.MessageTypeConstants.audio)
        });
      }
    case _.MessageTypeConstants.file:
      {
        return new CometChatMessageTemplate({
          type: _.MessageTypeConstants.file,
          icon: _fileUpload.default,
          name: (0, _.localize)("ATTACH_FILE"),
          onActionClick: null,
          customView: null,
          options: _2.CometChatMessageOptions.getDefaultOptions(_.MessageTypeConstants.file)
        });
      }
    case _.MessageTypeConstants.poll:
      {
        return new CometChatMessageTemplate({
          type: _.MessageTypeConstants.poll,
          icon: _polls.default,
          name: (0, _.localize)("CREATE_POLL"),
          onActionClick: null,
          customView: null,
          options: _2.CometChatMessageOptions.getDefaultOptions(_.MessageTypeConstants.poll)
        });
      }
    case _.MessageTypeConstants.sticker:
      {
        return new CometChatMessageTemplate({
          type: _.MessageTypeConstants.sticker,
          icon: _stickers.default,
          name: (0, _.localize)("STICKER"),
          onActionClick: null,
          customView: null,
          options: _2.CometChatMessageOptions.getDefaultOptions(_.MessageTypeConstants.sticker)
        });
      }
    case _.MessageTypeConstants.document:
      {
        return new CometChatMessageTemplate({
          type: _.MessageTypeConstants.document,
          icon: _collaborativedocument.default,
          name: (0, _.localize)("COLLABORATIVE_DOCUMENT"),
          onActionClick: null,
          customView: null,
          options: _2.CometChatMessageOptions.getDefaultOptions(_.MessageTypeConstants.document)
        });
      }
    case _.MessageTypeConstants.whiteboard:
      {
        return new CometChatMessageTemplate({
          type: _.MessageTypeConstants.whiteboard,
          icon: _collaborativewhiteboard.default,
          name: (0, _.localize)("COLLABORATIVE_WHITEBOARD"),
          onActionClick: null,
          customView: null,
          options: _2.CometChatMessageOptions.getDefaultOptions(_.MessageTypeConstants.whiteboard)
        });
      }
    default:
      return null;
  }
});
(0, _defineProperty2.default)(CometChatMessageTemplate, "getDefaultTypes", function () {
  var types = [CometChatMessageTemplate.getMessageTemplate(_.MessageTypeConstants.text), CometChatMessageTemplate.getMessageTemplate(_.MessageTypeConstants.image), CometChatMessageTemplate.getMessageTemplate(_.MessageTypeConstants.file), CometChatMessageTemplate.getMessageTemplate(_.MessageTypeConstants.audio), CometChatMessageTemplate.getMessageTemplate(_.MessageTypeConstants.video), CometChatMessageTemplate.getMessageTemplate(_.MessageTypeConstants.poll), CometChatMessageTemplate.getMessageTemplate(_.MessageTypeConstants.sticker), CometChatMessageTemplate.getMessageTemplate(_.MessageTypeConstants.document), CometChatMessageTemplate.getMessageTemplate(_.MessageTypeConstants.whiteboard)];
  return types;
});
var getDefaultTypes = CometChatMessageTemplate.getDefaultTypes;
exports.getDefaultTypes = getDefaultTypes;
