// react-ui-kit-testing/dist/Messages/CometChatCreatePollOptions/CreatePollOptionConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CreatePollOptionConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _delete = _interopRequireDefault(require("./resources/delete.svg"));
var _ = require("../../../");
var _2 = require("../");
/**
 * @class CreatePollOptionConfiguration
 * @description CreatePollOptionConfiguration class is used for defining the MessageComposer templates.
 * @param {Function} onDeleteClick
 * @param {Function} onChangeHandler
 * @param {String} option
 * @param {Boolean} hasDelete
 * @param {String} placeholderText
 * @param {String} deleteIconUrl
 * @param {Object} style
 */
var CreatePollOptionConfiguration = /*#__PURE__*/(0, _createClass2.default)(function CreatePollOptionConfiguration(_ref) {
  var _ref$hasDelete = _ref.hasDelete,
    hasDelete = _ref$hasDelete === void 0 ? false : _ref$hasDelete,
    _ref$placeholderText = _ref.placeholderText,
    placeholderText = _ref$placeholderText === void 0 ? (0, _.localize)("ANSWER") : _ref$placeholderText,
    _ref$deleteIconURL = _ref.deleteIconURL,
    deleteIconURL = _ref$deleteIconURL === void 0 ? _delete.default : _ref$deleteIconURL,
    _ref$onDeleteClick = _ref.onDeleteClick,
    onDeleteClick = _ref$onDeleteClick === void 0 ? null : _ref$onDeleteClick,
    _ref$onChangeHandler = _ref.onChangeHandler,
    onChangeHandler = _ref$onChangeHandler === void 0 ? null : _ref$onChangeHandler,
    _ref$option = _ref.option,
    option = _ref$option === void 0 ? null : _ref$option,
    _ref$style = _ref.style,
    style = _ref$style === void 0 ? new _2.CreatePollOptionStyles({}) : _ref$style;
  (0, _classCallCheck2.default)(this, CreatePollOptionConfiguration);
  this.hasDelete = hasDelete;
  this.placeholderText = placeholderText;
  this.deleteIconURL = deleteIconURL;
  this.onDeleteClick = onDeleteClick;
  this.onChangeHandler = onChangeHandler;
  this.option = option;
  this.style = new _2.CreatePollOptionStyles(style !== null && style !== void 0 ? style : {});
});
exports.CreatePollOptionConfiguration = CreatePollOptionConfiguration;
