// react-ui-kit-testing/dist/Messages/CometChatCreatePollOptions/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatCreatePollOptions = void 0;
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _style = require("./style");
var _delete = _interopRequireDefault(require("./resources/delete.svg"));
var _this = void 0;
var CometChatCreatePollOptions = function CometChatCreatePollOptions(props) {
  var onChange = function onChange(event) {
    props.onChangeHandler(event, props.option);
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "polloption__answer__wrapper",
    style: (0, _style.pollOptionAnswerStyle)()
  }, /*#__PURE__*/_react.default.createElement("input", {
    autoFocus: true,
    type: "text",
    tabIndex: "1",
    autoComplete: "off",
    style: (0, _style.pollOptionInputStyle)(props),
    value: props.option.value,
    placeholder: props.placeholderText,
    onChange: onChange.bind(_this)
  }), /*#__PURE__*/_react.default.createElement("span", {
    style: (0, _style.removeOptionIconStyle)(props, _delete.default),
    onClick: function onClick() {
      return props.onDeleteClick(props.option);
    }
  }));
};
exports.CometChatCreatePollOptions = CometChatCreatePollOptions;
CometChatCreatePollOptions.defaultProps = {
  hasDelete: false,
  placeholderText: "",
  deleteIconUrl: "",
  onDeleteClick: null,
  onChangeHandler: null,
  option: null,
  style: {
    width: "100%",
    height: "46px",
    deleteIconTint: "RGBA(20, 20, 20, 0.6)",
    border: "1px solid RGBA(20, 20, 20, 0.04)",
    borderRadius: "8px",
    boxShadow: "RGBA(20, 20, 20, 0.04) 0 16px 32px, RGBA(20, 20, 20, 0.04) 0 0 0 1px",
    background: "RGBA(20, 20, 20, 0.04)",
    placeholderTextFont: "400 15px Inter,sans-serif",
    placeholderTextColor: "RGBA(20, 20, 20, 0.6)",
    inputTextFont: "400 15px Inter,sans-serif",
    inputTextColor: "RGB(20, 20, 20)",
    inputStyle: null
  }
};
CometChatCreatePollOptions.propTypes = {
  option: _propTypes.default.object,
  hasDelete: _propTypes.default.bool,
  placeholderText: _propTypes.default.string,
  deleteIconUrl: _propTypes.default.string,
  style: _propTypes.default.object,
  onDeleteClick: _propTypes.default.func,
  onChangeHandler: _propTypes.default.func
};
