// cometchat-react-ui-kit/dist/Shared/UtilityComponents/CometChatListItem/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatListItem = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _style = require("./style");

var CometChatListItem = function CometChatListItem(props) {
  var getIcon = function getIcon() {
    return props.iconURL ? /*#__PURE__*/_react.default.createElement("div", {
      className: "item__background",
      style: (0, _style.iconBackgroundStyle)(props)
    }, /*#__PURE__*/_react.default.createElement("span", {
      className: "item__icon",
      style: (0, _style.listItemIconStyle)(props)
    })) : null;
  };

  var getText = function getText() {
    return props.text ? /*#__PURE__*/_react.default.createElement("div", {
      style: (0, _style.listTitle)(props),
      className: "item__text"
    }, props.text) : null;
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    id: props.id,
    style: (0, _style.listItem)(props),
    className: "list__item",
    onClick: props.onItemClick
  }, getIcon(), getText(), props.tail);
};

exports.CometChatListItem = CometChatListItem;
CometChatListItem.defaultProps = {
  id: "123",
  text: "",
  tail: null,
  iconURL: "",
  onItemClick: function onItemClick() {},
  style: {
    width: "",
    height: "",
    iconWidth: "",
    iconHeight: "",
    iconTint: "",
    borderRadius: "8px",
    iconBackground: "white",
    textColor: "rgb(51,153,255)",
    border: "",
    background: "rgba(255,255,255, 0.6)",
    textFont: "600 15px Inter, sans-serif"
  }
};
CometChatListItem.propTypes = {
  id: _propTypes.default.string,
  text: _propTypes.default.string,
  tail: _propTypes.default.object,
  iconURL: _propTypes.default.string,
  style: _propTypes.default.object,
  onItemClick: _propTypes.default.func
};
