// cometchat-react-ui-kit/dist/Messages/CometChatGroupActionBubble/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatGroupActionBubble = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _chat = require("@cometchat-pro/chat");

var _style = require("./style");

var _ = require("../..");

var CometChatGroupActionBubble = function CometChatGroupActionBubble(props) {
  var getActionMessage = function getActionMessage(props) {
    var _props$messageObject, _props$messageObject$, _props$messageObject2, _props$messageObject3;

    var actionMessage = null;
    var byUser = (_props$messageObject = props.messageObject) === null || _props$messageObject === void 0 ? void 0 : (_props$messageObject$ = _props$messageObject.actionBy) === null || _props$messageObject$ === void 0 ? void 0 : _props$messageObject$.name;
    var onUser = (_props$messageObject2 = props.messageObject) === null || _props$messageObject2 === void 0 ? void 0 : (_props$messageObject3 = _props$messageObject2.actionOn) === null || _props$messageObject3 === void 0 ? void 0 : _props$messageObject3.name;

    switch (props.messageObject.action) {
      case _chat.CometChat.ACTION_TYPE.MEMBER_JOINED:
        actionMessage = "".concat(byUser, " ").concat((0, _.localize)("JOINED"), " ");
        break;

      case _chat.CometChat.ACTION_TYPE.MEMBER_LEFT:
        actionMessage = "".concat(byUser, "  ").concat((0, _.localize)("LEFT"));
        break;

      case _chat.CometChat.ACTION_TYPE.MEMBER_ADDED:
        actionMessage = "".concat(byUser, " ").concat((0, _.localize)("ADDED"), " ").concat(onUser);
        break;

      case _chat.CometChat.ACTION_TYPE.MEMBER_KICKED:
        actionMessage = "".concat(byUser, "  ").concat((0, _.localize)("KICKED"), "  ").concat(onUser);
        break;

      case _chat.CometChat.ACTION_TYPE.MEMBER_BANNED:
        actionMessage = "".concat(byUser, "  ").concat((0, _.localize)("BANNED"), "  ").concat(onUser);
        break;

      case _chat.CometChat.ACTION_TYPE.MEMBER_UNBANNED:
        actionMessage = "".concat(byUser, "  ").concat((0, _.localize)("UNBANNED"), "  ").concat(onUser);
        break;

      case _chat.CometChat.ACTION_TYPE.MEMBER_SCOPE_CHANGED:
        {
          var _props$messageObject4;

          var newScope = (_props$messageObject4 = props.messageObject) === null || _props$messageObject4 === void 0 ? void 0 : _props$messageObject4.newScope;
          actionMessage = "".concat(byUser, " ").concat((0, _.localize)("MADE"), " ").concat(onUser, " ").concat(newScope);
          break;
        }

      default:
        break;
    }

    return actionMessage;
  };

  var message = function message() {
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
      style: (0, _style.actionMessageStyle)(props),
      className: "action__message"
    }, /*#__PURE__*/_react.default.createElement("p", {
      style: (0, _style.actionMessageTxtStyle)(props),
      className: "message__text"
    }, props.text ? props.text : getActionMessage(props))));
  };

  return message();
}; // Specifies the default values for props:


exports.CometChatGroupActionBubble = CometChatGroupActionBubble;
CometChatGroupActionBubble.defaultProps = {
  messagObject: {},
  text: "",
  style: {
    width: "",
    height: "",
    border: "",
    background: "",
    borderRadius: "",
    textFont: "",
    textColor: ""
  }
};
CometChatGroupActionBubble.propTypes = {
  messagObject: _propTypes.default.object,
  text: _propTypes.default.string,
  style: _propTypes.default.object
};
