// cometchat-react-ui-kit/dist/Shared/PrimaryComponents/CometChatConfiguration/ConversationsConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ConversationsConfiguration = void 0;

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _ = require(".");

var _Chats = require("../../../Chats/");

/**
 * @class ConversationsConfiguration
 * @param {String} backButtonIconURL
 * @param {String} startConversationIconURL
 * @param {String} searchIconURL
 * @param {String} showBackButton
 * @param {Boolean} hideStartConversation
 * @param {Boolean} hideSearch
 * @param {Object} style
 * @param {Object} conversationListConfiguration
 */
var ConversationsConfiguration = /*#__PURE__*/(0, _createClass2.default)(function ConversationsConfiguration(_ref) {
  var _ref$backButtonIconUR = _ref.backButtonIconURL,
      backButtonIconURL = _ref$backButtonIconUR === void 0 ? "" : _ref$backButtonIconUR,
      _ref$startConversatio = _ref.startConversationIconURL,
      startConversationIconURL = _ref$startConversatio === void 0 ? "" : _ref$startConversatio,
      _ref$searchIconURL = _ref.searchIconURL,
      searchIconURL = _ref$searchIconURL === void 0 ? "" : _ref$searchIconURL,
      _ref$showBackButton = _ref.showBackButton,
      showBackButton = _ref$showBackButton === void 0 ? false : _ref$showBackButton,
      _ref$hideStartConvers = _ref.hideStartConversation,
      hideStartConversation = _ref$hideStartConvers === void 0 ? true : _ref$hideStartConvers,
      _ref$hideSearch = _ref.hideSearch,
      hideSearch = _ref$hideSearch === void 0 ? true : _ref$hideSearch,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? new _Chats.ConversationsStyles({}) : _ref$style,
      _ref$conversationList = _ref.conversationListConfiguration,
      conversationListConfiguration = _ref$conversationList === void 0 ? new _.ConversationListConfiguration({}) : _ref$conversationList;
  (0, _classCallCheck2.default)(this, ConversationsConfiguration);
  this.backButtonIconURL = backButtonIconURL;
  this.startConversationIconURL = startConversationIconURL;
  this.searchIconURL = searchIconURL;
  this.showBackButton = showBackButton;
  this.hideStartConversation = hideStartConversation;
  this.hideSearch = hideSearch;
  this.style = new _Chats.ConversationsStyles(style !== null && style !== void 0 ? style : {});
  this.conversationListConfiguration = new _.ConversationListConfiguration(conversationListConfiguration !== null && conversationListConfiguration !== void 0 ? conversationListConfiguration : {});
});
exports.ConversationsConfiguration = ConversationsConfiguration;
