// react-ui-kit-testing/dist/Shared/UtilityComponents/CometChatListItem/ListItemConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListItemConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _fileUpload = _interopRequireDefault(require("../../../Messages/CometChatMessageComposer/resources/file-upload.svg"));
var _Shared = require("../../../Shared");
/**
 * @class ListItemConfiguration
 * @description ListItemConfiguration class is used for defining the ListItem templates.
 * @param {String} iconURL
 * @param {String} text
 * @param {Object} tail
 * @param {Function} onItemClick
 * @param {Number} id
 * @param {Object} style
 */
var ListItemConfiguration = /*#__PURE__*/(0, _createClass2.default)(function ListItemConfiguration() {
  var id = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : null;
  var text = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : null;
  var tail = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : null;
  var iconURL = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : null;
  var onItemClick = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : null;
  var style = arguments.length > 5 && arguments[5] !== undefined ? arguments[5] : new _Shared.ListItemStyles({});
  (0, _classCallCheck2.default)(this, ListItemConfiguration);
  this.id = id;
  this.text = text;
  this.tail = tail;
  this.iconURL = iconURL || _fileUpload.default;
  this.onItemClick = onItemClick;
  this.style = new _Shared.ListItemStyles(style !== null && style !== void 0 ? style : {});
});
exports.ListItemConfiguration = ListItemConfiguration;
