// @cometchat-pro/react-ui-kit/dist/UserProfile/CometChatUserProfileItem/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatUserProfileItem = void 0;
var _react = require("react");
var _core = require("@emotion/core");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _chat = require("@cometchat-pro/chat");
var _ = require("../..");
var _Shared = require("../../Shared");
var _style = require("./style");
/** @jsxRuntime classic */
/** @jsx jsx */

var CometChatUserProfileItem = function CometChatUserProfileItem(props) {
  return (0, _core.jsx)("div", {
    css: (0, _style.detailStyle)(),
    className: "userinfo__detail"
  }, (0, _core.jsx)("div", {
    css: (0, _style.thumbnailStyle)(),
    className: "detail__thumbnail"
  }, props.avatar), (0, _core.jsx)("div", {
    css: (0, _style.userDetailStyle)(),
    className: "detail__user",
    dir: (0, _.localize)(props.lang)
  }, (0, _core.jsx)("div", {
    css: (0, _style.userNameStyle)(),
    className: "user__name"
  }, props.title), (0, _core.jsx)("p", {
    css: (0, _style.userStatusStyle)(props),
    className: "user__status"
  }, (0, _.localize)("ONLINE"))));
};

// Specifies the default values for props:
exports.CometChatUserProfileItem = CometChatUserProfileItem;
CometChatUserProfileItem.defaultProps = {
  user: null,
  title: "",
  titleColor: "#141414",
  titleFont: "600 15px Inter",
  subTitle: "",
  subTitleColor: "rgba(20, 20, 20, 0.8)",
  subTitleFont: "400 13px Inter",
  backgroundColor: "#ffffff",
  statusIndicator: null,
  avatar: (0, _core.jsx)(_Shared.CometChatAvatar, {
    user: void 0
  })
};
CometChatUserProfileItem.propTypes = {
  user: _propTypes["default"].object,
  title: _propTypes["default"].string,
  titleColor: _propTypes["default"].string,
  titleFont: _propTypes["default"].string,
  subTitle: _propTypes["default"].string,
  subTitleColor: _propTypes["default"].string,
  subTitleFont: _propTypes["default"].string,
  backgroundColor: _propTypes["default"].string,
  statusIndicator: _propTypes["default"].object,
  avatar: _propTypes["default"].object
};
