// @cometchat-pro/react-ui-kit/dist/UserProfile/CometChatUserProfileItem/style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.userStatusStyle = exports.userNameStyle = exports.userDetailStyle = exports.thumbnailStyle = exports.detailStyle = void 0;
var detailStyle = function detailStyle() {
  return {
    padding: "16px",
    display: "flex",
    flexDirection: "row",
    justifyContent: "left",
    alignItems: "center"
  };
};
exports.detailStyle = detailStyle;
var thumbnailStyle = function thumbnailStyle() {
  return {
    display: "inline-block",
    width: "36px",
    height: "36px",
    flexShrink: "0"
  };
};
exports.thumbnailStyle = thumbnailStyle;
var userDetailStyle = function userDetailStyle() {
  return {
    width: "calc(100% - 45px)",
    flexGrow: "1",
    paddingLeft: "16px",
    "&[dir=rtl]": {
      paddingRight: "16px",
      paddingLeft: "0"
    }
  };
};
exports.userDetailStyle = userDetailStyle;
var userNameStyle = function userNameStyle() {
  return {
    margin: "0",
    fontSize: "15px",
    fontWeight: "600",
    display: "block",
    maxWidth: "100%",
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap"
  };
};
exports.userNameStyle = userNameStyle;
var userStatusStyle = function userStatusStyle(props) {
  return {
    fontSize: "13px",
    margin: "0",
    color: "#39f"
  };
};
exports.userStatusStyle = userStatusStyle;
