// @cometchat-pro/react-ui-kit/dist/Users/CometChatUserConstants/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.userStatusColorCode = exports.userStatus = exports.userOptions = exports.userObject = void 0;
var _chat = _interopRequireDefault(require("@cometchat-pro/chat"));
var userObject = {
  uid: "uid",
  avatar: "avatar",
  name: "name",
  status: "status",
  role: "role",
  metadata: "metadata"
};
exports.userObject = userObject;
var userStatus = {
  online: _chat["default"] === null || _chat["default"] === void 0 ? void 0 : _chat["default"].USER_STATUS.ONLINE,
  offline: _chat["default"] === null || _chat["default"] === void 0 ? void 0 : _chat["default"].USER_STATUS.OFFLINE
};
exports.userStatus = userStatus;
var userStatusColorCode = {
  online: "RGB(0, 200, 111)"
};
exports.userStatusColorCode = userStatusColorCode;
var userOptions = {
  videocall: "videocall",
  voicecall: "voicecall",
  info: "info"
};
exports.userOptions = userOptions;
