// react-ui-kit-testing/dist/Messages/CometChatNewMessageIndicator/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatNewMessageIndicator = void 0;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _Shared = require("../../Shared");
var _style = require("./style");
var _hooks = require("./hooks");
var CometChatNewMessageIndicator = function CometChatNewMessageIndicator(props) {
  var text = props.text,
    onClick = props.onClick,
    style = props.style,
    theme = props.theme;
  var _theme = theme || new _Shared.CometChatTheme({});
  var _React$useState = _react.default.useState(),
    _React$useState2 = (0, _slicedToArray2.default)(_React$useState, 2),
    messageText = _React$useState2[0],
    setMessageText = _React$useState2[1];
  (0, _hooks.Hooks)(text, setMessageText);
  return /*#__PURE__*/_react.default.createElement("div", {
    style: (0, _style.messagePaneTopStyle)(),
    className: "message_pane__top"
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: (0, _style.messagePaneBannerStyle)(style),
    className: "message_pane__banner"
  }, /*#__PURE__*/_react.default.createElement("div", {
    style: (0, _style.messagePaneUnreadBannerStyle)(),
    className: "message_pane__unread_banner__banner",
    title: (0, _Shared.localize)("JUMP")
  }, /*#__PURE__*/_react.default.createElement("button", {
    type: "button",
    style: (0, _style.messagePaneUnreadBannerMessageStyle)(style, _theme),
    className: "message_pane__unread_banner__msg",
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement("span", {
    style: (0, _style.iconArrowDownStyle)(style, _theme),
    className: "icon--arrow-down"
  }, "\u2193", " "), messageText))));
};
exports.CometChatNewMessageIndicator = CometChatNewMessageIndicator;
CometChatNewMessageIndicator.defaultProps = {
  text: "new messages",
  style: {
    textFont: "",
    textColor: "#ffffff",
    border: "",
    borderRadius: "6px",
    background: "transparent"
  },
  onClick: function onClick() {}
};
CometChatNewMessageIndicator.propTypes = {
  text: _propTypes.default.string,
  iconURL: _propTypes.default.string,
  style: _propTypes.default.object,
  onClick: _propTypes.default.func
};
