// react-ui-kit-testing/dist/Shared/UtilityComponents/CometChatActionSheetItem/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatActionSheetItem = void 0;
var React = _interopRequireWildcard(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _style = require("./style");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
//import { actionSheetItemWrapperStyle, actionSheetItemIconStyle, actionsheetItemTitleStyle } from "./style";

var CometChatActionSheetItem = function CometChatActionSheetItem(props) {
  var clickHandler = function clickHandler(event) {
    props.onActionItemClick(props);
  };
  return /*#__PURE__*/React.createElement("div", {
    className: "sheet__item",
    style: (0, _style.actionSheetItemWrapperStyle)(props),
    onClick: clickHandler
  }, /*#__PURE__*/React.createElement("div", {
    className: "item__iconbackground",
    style: (0, _style.IconBackgroundStyle)(props)
  }, /*#__PURE__*/React.createElement("span", {
    className: "item__icon",
    style: (0, _style.IconStyle)(props),
    title: props.title
  })), /*#__PURE__*/React.createElement("div", {
    className: "item__title",
    style: (0, _style.actionSheetItemTitleStyle)(props)
  }, props.title));
};
exports.CometChatActionSheetItem = CometChatActionSheetItem;
CometChatActionSheetItem.defaultProps = {
  id: "uploadphoto",
  width: "50px",
  height: "70px",
  borderRadius: "10px",
  background: "#eee",
  border: "0 none",
  title: "Upload photo",
  titleFont: "13px 400 Inter",
  titleColor: "#141414",
  iconURL: "",
  iconTint: "white",
  iconBackground: "rgb(105, 41, 202)",
  iconborderRadius: "10px",
  onActionItemClick: function onActionItemClick() {}
};
CometChatActionSheetItem.propTypes = {
  id: _propTypes.default.string,
  background: _propTypes.default.string,
  borderRadius: _propTypes.default.string,
  border: _propTypes.default.string,
  width: _propTypes.default.string,
  height: _propTypes.default.string,
  title: _propTypes.default.string,
  titleFont: _propTypes.default.string,
  titleColor: _propTypes.default.string,
  iconURL: _propTypes.default.string,
  iconTint: _propTypes.default.string,
  iconBackground: _propTypes.default.string,
  iconborderRadius: _propTypes.default.string,
  onActionItemClick: _propTypes.default.func
};
