// react-ui-kit-testing/dist/Groups/CometChatGroupConstants/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.groupTypeColorCode = exports.groupType = exports.groupOptions = exports.groupObject = exports.groupMemberScope = void 0;
var _chat = require("@cometchat-pro/chat");
var groupObject = {
  guid: "guid",
  icon: "icon",
  name: "name",
  type: "type",
  membersCount: "membersCount",
  onlineMembersCount: "onlineMembersCount",
  hasJoined: "hasJoined"
};
exports.groupObject = groupObject;
var groupType = {
  private: _chat.CometChat.GROUP_TYPE.PRIVATE,
  password: _chat.CometChat.GROUP_TYPE.PASSWORD,
  public: _chat.CometChat.GROUP_TYPE.PUBLIC
};
exports.groupType = groupType;
var groupTypeColorCode = {
  private: "#F7A500",
  password: "#00C86F"
};
exports.groupTypeColorCode = groupTypeColorCode;
var groupMemberScope = {
  // admin: CometChat.SCOPE.ADMIN,
  // moderator: CometChat.SCOPE.MODERATOR,
  // participant: CometChat.SCOPE.PARTICIPANT
};
exports.groupMemberScope = groupMemberScope;
var groupOptions = {
  videocall: "videocall",
  voicecall: "voicecall",
  info: "info"
};
exports.groupOptions = groupOptions;
