// @cometchat-pro/react-ui-kit/dist/UserProfile/CometChatUserProfile/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatUserProfile = void 0;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = _interopRequireDefault(require("react"));
var _core = require("@emotion/core");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _chat = require("@cometchat-pro/chat");
var _Shared = require("../../Shared");
var _ = require("..");
var _2 = require("../..");
var _style = require("./style");
var _notify = _interopRequireDefault(require("./resources/notify.svg"));
var _privacy = _interopRequireDefault(require("./resources/privacy.svg"));
var _chats = _interopRequireDefault(require("./resources/chats.svg"));
var _help = _interopRequireDefault(require("./resources/help.svg"));
var _warning = _interopRequireDefault(require("./resources/warning.svg"));
var _hooks = require("./hooks");
var _this = void 0;
var CometChatUserProfile = function CometChatUserProfile(props) {
  var _React$useState = _react["default"].useState(null),
    _React$useState2 = (0, _slicedToArray2["default"])(_React$useState, 2),
    loggedInUser = _React$useState2[0],
    setLoggedInUser = _React$useState2[1];
  var _React$useState3 = _react["default"].useState(null),
    _React$useState4 = (0, _slicedToArray2["default"])(_React$useState3, 2),
    error = _React$useState4[0],
    setError = _React$useState4[1];
  useEffect(function () {
    var userProfile = null;
    if (loggedInUser) {
      //let avatar = <CometChatAvatar user={state.loggedInUser} />;
      userProfile = (0, _core.jsx)(_react["default"].Fragment, null, (0, _core.jsx)("div", {
        css: (0, _style.headerStyle)(props),
        className: "userinfo__header"
      }, (0, _core.jsx)("h4", {
        css: (0, _style.headerTitleStyle)(props),
        className: "header__title"
      }, props.title)), (0, _core.jsx)(_.CometChatUserProfileItem, {
        user: loggedInUser
      }), (0, _core.jsx)("div", {
        css: (0, _style.optionsStyle)(),
        className: "userinfo__options"
      }, (0, _core.jsx)("div", {
        css: (0, _style.optionTitleStyle)(props),
        className: "options__title"
      }, (0, _2.localize)("PREFERENCES", props.lang)), (0, _core.jsx)("div", {
        css: (0, _style.optionListStyle)(),
        className: "options_list"
      }, (0, _core.jsx)("div", {
        css: (0, _style.optionStyle)(_notify["default"]),
        className: "option option-notification"
      }, (0, _core.jsx)("div", {
        css: (0, _style.optionNameStyle)(),
        className: "option_name"
      }, (0, _2.localize)("NOTIFICATIONS", props.lang))), (0, _core.jsx)("div", {
        css: (0, _style.optionStyle)(_privacy["default"]),
        className: "option option-privacy"
      }, (0, _core.jsx)("div", {
        css: (0, _style.optionNameStyle)(),
        className: "option_name"
      }, (0, _2.localize)("PRIVACY_AND_SECURITY", props.lang))), (0, _core.jsx)("div", {
        css: (0, _style.optionStyle)(_chats["default"]),
        className: "option option-chats"
      }, (0, _core.jsx)("div", {
        css: (0, _style.optionNameStyle)(),
        className: "option_name"
      }, (0, _2.localize)("CHATS", props.lang)))), (0, _core.jsx)("div", {
        css: (0, _style.optionTitleStyle)(_this.props),
        className: "options__title"
      }, (0, _2.localize)("OTHER", props.lang)), (0, _core.jsx)("div", {
        css: (0, _style.optionListStyle)(),
        className: "options_list"
      }, (0, _core.jsx)("div", {
        css: (0, _style.optionStyle)(_help["default"]),
        className: "option option-help"
      }, (0, _core.jsx)("div", {
        css: (0, _style.optionNameStyle)(),
        className: "option_name"
      }, (0, _2.localize)("HELP", props.lang))), (0, _core.jsx)("div", {
        css: (0, _style.optionStyle)(_warning["default"]),
        className: "option option-report"
      }, (0, _core.jsx)("div", {
        css: (0, _style.optionNameStyle)(),
        className: "option_name"
      }, (0, _2.localize)("REPORT_PROBLEM", props.lang))))));
    }
  }, []);
  (0, _hooks.hooks)(props, setLoggedInUser, setError);
  return (0, _core.jsx)("div", {
    css: (0, _style.userInfoScreenStyle)(props),
    className: "userinfo"
  }, userProfile, (0, _core.jsx)(_Shared.CometChatToastNotification, {
    ref: error
  }));
};

// Specifies the default values for props:
exports.CometChatUserProfile = CometChatUserProfile;
CometChatUserProfile.defaultProps = {
  lang: Translator.getDefaultLanguage(),
  theme: theme,
  title: Translator.translate("MORE", Translator.getDefaultLanguage()),
  titleFont: "bold 22px Inter",
  titleColor: "#141414",
  backgroundColor: "#ffffff"
};
CometChatUserProfile.propTypes = {
  lang: _propTypes["default"].string,
  theme: _propTypes["default"].object,
  title: _propTypes["default"].string,
  titleFont: _propTypes["default"].string,
  titleColor: _propTypes["default"].string,
  backgroundColor: _propTypes["default"].string
};
