// @cometchat-pro/react-ui-kit/dist/UserProfile/CometChatUserProfile/style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.userInfoScreenStyle = exports.optionsStyle = exports.optionTitleStyle = exports.optionStyle = exports.optionNameStyle = exports.optionListStyle = exports.headerTitleStyle = exports.headerStyle = void 0;
var userInfoScreenStyle = function userInfoScreenStyle(props) {
  return {
    display: "flex",
    flexDirection: "column!important",
    height: "100%",
    fontFamily: "".concat(props.theme.fontFamily),
    backgroundColor: props.backgroundColor,
    "*": {
      boxSizing: "border-box",
      fontFamily: "".concat(props.theme.fontFamily)
    }
  };
};
exports.userInfoScreenStyle = userInfoScreenStyle;
var headerStyle = function headerStyle(props) {
  return {
    padding: "16px",
    position: "relative",
    height: "70px",
    display: "flex",
    justifyContent: "center",
    alignItems: "center"
  };
};
exports.headerStyle = headerStyle;
var headerTitleStyle = function headerTitleStyle(props) {
  return {
    margin: "0",
    font: props.titleFont,
    color: props.titleColor,
    lineHeight: "26px",
    display: "inline-block",
    width: "100%",
    textAlign: "center"
  };
};
exports.headerTitleStyle = headerTitleStyle;
var optionsStyle = function optionsStyle() {
  return {
    height: "calc(100% - 145px)",
    overflowY: "auto",
    padding: "0 16px",
    display: "flex",
    flexDirection: "column",
    justifyContent: "left",
    alignItems: "flex-start"
  };
};
exports.optionsStyle = optionsStyle;
var optionTitleStyle = function optionTitleStyle(props) {
  return {
    margin: "5px 0",
    width: "100%",
    fontSize: "12px",
    color: "".concat(props.theme.color.helpText),
    textTransform: "uppercase"
  };
};
exports.optionTitleStyle = optionTitleStyle;
var optionListStyle = function optionListStyle() {
  return {
    padding: "10px 0",
    width: "100%",
    fontSize: "15px"
  };
};
exports.optionListStyle = optionListStyle;
var optionStyle = function optionStyle(img) {
  return {
    width: "100%",
    padding: "16px 16px 16px 36px",
    fontWeight: 600,
    background: "url(".concat(img, ") left center no-repeat")
  };
};
exports.optionStyle = optionStyle;
var optionNameStyle = function optionNameStyle() {
  return {
    width: "100%"
  };
};
exports.optionNameStyle = optionNameStyle;
