// cometchat-react-ui-kit/dist/Groups/CometChatJoinProtectedGroup/style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.joinGroupTitleStyle = exports.joinGroupPasswordInputStyle = exports.joinGroupContainerStyle = exports.joinGroupButtonStyle = exports.errorTextStyle = exports.errorIconStyle = exports.errorContainerStyle = void 0;

var _ = require("../..");

var joinGroupContainerStyle = function joinGroupContainerStyle() {
  return {
    margin: "auto",
    width: "100%",
    height: "100%",
    borderRadius: "8px",
    zIndex: "4",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center"
  };
};

exports.joinGroupContainerStyle = joinGroupContainerStyle;

var joinGroupTitleStyle = function joinGroupTitleStyle(style, theme) {
  return {
    font: style === null || style === void 0 ? void 0 : style.titleTextFont,
    color: style === null || style === void 0 ? void 0 : style.titleTextColor
  };
};

exports.joinGroupTitleStyle = joinGroupTitleStyle;

var joinGroupPasswordInputStyle = function joinGroupPasswordInputStyle(style, theme) {
  return {
    width: "320px",
    height: "46px",
    color: style === null || style === void 0 ? void 0 : style.passwordTextFont,
    font: style === null || style === void 0 ? void 0 : style.passwordTextColor,
    background: style === null || style === void 0 ? void 0 : style.passwordInputBackground,
    border: style === null || style === void 0 ? void 0 : style.passwordInputBorder,
    borderRadius: style === null || style === void 0 ? void 0 : style.passwordInputBorderRadius,
    boxShadow: style === null || style === void 0 ? void 0 : style.passwordInputBoxShadow,
    margin: "38px 0px 16px 0px"
  };
};

exports.joinGroupPasswordInputStyle = joinGroupPasswordInputStyle;

var joinGroupButtonStyle = function joinGroupButtonStyle(style, theme) {
  return {
    width: "320px",
    height: "46px",
    font: style === null || style === void 0 ? void 0 : style.joinButtonTextFont,
    color: style === null || style === void 0 ? void 0 : style.joinButtonTextColor,
    background: style === null || style === void 0 ? void 0 : style.joinButtonBackground,
    borderRadius: style === null || style === void 0 ? void 0 : style.joinButtonBorderRadius
  };
};

exports.joinGroupButtonStyle = joinGroupButtonStyle;

var errorContainerStyle = function errorContainerStyle(style, theme) {
  return {
    width: "328px",
    height: "60px",
    margin: "16px 0px 32px 0px",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    background: "rgba(255, 59, 48, 0.1)",
    border: "none",
    borderRadius: "8px"
  };
};

exports.errorContainerStyle = errorContainerStyle;

var errorIconStyle = function errorIconStyle() {
  return {
    iconBackground: "RGB(255, 59, 48)",
    iconTint: "white",
    iconBackgroundWidth: "38px",
    iconBackgroundHeight: "38px",
    iconWidth: "24px",
    iconHeight: "24px",
    margin: "0px 12px"
  };
};

exports.errorIconStyle = errorIconStyle;

var errorTextStyle = function errorTextStyle(theme) {
  return {
    color: theme.palette.getError(),
    font: (0, _.fontHelper)(theme.typography.text1)
  };
};

exports.errorTextStyle = errorTextStyle;
