// @cometchat-pro/react-ui-kit/dist/Shared/SecondaryComponents/CometChatStatusIndicator/StatusIndicatorStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StatusIndicatorStyles = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _Shared = require("../../../Shared");
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
 * @class StatusIndicatorStyles
 * @param {String} backgroundColor
 */var StatusIndicatorStyles = /*#__PURE__*/function (_BaseStyles) {
  (0, _inherits2["default"])(StatusIndicatorStyles, _BaseStyles);
  var _super = _createSuper(StatusIndicatorStyles);
  function StatusIndicatorStyles(_ref) {
    var _this;
    var _ref$backgroundColor = _ref.backgroundColor,
      backgroundColor = _ref$backgroundColor === void 0 ? "" : _ref$backgroundColor,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "14px" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "14px" : _ref$height,
      background = _ref.background,
      activeBackground = _ref.activeBackground,
      _ref$border = _ref.border,
      border = _ref$border === void 0 ? "" : _ref$border,
      _ref$borderRadius = _ref.borderRadius,
      borderRadius = _ref$borderRadius === void 0 ? "50%" : _ref$borderRadius;
    (0, _classCallCheck2["default"])(this, StatusIndicatorStyles);
    _this = _super.call(this, {
      width: width,
      height: height,
      background: background,
      activeBackground: activeBackground,
      border: border,
      borderRadius: borderRadius
    });
    _this.backgroundColor = backgroundColor;
    return _this;
  }
  return (0, _createClass2["default"])(StatusIndicatorStyles);
}(_Shared.BaseStyles);
exports.StatusIndicatorStyles = StatusIndicatorStyles;
