// react-ui-kit-testing/dist/Messages/Bubbles/CometChatLocationBubble/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatLocationBubble = void 0;
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _hooks = require("./hooks");
require("./style");
/**
 *
 * CometChatTextBubble is UI component for file message bubble.
 *
 * @version 1.0.0
 * @author CometChatTeam
 * @copyright © 2022 CometChat Inc.
 *
 */
var CometChatLocationBubble = function CometChatLocationBubble(props) {
  var fileMessage = function fileMessage() {
    console.log("location bubble");
  };
  (0, _hooks.Hooks)(props);
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("h1", null, "Location"));
};
exports.CometChatLocationBubble = CometChatLocationBubble;
CometChatLocationBubble.defaultProps = {
  messageObject: {},
  title: "",
  subTitle: "",
  mimeType: "",
  iconURL: "",
  style: {
    width: "",
    height: "",
    titleFont: "400 15px Inter, sans-serif",
    titleColor: "rgb(20,20,20)",
    subTitleFont: "400 12px Inter, sans-serif",
    subTitleColor: "rgb(20,20,20, 0.60)",
    iconTint: "#fff",
    background: "",
    borderRadius: ""
  }
};
CometChatLocationBubble.propTypes = {
  messageObject: _propTypes.default.object,
  iconURL: _propTypes.default.string,
  title: _propTypes.default.string,
  subTitle: _propTypes.default.string,
  mimeType: _propTypes.default.string,
  style: _propTypes.default.object
};
