// react-ui-kit-testing/dist/Messages/CometChatMessages/style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.messageListStyle = exports.messageHeaderStyle = exports.messageComposerStyle = exports.liveReactionWrapperStyle = exports.liveReactionStyle = exports.chatWrapperStyle = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _Shared = require("../../Shared");
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
var chatWrapperStyle = function chatWrapperStyle(style) {
  return {
    display: "flex",
    flexDirection: "column",
    width: style.width,
    height: style.height,
    margin: "none",
    boxSizing: "border-box",
    position: "relative",
    overflowX: "hidden",
    border: style.border
  };
};
exports.chatWrapperStyle = chatWrapperStyle;
var liveReactionWrapperStyle = function liveReactionWrapperStyle() {
  return _objectSpread(_objectSpread({}, new _Shared.BaseStyles({
    width: "100%",
    height: "100%",
    background: "none",
    border: "none",
    borderRadius: "none",
    activeBackground: ""
  })), {}, {
    position: "absolute",
    top: "0",
    right: "0",
    zIndex: "2",
    display: "flex",
    justifyContent: "left",
    alignItems: "center",
    boxSizing: "border-box"
  });
};
exports.liveReactionWrapperStyle = liveReactionWrapperStyle;
var messageComposerStyle = function messageComposerStyle(theme, config) {
  var _theme$palette, _theme$palette2, _theme$typography, _theme$palette3, _theme$typography2, _theme$palette4, _theme$palette5, _theme$palette6, _theme$palette7, _theme$palette8, _theme$palette9, _theme$palette10;
  return _objectSpread(_objectSpread({}, new _Shared.BaseStyles({
    width: "100%",
    height: "auto",
    background: theme === null || theme === void 0 ? void 0 : (_theme$palette = theme.palette) === null || _theme$palette === void 0 ? void 0 : _theme$palette.getBackground(),
    border: "none",
    borderRadius: "12px",
    activeBackground: "none"
  })), {}, {
    inputBorderRadius: "8px",
    inputBackground: theme === null || theme === void 0 ? void 0 : (_theme$palette2 = theme.palette) === null || _theme$palette2 === void 0 ? void 0 : _theme$palette2.getAccent50(),
    inputTextFont: (0, _Shared.fontHelper)(theme === null || theme === void 0 ? void 0 : (_theme$typography = theme.typography) === null || _theme$typography === void 0 ? void 0 : _theme$typography.subtitle1),
    inputTextColor: theme === null || theme === void 0 ? void 0 : (_theme$palette3 = theme.palette) === null || _theme$palette3 === void 0 ? void 0 : _theme$palette3.getAccent(),
    placeholderTextFont: (0, _Shared.fontHelper)(theme === null || theme === void 0 ? void 0 : (_theme$typography2 = theme.typography) === null || _theme$typography2 === void 0 ? void 0 : _theme$typography2.subtitle1),
    placeholderTextColor: theme === null || theme === void 0 ? void 0 : (_theme$palette4 = theme.palette) === null || _theme$palette4 === void 0 ? void 0 : _theme$palette4.getAccent600(),
    emojiIconTint: theme === null || theme === void 0 ? void 0 : (_theme$palette5 = theme.palette) === null || _theme$palette5 === void 0 ? void 0 : _theme$palette5.getAccent500(),
    attachmentIconTint: theme === null || theme === void 0 ? void 0 : (_theme$palette6 = theme.palette) === null || _theme$palette6 === void 0 ? void 0 : _theme$palette6.getAccent500(),
    microphoneIconTint: theme === null || theme === void 0 ? void 0 : (_theme$palette7 = theme.palette) === null || _theme$palette7 === void 0 ? void 0 : _theme$palette7.getAccent500(),
    sendButtonIconTint: theme === null || theme === void 0 ? void 0 : (_theme$palette8 = theme.palette) === null || _theme$palette8 === void 0 ? void 0 : _theme$palette8.getAccent300(),
    stickerIconTint: theme === null || theme === void 0 ? void 0 : (_theme$palette9 = theme.palette) === null || _theme$palette9 === void 0 ? void 0 : _theme$palette9.getAccent500(),
    stickerCloseIconTint: theme === null || theme === void 0 ? void 0 : (_theme$palette10 = theme.palette) === null || _theme$palette10 === void 0 ? void 0 : _theme$palette10.getPrimary()
  });
};
exports.messageComposerStyle = messageComposerStyle;
var messageHeaderStyle = function messageHeaderStyle(theme) {
  var _theme$palette11, _theme$palette12;
  return _objectSpread(_objectSpread({}, new _Shared.BaseStyles({
    width: "100%",
    height: "auto",
    background: theme === null || theme === void 0 ? void 0 : (_theme$palette11 = theme.palette) === null || _theme$palette11 === void 0 ? void 0 : _theme$palette11.getBackground(),
    border: "none",
    borderRadius: "none",
    activeBackground: "none"
  })), {}, {
    backButtonIconTint: theme === null || theme === void 0 ? void 0 : (_theme$palette12 = theme.palette) === null || _theme$palette12 === void 0 ? void 0 : _theme$palette12.getPrimary()
  });
};
exports.messageHeaderStyle = messageHeaderStyle;
var messageListStyle = function messageListStyle(props, theme) {
  var _theme$palette13, _theme$typography3, _theme$palette14;
  return _objectSpread(_objectSpread({}, new _Shared.BaseStyles({
    width: "100%",
    height: "100%",
    background: theme === null || theme === void 0 ? void 0 : (_theme$palette13 = theme.palette) === null || _theme$palette13 === void 0 ? void 0 : _theme$palette13.getBackground(),
    border: "none",
    borderRadius: "8px",
    activeBackground: "none"
  })), {}, {
    textFont: (0, _Shared.fontHelper)(theme === null || theme === void 0 ? void 0 : (_theme$typography3 = theme.typography) === null || _theme$typography3 === void 0 ? void 0 : _theme$typography3.subtitle2),
    textColor: theme === null || theme === void 0 ? void 0 : (_theme$palette14 = theme.palette) === null || _theme$palette14 === void 0 ? void 0 : _theme$palette14.getAccent()
  });
};
exports.messageListStyle = messageListStyle;
var liveReactionStyle = function liveReactionStyle() {
  return _objectSpread({}, new _Shared.BaseStyles({
    width: "20px",
    height: "20px",
    background: "transparent",
    border: "none",
    borderRadius: "none",
    activeBackground: "transparent"
  }));
};
exports.liveReactionStyle = liveReactionStyle;
