// @cometchat-pro/react-ui-kit/dist/Groups/CometChatGroupsWithMessages/GroupsWithMessagesConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GroupsWithMessagesConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _ = require("../../../");
var _Groups = require("../../Groups");
var _JoinProtectedGroupConfiguration = require("../CometChatJoinProtectedGroup/JoinProtectedGroupConfiguration");
/**
 * @class GroupsWithMessagesConfiguration
 * @param {Object} style
 * @param {Boolean} isMobileView
 * @param {Object} groupsConfiguration
 * @param {Object} MessagesConfiguration
 * @param {Object} joinProtectedGroupConfiguration
 */var GroupsWithMessagesConfiguration = /*#__PURE__*/(0, _createClass2["default"])(function GroupsWithMessagesConfiguration(_ref) {
  var _ref$style = _ref.style,
    style = _ref$style === void 0 ? new _Groups.GroupsWithMessagesStyles({}) : _ref$style,
    _ref$isMobileView = _ref.isMobileView,
    isMobileView = _ref$isMobileView === void 0 ? false : _ref$isMobileView,
    _ref$groupsConfigurat = _ref.groupsConfiguration,
    groupsConfiguration = _ref$groupsConfigurat === void 0 ? new _.GroupsConfiguration({}) : _ref$groupsConfigurat,
    _ref$messagesConfigur = _ref.messagesConfiguration,
    messagesConfiguration = _ref$messagesConfigur === void 0 ? new _.MessagesConfiguration({}) : _ref$messagesConfigur,
    _ref$joinProtectedGro = _ref.joinProtectedGroupConfiguration,
    joinProtectedGroupConfiguration = _ref$joinProtectedGro === void 0 ? new _JoinProtectedGroupConfiguration.JoinProtectedGroupConfiguration({}) : _ref$joinProtectedGro;
  (0, _classCallCheck2["default"])(this, GroupsWithMessagesConfiguration);
  this.style = new _Groups.GroupsWithMessagesStyles(style !== null && style !== void 0 ? style : {});
  this.isMobileView = isMobileView;
  this.groupsConfiguration = new _.GroupsConfiguration(groupsConfiguration !== null && groupsConfiguration !== void 0 ? groupsConfiguration : {});
  this.messagesConfiguration = new _.MessagesConfiguration(messagesConfiguration !== null && messagesConfiguration !== void 0 ? messagesConfiguration : {});
  this.joinProtectedGroupConfiguration = new _JoinProtectedGroupConfiguration.JoinProtectedGroupConfiguration(joinProtectedGroupConfiguration !== null && joinProtectedGroupConfiguration !== void 0 ? joinProtectedGroupConfiguration : {});
});
exports.GroupsWithMessagesConfiguration = GroupsWithMessagesConfiguration;
