// @cometchat-pro/react-ui-kit/dist/Shared/UtilityComponents/CometChatActionSheetItem/style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.actionSheetItemWrapperStyle = exports.actionSheetItemTitleStyle = exports.IconStyle = exports.IconBackgroundStyle = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
var actionSheetItemWrapperStyle = function actionSheetItemWrapperStyle(props) {
  return _objectSpread({
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    background: props.background,
    padding: "8px 16px",
    width: props.width,
    height: props.height,
    overflowX: "hidden",
    overflowY: "auto",
    cursor: "pointer"
  }, props.style);
};
exports.actionSheetItemWrapperStyle = actionSheetItemWrapperStyle;
var IconBackgroundStyle = function IconBackgroundStyle(props) {
  return {
    width: "38px",
    height: "38px",
    borderRadius: "50%",
    background: props.iconBackground,
    cursor: "pointer",
    display: "flex",
    justifyContent: "center",
    alignItems: "center"
  };
};
exports.IconBackgroundStyle = IconBackgroundStyle;
var IconStyle = function IconStyle(props) {
  return {
    width: "24px",
    height: "24px",
    WebkitMask: "url(".concat(props.iconURL, ") center center no-repeat"),
    backgroundColor: props.iconTint,
    flexShrink: "0",
    display: "inline-block"
  };
};
exports.IconStyle = IconStyle;
var actionSheetItemTitleStyle = function actionSheetItemTitleStyle(props) {
  return {
    font: props.titleFont,
    color: props.titleColor,
    wordBreak: "break-word",
    margin: "6px"
  };
};
exports.actionSheetItemTitleStyle = actionSheetItemTitleStyle;
