// @cometchat-pro/react-ui-kit/dist/Messages/CometChatMessageHeader/MessageHeaderconfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageHeaderConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _back = _interopRequireDefault(require("../CometChatMessages//resources/back.svg"));
var _Shared = require("../../Shared");
/**
 * @class MessageHeaderConfiguration
 * @description MessageHeaderConfiguration class is used for defining the MessageHeader templates.
 * @param {Boolean} showBackButton
 * @param {Boolean} enableTypingIndicator
 * @param {String} backButtonIconURL
 * @param {Array} options
 */var MessageHeaderConfiguration = /*#__PURE__*/(0, _createClass2["default"])(function MessageHeaderConfiguration(_ref) {
  var _ref$showBackButton = _ref.showBackButton,
    showBackButton = _ref$showBackButton === void 0 ? true : _ref$showBackButton,
    _ref$backButtonIconUR = _ref.backButtonIconURL,
    backButtonIconURL = _ref$backButtonIconUR === void 0 ? _back["default"] : _ref$backButtonIconUR,
    _ref$enableTypingIndi = _ref.enableTypingIndicator,
    enableTypingIndicator = _ref$enableTypingIndi === void 0 ? false : _ref$enableTypingIndi,
    _ref$dataItemConfigur = _ref.dataItemConfiguration,
    dataItemConfiguration = _ref$dataItemConfigur === void 0 ? new _Shared.DataItemConfiguration({}) : _ref$dataItemConfigur;
  (0, _classCallCheck2["default"])(this, MessageHeaderConfiguration);
  this.showBackButton = showBackButton;
  this.backButtonIconURL = backButtonIconURL;
  this.enableTypingIndicator = enableTypingIndicator;
  //this.options = options;
  this.dataItemConfiguration = dataItemConfiguration;
});
exports.MessageHeaderConfiguration = MessageHeaderConfiguration;
