// react-ui-kit-testing/dist/Messages/Bubbles/CometChatAudioBubble/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatAudioBubble = void 0;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _hooks = require("./hooks");
var _style = require("./style");
var CometChatAudioBubble = function CometChatAudioBubble(props) {
  var _React$useState = _react.default.useState(""),
    _React$useState2 = (0, _slicedToArray2.default)(_React$useState, 2),
    audioURL = _React$useState2[0],
    setAudioURL = _React$useState2[1];
  var audioMessage = function audioMessage() {
    return /*#__PURE__*/_react.default.createElement("div", {
      style: (0, _style.messageAudioBubbleStyle)(props),
      className: "message_kit__blocks"
    }, /*#__PURE__*/_react.default.createElement("div", {
      style: (0, _style.messageAudioBubbleBlockStyle)(props),
      className: "message__message-blocks"
    }, /*#__PURE__*/_react.default.createElement("audio", {
      controls: true,
      className: "message__message-audio",
      style: (0, _style.messageBlockAudioStyle)(props),
      src: audioURL
    })));
  };
  (0, _hooks.Hooks)(props, setAudioURL);
  return audioMessage();
};
exports.CometChatAudioBubble = CometChatAudioBubble;
CometChatAudioBubble.defaultProps = {
  messageObject: null,
  audioURL: null,
  style: {
    width: "228px",
    height: "auto",
    borderRadius: "12px",
    background: "linear-gradient(to right, red , yellow)",
    border: "0 none"
  }
};
CometChatAudioBubble.propTypes = {
  messageObject: _propTypes.default.object,
  audioURL: _propTypes.default.string,
  style: _propTypes.default.object
};
