// @cometchat-pro/react-ui-kit/dist/Shared/InputData/InputData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputData = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
/**
 * @class InputData
 * @description InputData class is data model for several list item and data Item components
 * @param {String} id
 * @param {String} thumbnail
 * @param {String} status
 * @param {String} title
 * @param {String} subtitle
 */var InputData = /*#__PURE__*/(0, _createClass2["default"])(function InputData(_ref) {
  var _ref$id = _ref.id,
    id = _ref$id === void 0 ? "" : _ref$id,
    _ref$thumbnail = _ref.thumbnail,
    thumbnail = _ref$thumbnail === void 0 ? true : _ref$thumbnail,
    _ref$status = _ref.status,
    status = _ref$status === void 0 ? true : _ref$status,
    _ref$title = _ref.title,
    title = _ref$title === void 0 ? true : _ref$title,
    _ref$subtitle = _ref.subtitle,
    subtitle = _ref$subtitle === void 0 ? null : _ref$subtitle;
  (0, _classCallCheck2["default"])(this, InputData);
  this.id = id;
  this.thumbnail = thumbnail;
  this.status = status;
  this.title = title;
  this.subtitle = subtitle;
});
exports.InputData = InputData;
