// cometchat-react-ui-kit/dist/Users/CometChatUsers/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CometChatUsers = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ = require("../..");

var _style = require("./style");

var _back = _interopRequireDefault(require("./resources/back.svg"));

var _search = _interopRequireDefault(require("./resources/search.svg"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/**
 *
 * CometChatConversations is a container component that wraps and
 * formats CometChatListBase and CometChatConversationList component, with no behavior of its own.
 *
 * @version 1.0.0
 * @author CometChatTeam
 * @copyright © 2022 CometChat Inc.
 *
 */
var Users = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  var title = props.title,
      searchPlaceholder = props.searchPlaceholder,
      backButtonIconURL = props.backButtonIconURL,
      showBackButton = props.showBackButton,
      searchIconURL = props.searchIconURL,
      hideSearch = props.hideSearch,
      activeUser = props.activeUser,
      theme = props.theme,
      style = props.style,
      userListConfiguration = props.userListConfiguration;
  /**
   * Component internal state
   */

  var userListRef = _react.default.useRef(null);

  var _React$useState = _react.default.useState(""),
      _React$useState2 = (0, _slicedToArray2.default)(_React$useState, 2),
      searchInput = _React$useState2[0],
      setSearchInput = _React$useState2[1];

  var _theme = new _.CometChatTheme(theme || {});

  var _userListConfiguration = userListConfiguration || new _.UserListConfiguration({});

  _react.default.useImperativeHandle(ref, function () {
    return {
      userListRef: userListRef.current
    };
  });

  var searchHandler = function searchHandler(searchText) {
    setSearchInput(searchText);
  };

  var updateUsers = function updateUsers(payload) {
    if (userListRef !== null && userListRef !== void 0 && userListRef.current) {
      userListRef.current.updateUser(payload);
    }
  };

  _.CometChatUserEvents.addListener(_.CometChatUserEvents.onUserBlock, "onUserBlock", updateUsers);

  _.CometChatUserEvents.addListener(_.CometChatUserEvents.onUserUnblock, "onUserUnblock", updateUsers);

  var getListBaseStyle = function getListBaseStyle() {
    var _props$style, _props$style2, _theme$palette, _props$style3, _theme$palette2, _props$style4, _theme$typography, _props$style5, _theme$palette3, _theme$palette4, _props$style6, _theme$palette5, _props$style7, _theme$typography2, _props$style8, _theme$palette6, _props$style9, _theme$palette7;

    return {
      width: "100%",
      height: "100%",
      border: "0px none",
      borderRadius: (_props$style = props.style) === null || _props$style === void 0 ? void 0 : _props$style.borderRadius,
      background: ((_props$style2 = props.style) === null || _props$style2 === void 0 ? void 0 : _props$style2.background) || (_theme === null || _theme === void 0 ? void 0 : (_theme$palette = _theme.palette) === null || _theme$palette === void 0 ? void 0 : _theme$palette.getBackground()),
      titleColor: ((_props$style3 = props.style) === null || _props$style3 === void 0 ? void 0 : _props$style3.titleColor) || (_theme === null || _theme === void 0 ? void 0 : (_theme$palette2 = _theme.palette) === null || _theme$palette2 === void 0 ? void 0 : _theme$palette2.getAccent()),
      titleFont: ((_props$style4 = props.style) === null || _props$style4 === void 0 ? void 0 : _props$style4.titleFont) || (0, _.fontHelper)(_theme === null || _theme === void 0 ? void 0 : (_theme$typography = _theme.typography) === null || _theme$typography === void 0 ? void 0 : _theme$typography.heading),
      backIconTint: ((_props$style5 = props.style) === null || _props$style5 === void 0 ? void 0 : _props$style5.backIconTint) || (_theme === null || _theme === void 0 ? void 0 : (_theme$palette3 = _theme.palette) === null || _theme$palette3 === void 0 ? void 0 : _theme$palette3.getPrimary()),
      searchBorder: "1px solid ".concat(_theme === null || _theme === void 0 ? void 0 : (_theme$palette4 = _theme.palette) === null || _theme$palette4 === void 0 ? void 0 : _theme$palette4.getAccent50()),
      searchborderRadius: "8px",
      searchBackground: ((_props$style6 = props.style) === null || _props$style6 === void 0 ? void 0 : _props$style6.searchBackground) || (_theme === null || _theme === void 0 ? void 0 : (_theme$palette5 = _theme.palette) === null || _theme$palette5 === void 0 ? void 0 : _theme$palette5.getAccent50()),
      searchTextFont: ((_props$style7 = props.style) === null || _props$style7 === void 0 ? void 0 : _props$style7.searchTextFont) || (0, _.fontHelper)(_theme === null || _theme === void 0 ? void 0 : (_theme$typography2 = _theme.typography) === null || _theme$typography2 === void 0 ? void 0 : _theme$typography2.subtitle1),
      searchTextColor: ((_props$style8 = props.style) === null || _props$style8 === void 0 ? void 0 : _props$style8.searchTextColor) || (_theme === null || _theme === void 0 ? void 0 : (_theme$palette6 = _theme.palette) === null || _theme$palette6 === void 0 ? void 0 : _theme$palette6.getAccent500()),
      searchIconTint: ((_props$style9 = props.style) === null || _props$style9 === void 0 ? void 0 : _props$style9.searchIconTint) || (_theme === null || _theme === void 0 ? void 0 : (_theme$palette7 = _theme.palette) === null || _theme$palette7 === void 0 ? void 0 : _theme$palette7.getAccent500())
    };
  };

  var getListStyle = function getListStyle() {
    var _userListConfiguratio, _userListConfiguratio2, _userListConfiguratio3, _userListConfiguratio4, _theme$palette8, _userListConfiguratio5, _userListConfiguratio6, _userListConfiguratio7, _theme$typography3, _userListConfiguratio8, _theme$palette9, _userListConfiguratio9, _theme$typography4, _userListConfiguratio10, _theme$palette10;

    return {
      width: ((_userListConfiguratio = _userListConfiguration.style) === null || _userListConfiguratio === void 0 ? void 0 : _userListConfiguratio.width) || "100%",
      height: ((_userListConfiguratio2 = _userListConfiguration.style) === null || _userListConfiguratio2 === void 0 ? void 0 : _userListConfiguratio2.height) || "100%",
      border: ((_userListConfiguratio3 = _userListConfiguration.style) === null || _userListConfiguratio3 === void 0 ? void 0 : _userListConfiguratio3.border) || "0 none",
      loadingIconTint: ((_userListConfiguratio4 = _userListConfiguration.style) === null || _userListConfiguratio4 === void 0 ? void 0 : _userListConfiguratio4.loadingIconTint) || (_theme === null || _theme === void 0 ? void 0 : (_theme$palette8 = _theme.palette) === null || _theme$palette8 === void 0 ? void 0 : _theme$palette8.getAccent600()),
      borderRadius: ((_userListConfiguratio5 = _userListConfiguration.style) === null || _userListConfiguratio5 === void 0 ? void 0 : _userListConfiguratio5.borderRadius) || "0px",
      background: ((_userListConfiguratio6 = _userListConfiguration.style) === null || _userListConfiguratio6 === void 0 ? void 0 : _userListConfiguratio6.background) || "transparent",
      emptyTextFont: ((_userListConfiguratio7 = _userListConfiguration.style) === null || _userListConfiguratio7 === void 0 ? void 0 : _userListConfiguratio7.emptyTextFont) || (0, _.fontHelper)(_theme === null || _theme === void 0 ? void 0 : (_theme$typography3 = _theme.typography) === null || _theme$typography3 === void 0 ? void 0 : _theme$typography3.heading),
      emptyTextColor: ((_userListConfiguratio8 = _userListConfiguration.style) === null || _userListConfiguratio8 === void 0 ? void 0 : _userListConfiguratio8.emptyTextColor) || (_theme === null || _theme === void 0 ? void 0 : (_theme$palette9 = _theme.palette) === null || _theme$palette9 === void 0 ? void 0 : _theme$palette9.getAccent400()),
      errorTextFont: ((_userListConfiguratio9 = _userListConfiguration.style) === null || _userListConfiguratio9 === void 0 ? void 0 : _userListConfiguratio9.errorTextColor) || (0, _.fontHelper)(_theme === null || _theme === void 0 ? void 0 : (_theme$typography4 = _theme.typography) === null || _theme$typography4 === void 0 ? void 0 : _theme$typography4.heading),
      errorTextColor: ((_userListConfiguratio10 = _userListConfiguration.style) === null || _userListConfiguratio10 === void 0 ? void 0 : _userListConfiguratio10.errorTextFont) || (_theme === null || _theme === void 0 ? void 0 : (_theme$palette10 = _theme.palette) === null || _theme$palette10 === void 0 ? void 0 : _theme$palette10.getAccent400())
    };
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    style: (0, _style.containerStyle)(style, _theme),
    className: "cometchat__users"
  }, /*#__PURE__*/_react.default.createElement(_.CometChatListBase, {
    title: title,
    searchPlaceholder: searchInput ? searchInput : searchPlaceholder,
    onSearch: searchHandler,
    style: getListBaseStyle(),
    backButtonIconURL: backButtonIconURL || _back.default,
    searchIconURL: searchIconURL || _search.default,
    showBackButton: showBackButton,
    hideSearch: hideSearch
  }, /*#__PURE__*/_react.default.createElement(_.CometChatUserList, {
    ref: userListRef,
    limit: _userListConfiguration.limit,
    searchKeyword: searchInput ? searchInput : _userListConfiguration.searchKeyword,
    status: _userListConfiguration.status,
    friendsOnly: _userListConfiguration.friendsOnly,
    hideBlockedUsers: _userListConfiguration.hideBlockedUsers,
    tags: _userListConfiguration.tags,
    uids: _userListConfiguration.uids,
    roles: _userListConfiguration.roles,
    style: getListStyle(),
    customView: _userListConfiguration.customView,
    loadingIconURL: _userListConfiguration.loadingIconURL,
    hideError: _userListConfiguration.hideError,
    emptyText: (0, _.localize)("NO_USERS_FOUND"),
    errorText: (0, _.localize)("SOMETHING_WRONG"),
    activeUser: activeUser,
    theme: _theme,
    dataItemConfigurations: _userListConfiguration.dataItemConfigurations
  })));
});
Users.propTypes = {
  title: _propTypes.default.string,
  searchPlaceholder: _propTypes.default.string,
  activeUser: _propTypes.default.object,
  style: _propTypes.default.object,
  backButtonIconURL: _propTypes.default.string,
  searchIconURL: _propTypes.default.string,
  showBackButton: _propTypes.default.bool,
  hideSearch: _propTypes.default.bool,
  configurations: _propTypes.default.object
};
Users.defaultProps = {
  title: "Users",
  searchPlaceholder: "Search",
  activeUser: {},
  style: {
    width: "100%",
    height: "100%",
    background: "",
    border: "1px solid #808080",
    borderRadius: "8px",
    titleFont: "700 22px Inter, sans-serif",
    titleColor: "",
    backIconTint: "#3399FF",
    searchBorder: "1px solid #141414",
    searchborderRadius: "8px",
    searchBackground: "",
    searchTextFont: "",
    searchTextColor: "",
    searchIconTint: ""
  },
  backButtonIconURL: _back.default,
  searchIconURL: _search.default,
  showBackButton: false,
  hideSearch: false,
  configurations: {}
};

var CometChatUsers = /*#__PURE__*/_react.default.memo(Users);

exports.CometChatUsers = CometChatUsers;
