// cometchat-react-ui-kit/dist/Shared/Base/BaseStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BaseStyles = void 0;

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

/**
 * @class Styles
 * @description Styles class is used for defining the basic styles .
 * @param {String} width
 * @param {String} height
 * @param {String} background
 * @param {String} border
 * @param {String} borderRadius
 * @param {String} activeBackground
 */
var BaseStyles = /*#__PURE__*/(0, _createClass2.default)(function BaseStyles(_ref) {
  var _ref$width = _ref.width,
      width = _ref$width === void 0 ? "100%" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "100%" : _ref$height,
      _ref$background = _ref.background,
      background = _ref$background === void 0 ? "" : _ref$background,
      _ref$border = _ref.border,
      border = _ref$border === void 0 ? "" : _ref$border,
      _ref$borderRadius = _ref.borderRadius,
      borderRadius = _ref$borderRadius === void 0 ? "" : _ref$borderRadius,
      _ref$activeBackground = _ref.activeBackground,
      activeBackground = _ref$activeBackground === void 0 ? "" : _ref$activeBackground;
  (0, _classCallCheck2.default)(this, BaseStyles);
  this.width = width;
  this.height = height;
  this.background = background;
  this.border = border;
  this.borderRadius = borderRadius;
  this.activeBackground = activeBackground;
});
exports.BaseStyles = BaseStyles;
