// @cometchat-pro/react-ui-kit/dist/Shared/SDKDerivedComponents/CometChatDataItem/DataItemConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DataItemConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _Shared = require("../../../Shared");
/**
 * @class DataItemConfiguration
 * @description DataItemConfiguration class is used for defining the DataItem templates.
 * @param {Object} inputData
 * @param {Boolean} isActive
 */var DataItemConfiguration = /*#__PURE__*/(0, _createClass2["default"])(function DataItemConfiguration(_ref) {
  var _ref$inputData = _ref.inputData,
    inputData = _ref$inputData === void 0 ? new _Shared.InputData({}) : _ref$inputData,
    _ref$style = _ref.style,
    style = _ref$style === void 0 ? new _Shared.DataItemStyles({}) : _ref$style,
    _ref$avatarConfigurat = _ref.avatarConfiguration,
    avatarConfiguration = _ref$avatarConfigurat === void 0 ? new _Shared.AvatarConfiguration({}) : _ref$avatarConfigurat,
    _ref$statusIndicatorC = _ref.statusIndicatorConfiguration,
    statusIndicatorConfiguration = _ref$statusIndicatorC === void 0 ? new _Shared.StatusIndicatorConfiguration({}) : _ref$statusIndicatorC;
  (0, _classCallCheck2["default"])(this, DataItemConfiguration);
  this.inputData = new _Shared.InputData(inputData !== null && inputData !== void 0 ? inputData : {});
  this.style = new _Shared.DataItemStyles(style !== null && style !== void 0 ? style : {});
  this.avatarConfiguration = new _Shared.AvatarConfiguration(avatarConfiguration !== null && avatarConfiguration !== void 0 ? avatarConfiguration : {});
  this.statusIndicatorConfiguration = new _Shared.StatusIndicatorConfiguration(statusIndicatorConfiguration !== null && statusIndicatorConfiguration !== void 0 ? statusIndicatorConfiguration : {});
});
exports.DataItemConfiguration = DataItemConfiguration;
