// react-ui-kit-testing/dist/Shared/SecondaryComponents/CometChatMessageReceipt/MessageReceiptConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageReceiptConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _Shared = require("../../../Shared");
var _wait = _interopRequireDefault(require("./resources/wait.svg"));
var _messageSent = _interopRequireDefault(require("./resources/message-sent.svg"));
var _messageDelivered = _interopRequireDefault(require("./resources/message-delivered.svg"));
var _messageRead = _interopRequireDefault(require("./resources/message-read.svg"));
var _warningSmall = _interopRequireDefault(require("./resources/warning-small.svg"));
/**
 * @class MessageReceiptConfiguration
 * @param {Object} style
 * @param {String} messageWaitIcon
 * @param {String} messageSentIcon
 * @param {String} messageDeliveredIcon
 * @param {String} messageReadIcon
 * @param {String} messageErrorIcon
 */
var MessageReceiptConfiguration = /*#__PURE__*/(0, _createClass2.default)(function MessageReceiptConfiguration(_ref) {
  var _ref$style = _ref.style,
    style = _ref$style === void 0 ? new _Shared.MessageReceiptStyles({}) : _ref$style,
    _ref$messageWaitIcon = _ref.messageWaitIcon,
    messageWaitIcon = _ref$messageWaitIcon === void 0 ? _wait.default : _ref$messageWaitIcon,
    _ref$messageSentIcon = _ref.messageSentIcon,
    messageSentIcon = _ref$messageSentIcon === void 0 ? _messageSent.default : _ref$messageSentIcon,
    _ref$messageDelivered = _ref.messageDeliveredIcon,
    messageDeliveredIcon = _ref$messageDelivered === void 0 ? _messageDelivered.default : _ref$messageDelivered,
    _ref$messageReadIcon = _ref.messageReadIcon,
    messageReadIcon = _ref$messageReadIcon === void 0 ? _messageRead.default : _ref$messageReadIcon,
    _ref$messageErrorIcon = _ref.messageErrorIcon,
    messageErrorIcon = _ref$messageErrorIcon === void 0 ? _warningSmall.default : _ref$messageErrorIcon;
  (0, _classCallCheck2.default)(this, MessageReceiptConfiguration);
  this.style = new _Shared.MessageReceiptStyles(style !== null && style !== void 0 ? style : {});
  this.messageWaitIcon = messageWaitIcon;
  this.messageSentIcon = messageSentIcon;
  this.messageDeliveredIcon = messageDeliveredIcon;
  this.messageReadIcon = messageReadIcon;
  this.messageErrorIcon = messageErrorIcon;
});
exports.MessageReceiptConfiguration = MessageReceiptConfiguration;
