// react-ui-kit-testing/dist/Groups/CometChatCreateGroup/CreateGroupConfiguration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CreateGroupConfiguration = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _CreateGroupStyles = require("./CreateGroupStyles");
var _close = _interopRequireDefault(require("./resources/close.svg"));
/**
 * @class CreateGroupConfiguration
 * @description CreateGroupConfiguration class is used for defining the CreateGroup template.
 *
 * @param {boolean} hideCloseButton
 * @param {string} closeButtonIconURL
 * @param {function} onClose
 * @param {function} onCreateGroup
 * @param {object} style
 */
var CreateGroupConfiguration = /*#__PURE__*/(0, _createClass2.default)(function CreateGroupConfiguration(_ref) {
  var _ref$hideCloseButton = _ref.hideCloseButton,
    hideCloseButton = _ref$hideCloseButton === void 0 ? false : _ref$hideCloseButton,
    _ref$closeButtonIconU = _ref.closeButtonIconURL,
    closeButtonIconURL = _ref$closeButtonIconU === void 0 ? _close.default : _ref$closeButtonIconU,
    _ref$onClose = _ref.onClose,
    onClose = _ref$onClose === void 0 ? null : _ref$onClose,
    _ref$onCreateGroup = _ref.onCreateGroup,
    onCreateGroup = _ref$onCreateGroup === void 0 ? null : _ref$onCreateGroup,
    _ref$style = _ref.style,
    style = _ref$style === void 0 ? new _CreateGroupStyles.CreateGroupStyles({}) : _ref$style;
  (0, _classCallCheck2.default)(this, CreateGroupConfiguration);
  this.hideCloseButton = hideCloseButton;
  this.closeButtonIconURL = closeButtonIconURL;
  this.onClose = onClose;
  this.onCreateGroup = onCreateGroup;
  this.style = new _CreateGroupStyles.CreateGroupStyles(style || {});
});
exports.CreateGroupConfiguration = CreateGroupConfiguration;
