// react-ui-kit-testing/dist/Groups/CometChatCreateGroup/CreateGroupStyles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CreateGroupStyles = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _Shared = require("../../Shared");
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
 * @class CreateGroupStyles
 * @description CreateGroupStyles class is used for defining the styles for CreateGroup component
 * @param {string}width
 * @param {string} height
 * @param {string} background
 * @param {string} border
 * @param {string} borderRadius
 * @param {string} boxShadow
 * @param {string} closeIconTint
 * @param {string} titleTextFont
 * @param {string} titleTextColor
 * @param {string} errorTextFont
 * @param {string} errorTextBackground
 * @param {string} errorTextBorderRadius
 * @param {string} errorTextBorder
 * @param {string} errorTextColor
 * @param {string} groupTypeTextFont
 * @param {string} groupTypeTextColor
 * @param {string} groupTypeTextBackground
 * @param {string} groupTypeTextActiveBackground
 * @param {string} namePlaceholderTextFont
 * @param {string} namePlaceholderTextColor
 * @param {string} nameInputBackground
 * @param {string} nameInputBorder
 * @param {string} nameInputBorderRadius
 * @param {string} nameInputBoxShadow
 * @param {string} passwordPlaceholderTextFont
 * @param {string} passwordPlaceholderTextColor
 * @param {string} passwordInputBackground
 * @param {string} passwordInputBorder
 * @param {string} passwordInputBorderRadius
 * @param {string} passwordInputBoxShadow
 * @param {string} createGroupButtonTextFont
 * @param {string} createGroupButtonTextColor
 * @param {string} createGroupButtonBackground
 * @param {string} createGroupButtonBorderRadius
 */
var CreateGroupStyles = /*#__PURE__*/function (_BaseStyles) {
  (0, _inherits2.default)(CreateGroupStyles, _BaseStyles);
  var _super = _createSuper(CreateGroupStyles);
  function CreateGroupStyles(_ref) {
    var _this;
    var _ref$width = _ref.width,
      width = _ref$width === void 0 ? "" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "" : _ref$height,
      _ref$background = _ref.background,
      background = _ref$background === void 0 ? "" : _ref$background,
      _ref$border = _ref.border,
      border = _ref$border === void 0 ? "" : _ref$border,
      _ref$borderRadius = _ref.borderRadius,
      borderRadius = _ref$borderRadius === void 0 ? "" : _ref$borderRadius,
      _ref$boxShadow = _ref.boxShadow,
      boxShadow = _ref$boxShadow === void 0 ? "" : _ref$boxShadow,
      _ref$closeIconTint = _ref.closeIconTint,
      closeIconTint = _ref$closeIconTint === void 0 ? "" : _ref$closeIconTint,
      _ref$titleTextFont = _ref.titleTextFont,
      titleTextFont = _ref$titleTextFont === void 0 ? "" : _ref$titleTextFont,
      _ref$titleTextColor = _ref.titleTextColor,
      titleTextColor = _ref$titleTextColor === void 0 ? "" : _ref$titleTextColor,
      _ref$errorTextFont = _ref.errorTextFont,
      errorTextFont = _ref$errorTextFont === void 0 ? "" : _ref$errorTextFont,
      _ref$errorTextBackgro = _ref.errorTextBackground,
      errorTextBackground = _ref$errorTextBackgro === void 0 ? "" : _ref$errorTextBackgro,
      _ref$errorTextBorderR = _ref.errorTextBorderRadius,
      errorTextBorderRadius = _ref$errorTextBorderR === void 0 ? "" : _ref$errorTextBorderR,
      _ref$errorTextBorder = _ref.errorTextBorder,
      errorTextBorder = _ref$errorTextBorder === void 0 ? "" : _ref$errorTextBorder,
      _ref$errorTextColor = _ref.errorTextColor,
      errorTextColor = _ref$errorTextColor === void 0 ? "" : _ref$errorTextColor,
      _ref$groupTypeTextFon = _ref.groupTypeTextFont,
      groupTypeTextFont = _ref$groupTypeTextFon === void 0 ? "" : _ref$groupTypeTextFon,
      _ref$groupTypeTextCol = _ref.groupTypeTextColor,
      groupTypeTextColor = _ref$groupTypeTextCol === void 0 ? "" : _ref$groupTypeTextCol,
      _ref$groupTypeTextBac = _ref.groupTypeTextBackground,
      groupTypeTextBackground = _ref$groupTypeTextBac === void 0 ? "" : _ref$groupTypeTextBac,
      _ref$groupTypeTextAct = _ref.groupTypeTextActiveBackground,
      groupTypeTextActiveBackground = _ref$groupTypeTextAct === void 0 ? "" : _ref$groupTypeTextAct,
      _ref$namePlaceholderT = _ref.namePlaceholderTextFont,
      namePlaceholderTextFont = _ref$namePlaceholderT === void 0 ? "" : _ref$namePlaceholderT,
      _ref$namePlaceholderT2 = _ref.namePlaceholderTextColor,
      namePlaceholderTextColor = _ref$namePlaceholderT2 === void 0 ? "" : _ref$namePlaceholderT2,
      _ref$nameInputBackgro = _ref.nameInputBackground,
      nameInputBackground = _ref$nameInputBackgro === void 0 ? "" : _ref$nameInputBackgro,
      _ref$nameInputBorder = _ref.nameInputBorder,
      nameInputBorder = _ref$nameInputBorder === void 0 ? "" : _ref$nameInputBorder,
      _ref$nameInputBorderR = _ref.nameInputBorderRadius,
      nameInputBorderRadius = _ref$nameInputBorderR === void 0 ? "" : _ref$nameInputBorderR,
      _ref$nameInputBoxShad = _ref.nameInputBoxShadow,
      nameInputBoxShadow = _ref$nameInputBoxShad === void 0 ? "" : _ref$nameInputBoxShad,
      _ref$passwordPlacehol = _ref.passwordPlaceholderTextFont,
      passwordPlaceholderTextFont = _ref$passwordPlacehol === void 0 ? "" : _ref$passwordPlacehol,
      _ref$passwordPlacehol2 = _ref.passwordPlaceholderTextColor,
      passwordPlaceholderTextColor = _ref$passwordPlacehol2 === void 0 ? "" : _ref$passwordPlacehol2,
      _ref$passwordInputBac = _ref.passwordInputBackground,
      passwordInputBackground = _ref$passwordInputBac === void 0 ? "" : _ref$passwordInputBac,
      _ref$passwordInputBor = _ref.passwordInputBorder,
      passwordInputBorder = _ref$passwordInputBor === void 0 ? "" : _ref$passwordInputBor,
      _ref$passwordInputBor2 = _ref.passwordInputBorderRadius,
      passwordInputBorderRadius = _ref$passwordInputBor2 === void 0 ? "" : _ref$passwordInputBor2,
      _ref$passwordInputBox = _ref.passwordInputBoxShadow,
      passwordInputBoxShadow = _ref$passwordInputBox === void 0 ? "" : _ref$passwordInputBox,
      _ref$createGroupButto = _ref.createGroupButtonTextFont,
      createGroupButtonTextFont = _ref$createGroupButto === void 0 ? "" : _ref$createGroupButto,
      _ref$createGroupButto2 = _ref.createGroupButtonTextColor,
      createGroupButtonTextColor = _ref$createGroupButto2 === void 0 ? "" : _ref$createGroupButto2,
      _ref$createGroupButto3 = _ref.createGroupButtonBackground,
      createGroupButtonBackground = _ref$createGroupButto3 === void 0 ? "" : _ref$createGroupButto3,
      _ref$createGroupButto4 = _ref.createGroupButtonBorderRadius,
      createGroupButtonBorderRadius = _ref$createGroupButto4 === void 0 ? "" : _ref$createGroupButto4;
    (0, _classCallCheck2.default)(this, CreateGroupStyles);
    _this = _super.call(this, {
      width: width,
      height: height,
      background: background,
      border: border,
      borderRadius: borderRadius
    });
    _this.boxShadow = boxShadow;
    _this.closeIconTint = closeIconTint;
    _this.titleTextFont = titleTextFont;
    _this.titleTextColor = titleTextColor;
    _this.errorTextFont = errorTextFont;
    _this.errorTextBackground = errorTextBackground;
    _this.errorTextBorderRadius = errorTextBorderRadius;
    _this.errorTextBorder = errorTextBorder;
    _this.errorTextColor = errorTextColor;
    _this.groupTypeTextFont = groupTypeTextFont;
    _this.groupTypeTextColor = groupTypeTextColor;
    _this.groupTypeTextBackground = groupTypeTextBackground;
    _this.groupTypeTextActiveBackground = groupTypeTextActiveBackground;
    _this.namePlaceholderTextFont = namePlaceholderTextFont;
    _this.namePlaceholderTextColor = namePlaceholderTextColor;
    _this.nameInputBackground = nameInputBackground;
    _this.nameInputBorder = nameInputBorder;
    _this.nameInputBorderRadius = nameInputBorderRadius;
    _this.nameInputBoxShadow = nameInputBoxShadow;
    _this.passwordPlaceholderTextFont = passwordPlaceholderTextFont;
    _this.passwordPlaceholderTextColor = passwordPlaceholderTextColor;
    _this.passwordInputBackground = passwordInputBackground;
    _this.passwordInputBorder = passwordInputBorder;
    _this.passwordInputBorderRadius = passwordInputBorderRadius;
    _this.passwordInputBoxShadow = passwordInputBoxShadow;
    _this.createGroupButtonTextFont = createGroupButtonTextFont;
    _this.createGroupButtonTextColor = createGroupButtonTextColor;
    _this.createGroupButtonBackground = createGroupButtonBackground;
    _this.createGroupButtonBorderRadius = createGroupButtonBorderRadius;
    return _this;
  }
  return (0, _createClass2.default)(CreateGroupStyles);
}(_Shared.BaseStyles);
exports.CreateGroupStyles = CreateGroupStyles;
