// react-ui-kit-testing/dist/Groups/CometChatCreateGroup/style.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.passwordInputStyle = exports.nameInputStyle = exports.errorTextStyle = exports.errorIconStyle = exports.errorContainerStyle = exports.createGroupWrapperStyle = exports.createGroupTitleStyle = exports.createGroupTabStyle = exports.createGroupTabListStyle = exports.createGroupTabContainerStyle = exports.createGroupInputPassword = exports.createGroupInputName = exports.createGroupInput = exports.createGroupHeader = exports.createGroupButton = exports.createGroupBodyStyle = exports.createButtonStyle = exports.closeIconStyle = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _Shared = require("../../Shared");
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
var createGroupWrapperStyle = function createGroupWrapperStyle(style, theme) {
  return {
    width: (style === null || style === void 0 ? void 0 : style.width) || "100%",
    height: (style === null || style === void 0 ? void 0 : style.height) || "100%",
    border: (style === null || style === void 0 ? void 0 : style.border) || "1px solid ".concat(theme.palette.accent50[theme.palette.mode]),
    boxShadow: style.boxShadow || "".concat(theme.palette.accent50[theme.palette.mode], " 0 16px 32px, ").concat(theme.palette.accent50[theme.palette.mode], " 0 0 0 1px"),
    background: (style === null || style === void 0 ? void 0 : style.background) || theme.palette.background[theme.palette.mode],
    borderRadius: style === null || style === void 0 ? void 0 : style.borderRadius,
    overflow: "hidden",
    position: "fixed",
    margin: "auto",
    cursor: "default"
  };
};
exports.createGroupWrapperStyle = createGroupWrapperStyle;
var createGroupBodyStyle = function createGroupBodyStyle() {
  return {
    display: "flex",
    flexDirection: "column",
    position: "relative",
    width: "100%",
    height: "calc(100% - 65px)",
    padding: "16px"
  };
};
exports.createGroupBodyStyle = createGroupBodyStyle;
var closeIconStyle = function closeIconStyle(style, img, theme) {
  var _theme$palette;
  return {
    position: "absolute",
    width: "24px",
    height: "24px",
    top: "3%",
    right: "3%",
    WebkitMask: "url(".concat(img, ") center center no-repeat"),
    background: style.closeIconTint || (theme === null || theme === void 0 ? void 0 : (_theme$palette = theme.palette) === null || _theme$palette === void 0 ? void 0 : _theme$palette.getPrimary()),
    cursor: "pointer",
    zIndex: 1
  };
};
exports.closeIconStyle = closeIconStyle;
var createGroupHeader = function createGroupHeader() {
  return {
    padding: "20px"
  };
};
exports.createGroupHeader = createGroupHeader;
var createGroupTitleStyle = function createGroupTitleStyle(style, theme) {
  var _theme$typography, _theme$palette2;
  return {
    margin: "0",
    font: (style === null || style === void 0 ? void 0 : style.titleFont) || (0, _Shared.fontHelper)(theme === null || theme === void 0 ? void 0 : (_theme$typography = theme.typography) === null || _theme$typography === void 0 ? void 0 : _theme$typography.heading),
    color: (style === null || style === void 0 ? void 0 : style.titleColor) || (theme === null || theme === void 0 ? void 0 : (_theme$palette2 = theme.palette) === null || _theme$palette2 === void 0 ? void 0 : _theme$palette2.getAccent()),
    textAlign: "center"
  };
};
exports.createGroupTitleStyle = createGroupTitleStyle;
var createGroupTabContainerStyle = function createGroupTabContainerStyle(style) {
  return {
    borderRadius: "8px",
    width: style === null || style === void 0 ? void 0 : style.width,
    margin: "0"
  };
};
exports.createGroupTabContainerStyle = createGroupTabContainerStyle;
var createGroupTabListStyle = function createGroupTabListStyle(style, theme) {
  var _theme$palette3, _theme$typography2, _theme$palette4;
  return {
    height: "inherit",
    width: "fit-content",
    padding: "0",
    borderRadius: "8px",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    margin: "auto",
    background: (style === null || style === void 0 ? void 0 : style.groupTypeTextBackground) || (theme === null || theme === void 0 ? void 0 : (_theme$palette3 = theme.palette) === null || _theme$palette3 === void 0 ? void 0 : _theme$palette3.getAccent100()),
    font: (0, _Shared.fontHelper)(theme === null || theme === void 0 ? void 0 : (_theme$typography2 = theme.typography) === null || _theme$typography2 === void 0 ? void 0 : _theme$typography2.subtitle2) || "500 13px Inter,sans-serif",
    color: (theme === null || theme === void 0 ? void 0 : (_theme$palette4 = theme.palette) === null || _theme$palette4 === void 0 ? void 0 : _theme$palette4.getAccent()) || "rgb(20,20,20)"
  };
};
exports.createGroupTabListStyle = createGroupTabListStyle;
var createGroupTabStyle = function createGroupTabStyle(style, theme, activeTab, tab, isHovering) {
  var _theme$palette7;
  var activeTabStyle;
  if (activeTab === tab) {
    var _theme$palette5, _theme$palette6;
    activeTabStyle = {
      background: (style === null || style === void 0 ? void 0 : style.groupTypeTextActiveBackground) || (theme === null || theme === void 0 ? void 0 : (_theme$palette5 = theme.palette) === null || _theme$palette5 === void 0 ? void 0 : _theme$palette5.getBackground()),
      height: "28px",
      boxShadow: "".concat(theme === null || theme === void 0 ? void 0 : (_theme$palette6 = theme.palette) === null || _theme$palette6 === void 0 ? void 0 : _theme$palette6.getAccent50(), " 0px 0px 0px 1px"),
      borderRadius: "8px"
    };
  }
  return _objectSpread({
    width: "110px",
    textFont: style === null || style === void 0 ? void 0 : style.groupTypeTextFont,
    textColor: style === null || style === void 0 ? void 0 : style.groupTypeTextColor,
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    background: isHovering ? (style === null || style === void 0 ? void 0 : style.groupTypeTextActiveBackground) || (theme === null || theme === void 0 ? void 0 : (_theme$palette7 = theme.palette) === null || _theme$palette7 === void 0 ? void 0 : _theme$palette7.getBackground()) : "transparent"
  }, activeTabStyle);
};
exports.createGroupTabStyle = createGroupTabStyle;
var createGroupInput = function createGroupInput(style) {
  return {
    width: style === null || style === void 0 ? void 0 : style.width,
    border: "none",
    outline: "none"
  };
};
exports.createGroupInput = createGroupInput;
var createGroupInputName = function createGroupInputName() {
  return {
    margin: "20px 0px 8px 0px",
    padding: "4px 0px"
  };
};
exports.createGroupInputName = createGroupInputName;
var createGroupInputPassword = function createGroupInputPassword() {
  return {
    padding: "4px 0px"
  };
};
exports.createGroupInputPassword = createGroupInputPassword;
var nameInputStyle = function nameInputStyle(style, theme) {
  var _theme$palette8, _theme$palette9;
  return {
    background: (style === null || style === void 0 ? void 0 : style.nameInputBackground) || (theme === null || theme === void 0 ? void 0 : (_theme$palette8 = theme.palette) === null || _theme$palette8 === void 0 ? void 0 : _theme$palette8.getAccent50()),
    border: (style === null || style === void 0 ? void 0 : style.nameInputBorder) || "none",
    borderRadius: (style === null || style === void 0 ? void 0 : style.nameInputBorderRadius) || "8px",
    boxShadow: (style === null || style === void 0 ? void 0 : style.nameInputBoxShadow) || "".concat(theme === null || theme === void 0 ? void 0 : (_theme$palette9 = theme.palette) === null || _theme$palette9 === void 0 ? void 0 : _theme$palette9.getAccent50(), " 0px 0px 0px 1px"),
    width: style === null || style === void 0 ? void 0 : style.width,
    height: "46px",
    outline: "none",
    padding: "6px 12px"
  };
};
exports.nameInputStyle = nameInputStyle;
var passwordInputStyle = function passwordInputStyle(style, theme) {
  var _theme$palette10, _theme$palette11;
  return {
    background: (style === null || style === void 0 ? void 0 : style.passwordInputBackground) || (theme === null || theme === void 0 ? void 0 : (_theme$palette10 = theme.palette) === null || _theme$palette10 === void 0 ? void 0 : _theme$palette10.getAccent50()),
    border: (style === null || style === void 0 ? void 0 : style.passwordInputBorder) || "none",
    borderRadius: (style === null || style === void 0 ? void 0 : style.passwordInputBorderRadius) || "8px",
    boxShadow: (style === null || style === void 0 ? void 0 : style.passInputBoxShadow) || "".concat(theme === null || theme === void 0 ? void 0 : (_theme$palette11 = theme.palette) === null || _theme$palette11 === void 0 ? void 0 : _theme$palette11.getAccent50(), " 0px 0px 0px 1px"),
    width: style === null || style === void 0 ? void 0 : style.width,
    height: "46px",
    outline: "none",
    padding: "6px 12px"
  };
};
exports.passwordInputStyle = passwordInputStyle;
var errorContainerStyle = function errorContainerStyle(style, theme) {
  return {
    width: style === null || style === void 0 ? void 0 : style.width,
    padding: "10px 0px",
    marginTop: "16px",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    background: (style === null || style === void 0 ? void 0 : style.errorTextBackground) || "rgba(255, 59, 48, 0.1)",
    border: (style === null || style === void 0 ? void 0 : style.errorTextBorder) || "none",
    borderRadius: (style === null || style === void 0 ? void 0 : style.errorTextBorderRadius) || "8px"
  };
};
exports.errorContainerStyle = errorContainerStyle;
var errorIconStyle = function errorIconStyle(theme) {
  var _theme$palette12, _theme$palette13;
  return {
    iconBackground: (theme === null || theme === void 0 ? void 0 : (_theme$palette12 = theme.palette) === null || _theme$palette12 === void 0 ? void 0 : _theme$palette12.getError()) || "RGB(255, 59, 48)",
    iconTint: theme === null || theme === void 0 ? void 0 : (_theme$palette13 = theme.palette) === null || _theme$palette13 === void 0 ? void 0 : _theme$palette13.getBackground("light"),
    iconBackgroundWidth: "38px",
    iconBackgroundHeight: "38px",
    iconWidth: "24px",
    iconHeight: "24px",
    margin: "0px 12px"
  };
};
exports.errorIconStyle = errorIconStyle;
var errorTextStyle = function errorTextStyle(style, theme) {
  var _theme$palette14, _theme$typography3;
  return {
    color: (style === null || style === void 0 ? void 0 : style.errorTextColor) || (theme === null || theme === void 0 ? void 0 : (_theme$palette14 = theme.palette) === null || _theme$palette14 === void 0 ? void 0 : _theme$palette14.getError()),
    font: (style === null || style === void 0 ? void 0 : style.errorTextFont) || (0, _Shared.fontHelper)(theme === null || theme === void 0 ? void 0 : (_theme$typography3 = theme.typography) === null || _theme$typography3 === void 0 ? void 0 : _theme$typography3.text1)
  };
};
exports.errorTextStyle = errorTextStyle;
var createGroupButton = function createGroupButton(style) {
  return {
    display: "inline-block",
    width: "calc(100% - 30px)",
    position: "absolute",
    bottom: "0",
    right: "0",
    left: "0",
    margin: "16px"
  };
};
exports.createGroupButton = createGroupButton;
var createButtonStyle = function createButtonStyle(style, theme) {
  var _theme$typography4, _theme$palette15, _theme$palette16;
  return {
    width: style === null || style === void 0 ? void 0 : style.width,
    height: "46px",
    outline: "0",
    borderRadius: (style === null || style === void 0 ? void 0 : style.createGroupButtonBorderRadius) || "8px",
    font: (style === null || style === void 0 ? void 0 : style.createGroupButtonTextFont) || (0, _Shared.fontHelper)(theme === null || theme === void 0 ? void 0 : (_theme$typography4 = theme.typography) === null || _theme$typography4 === void 0 ? void 0 : _theme$typography4.title2),
    color: (style === null || style === void 0 ? void 0 : style.createGroupButtonTextColor) || (theme === null || theme === void 0 ? void 0 : (_theme$palette15 = theme.palette) === null || _theme$palette15 === void 0 ? void 0 : _theme$palette15.getAccent900("light")),
    background: (style === null || style === void 0 ? void 0 : style.createGroupButtonBackground) || (theme === null || theme === void 0 ? void 0 : (_theme$palette16 = theme.palette) === null || _theme$palette16 === void 0 ? void 0 : _theme$palette16.getPrimary()),
    border: "none"
  };
};
exports.createButtonStyle = createButtonStyle;
