// cometchat-react-ui-kit/dist/Messages/CometChatMessagePreview/style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.previewTitleStyle = exports.previewSubTitleStyle = exports.previewHeadingStyle = exports.previewCloseStyle = exports.editPreviewContainerStyle = void 0;

var _ = require("../../..");

var editPreviewContainerStyle = function editPreviewContainerStyle(props, theme) {
  return {
    padding: "5px",
    margin: "0 10px 5px 5px",
    width: props.style.width,
    height: props.style.height,
    zIndex: "12",
    borderLeft: props.style.border || theme.palette.accent100[theme.palette.mode],
    background: props.style.messagePreviewBackground || theme.palette.background[theme.palette.mode]
  };
};

exports.editPreviewContainerStyle = editPreviewContainerStyle;

var previewHeadingStyle = function previewHeadingStyle() {
  return {
    display: "flex",
    marginBottom: "5px",
    alignItems: "center",
    justifyContent: "space-between"
  };
};

exports.previewHeadingStyle = previewHeadingStyle;

var previewTitleStyle = function previewTitleStyle(props, theme) {
  return {
    font: props.style.messagePreviewTitleFont || (0, _.fontHelper)(theme.typography.caption1),
    color: props.style.messagePreviewTitleColor || theme.palette.accent[theme.palette.mode],
    lineHeight: "1.33",
    letterSpacing: ".5px"
  };
};

exports.previewTitleStyle = previewTitleStyle;

var previewSubTitleStyle = function previewSubTitleStyle(props, theme) {
  return {
    font: props.style.messagePreviewSubtitleFont || (0, _.fontHelper)(theme.typography.subtitle2),
    color: props.style.messagePreviewSubtitleColor || theme.palette.accent600[theme.palette.mode],
    lineHeight: "1.38",
    letterSpacing: ".5px"
  };
};

exports.previewSubTitleStyle = previewSubTitleStyle;

var previewCloseStyle = function previewCloseStyle(props, theme) {
  return {
    width: "16px",
    height: "16px",
    cursor: "pointer",
    borderRadius: "50%",
    WebkitMask: "url(".concat(props.closeIconURL, ") center center no-repeat"),
    backgroundColor: props.style.closeIconTint || theme.palette.accent500[theme.palette.mode]
  };
};

exports.previewCloseStyle = previewCloseStyle;
